// @aws-sdk/client-iot/dist-cjs/commands/DescribeThingRegistrationTaskCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeThingRegistrationTaskCommand = void 0;
const middleware_serde_1 = require("@aws-sdk/middleware-serde");
const smithy_client_1 = require("@aws-sdk/smithy-client");
const models_1_1 = require("../models/models_1");
const Aws_restJson1_1 = require("../protocols/Aws_restJson1");
class DescribeThingRegistrationTaskCommand extends smithy_client_1.Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(middleware_serde_1.getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "IoTClient";
        const commandName = "DescribeThingRegistrationTaskCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: models_1_1.DescribeThingRegistrationTaskRequest.filterSensitiveLog,
            outputFilterSensitiveLog: models_1_1.DescribeThingRegistrationTaskResponse.filterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return Aws_restJson1_1.serializeAws_restJson1DescribeThingRegistrationTaskCommand(input, context);
    }
    deserialize(output, context) {
        return Aws_restJson1_1.deserializeAws_restJson1DescribeThingRegistrationTaskCommand(output, context);
    }
}
exports.DescribeThingRegistrationTaskCommand = DescribeThingRegistrationTaskCommand;
