// @aws-sdk/client-iot/dist-cjs/commands/PutVerificationStateOnViolationCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PutVerificationStateOnViolationCommand = void 0;
const middleware_serde_1 = require("@aws-sdk/middleware-serde");
const smithy_client_1 = require("@aws-sdk/smithy-client");
const models_2_1 = require("../models/models_2");
const Aws_restJson1_1 = require("../protocols/Aws_restJson1");
class PutVerificationStateOnViolationCommand extends smithy_client_1.Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(middleware_serde_1.getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "IoTClient";
        const commandName = "PutVerificationStateOnViolationCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: models_2_1.PutVerificationStateOnViolationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: models_2_1.PutVerificationStateOnViolationResponse.filterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return Aws_restJson1_1.serializeAws_restJson1PutVerificationStateOnViolationCommand(input, context);
    }
    deserialize(output, context) {
        return Aws_restJson1_1.deserializeAws_restJson1PutVerificationStateOnViolationCommand(output, context);
    }
}
exports.PutVerificationStateOnViolationCommand = PutVerificationStateOnViolationCommand;
