// @aws-sdk/client-iot/dist-cjs/models/models_1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeCustomMetricResponse = exports.DescribeCustomMetricRequest = exports.DescribeCertificateResponse = exports.CertificateDescription = exports.TransferData = exports.CertificateStatus = exports.CertificateMode = exports.DescribeCertificateRequest = exports.DescribeCACertificateResponse = exports.RegistrationConfig = exports.CACertificateDescription = exports.CertificateValidity = exports.CACertificateStatus = exports.DescribeCACertificateRequest = exports.DescribeBillingGroupResponse = exports.BillingGroupMetadata = exports.DescribeBillingGroupRequest = exports.DescribeAuthorizerResponse = exports.DescribeAuthorizerRequest = exports.DescribeAuditTaskResponse = exports.TaskStatistics = exports.DescribeAuditTaskRequest = exports.DescribeAuditSuppressionResponse = exports.DescribeAuditSuppressionRequest = exports.DescribeAuditMitigationActionsTaskResponse = exports.MitigationAction = exports.DescribeAuditMitigationActionsTaskRequest = exports.DescribeAuditFindingResponse = exports.DescribeAuditFindingRequest = exports.DescribeAccountAuditConfigurationResponse = exports.DescribeAccountAuditConfigurationRequest = exports.DeprecateThingTypeResponse = exports.DeprecateThingTypeRequest = exports.DeleteV2LoggingLevelRequest = exports.LogTargetType = exports.DeleteTopicRuleDestinationResponse = exports.DeleteTopicRuleDestinationRequest = exports.DeleteTopicRuleRequest = exports.DeleteThingTypeResponse = exports.DeleteThingTypeRequest = exports.DeleteThingGroupResponse = exports.DeleteThingGroupRequest = exports.DeleteThingResponse = exports.DeleteThingRequest = exports.DeleteStreamResponse = exports.DeleteStreamRequest = exports.DeleteSecurityProfileResponse = exports.DeleteSecurityProfileRequest = exports.DeleteScheduledAuditResponse = exports.DeleteScheduledAuditRequest = void 0;
exports.DescribeProvisioningTemplateVersionResponse = exports.DescribeProvisioningTemplateVersionRequest = exports.DescribeProvisioningTemplateResponse = exports.DescribeProvisioningTemplateRequest = exports.DescribeMitigationActionResponse = exports.MitigationActionType = exports.DescribeMitigationActionRequest = exports.DescribeManagedJobTemplateResponse = exports.DocumentParameter = exports.DescribeManagedJobTemplateRequest = exports.DescribeJobTemplateResponse = exports.DescribeJobTemplateRequest = exports.DescribeJobExecutionResponse = exports.JobExecution = exports.JobExecutionStatusDetails = exports.JobExecutionStatus = exports.DescribeJobExecutionRequest = exports.DescribeJobResponse = exports.Job = exports.JobStatus = exports.JobProcessDetails = exports.DescribeJobRequest = exports.DescribeIndexResponse = exports.IndexStatus = exports.DescribeIndexRequest = exports.DescribeFleetMetricResponse = exports.DescribeFleetMetricRequest = exports.DescribeEventConfigurationsResponse = exports.Configuration = exports.EventType = exports.DescribeEventConfigurationsRequest = exports.DescribeEndpointResponse = exports.DescribeEndpointRequest = exports.DescribeDomainConfigurationResponse = exports.ServerCertificateSummary = exports.ServerCertificateStatus = exports.DomainType = exports.DomainConfigurationStatus = exports.DescribeDomainConfigurationRequest = exports.DescribeDimensionResponse = exports.DescribeDimensionRequest = exports.DescribeDetectMitigationActionsTaskResponse = exports.DetectMitigationActionsTaskSummary = exports.ViolationEventOccurrenceRange = exports.DetectMitigationActionsTaskStatus = exports.DetectMitigationActionsTaskStatistics = exports.DetectMitigationActionsTaskTarget = exports.DescribeDetectMitigationActionsTaskRequest = exports.DescribeDefaultAuthorizerResponse = exports.DescribeDefaultAuthorizerRequest = void 0;
exports.ThingGroupIndexingConfiguration = exports.ThingGroupIndexingMode = exports.Field = exports.FieldType = exports.GetIndexingConfigurationRequest = exports.GetEffectivePoliciesResponse = exports.EffectivePolicy = exports.GetEffectivePoliciesRequest = exports.GetCardinalityResponse = exports.GetCardinalityRequest = exports.GetBucketsAggregationResponse = exports.Bucket = exports.GetBucketsAggregationRequest = exports.BucketsAggregationType = exports.TermsAggregation = exports.GetBehaviorModelTrainingSummariesResponse = exports.BehaviorModelTrainingSummary = exports.ModelStatus = exports.GetBehaviorModelTrainingSummariesRequest = exports.EnableTopicRuleRequest = exports.DisableTopicRuleRequest = exports.DetachThingPrincipalResponse = exports.DetachThingPrincipalRequest = exports.DetachSecurityProfileResponse = exports.DetachSecurityProfileRequest = exports.DetachPrincipalPolicyRequest = exports.DetachPolicyRequest = exports.DescribeThingTypeResponse = exports.ThingTypeMetadata = exports.DescribeThingTypeRequest = exports.DescribeThingRegistrationTaskResponse = exports.Status = exports.DescribeThingRegistrationTaskRequest = exports.DescribeThingGroupResponse = exports.ThingGroupMetadata = exports.GroupNameAndArn = exports.DynamicGroupStatus = exports.DescribeThingGroupRequest = exports.DescribeThingResponse = exports.DescribeThingRequest = exports.DescribeStreamResponse = exports.StreamInfo = exports.DescribeStreamRequest = exports.DescribeSecurityProfileResponse = exports.DescribeSecurityProfileRequest = exports.DescribeScheduledAuditResponse = exports.DescribeScheduledAuditRequest = exports.DescribeRoleAliasResponse = exports.RoleAliasDescription = exports.DescribeRoleAliasRequest = void 0;
exports.ListAuthorizersResponse = exports.ListAuthorizersRequest = exports.ListAuditTasksResponse = exports.ListAuditTasksRequest = exports.ListAuditSuppressionsResponse = exports.ListAuditSuppressionsRequest = exports.ListAuditMitigationActionsTasksResponse = exports.ListAuditMitigationActionsTasksRequest = exports.ListAuditMitigationActionsExecutionsResponse = exports.ListAuditMitigationActionsExecutionsRequest = exports.ListAuditFindingsResponse = exports.ListAuditFindingsRequest = exports.ListAttachedPoliciesResponse = exports.ListAttachedPoliciesRequest = exports.ListActiveViolationsResponse = exports.ListActiveViolationsRequest = exports.BehaviorCriteriaType = exports.GetV2LoggingOptionsResponse = exports.GetV2LoggingOptionsRequest = exports.GetTopicRuleDestinationResponse = exports.GetTopicRuleDestinationRequest = exports.GetTopicRuleResponse = exports.TopicRule = exports.GetTopicRuleRequest = exports.GetStatisticsResponse = exports.Statistics = exports.GetStatisticsRequest = exports.GetRegistrationCodeResponse = exports.GetRegistrationCodeRequest = exports.GetPolicyVersionResponse = exports.GetPolicyVersionRequest = exports.GetPolicyResponse = exports.GetPolicyRequest = exports.GetPercentilesResponse = exports.PercentPair = exports.GetPercentilesRequest = exports.GetOTAUpdateResponse = exports.OTAUpdateInfo = exports.ErrorInfo = exports.GetOTAUpdateRequest = exports.GetLoggingOptionsResponse = exports.GetLoggingOptionsRequest = exports.GetJobDocumentResponse = exports.GetJobDocumentRequest = exports.GetIndexingConfigurationResponse = exports.ThingIndexingConfiguration = exports.ThingIndexingMode = exports.ThingConnectivityIndexingMode = exports.NamedShadowIndexingMode = exports.DeviceDefenderIndexingMode = void 0;
exports.ListOTAUpdatesResponse = exports.OTAUpdateSummary = exports.ListOTAUpdatesRequest = exports.ListMitigationActionsResponse = exports.MitigationActionIdentifier = exports.ListMitigationActionsRequest = exports.ListManagedJobTemplatesResponse = exports.ManagedJobTemplateSummary = exports.ListManagedJobTemplatesRequest = exports.ListJobTemplatesResponse = exports.JobTemplateSummary = exports.ListJobTemplatesRequest = exports.ListJobsResponse = exports.JobSummary = exports.ListJobsRequest = exports.ListJobExecutionsForThingResponse = exports.JobExecutionSummaryForThing = exports.ListJobExecutionsForThingRequest = exports.ListJobExecutionsForJobResponse = exports.JobExecutionSummaryForJob = exports.JobExecutionSummary = exports.ListJobExecutionsForJobRequest = exports.ListIndicesResponse = exports.ListIndicesRequest = exports.ListFleetMetricsResponse = exports.FleetMetricNameAndArn = exports.ListFleetMetricsRequest = exports.ListDomainConfigurationsResponse = exports.DomainConfigurationSummary = exports.ListDomainConfigurationsRequest = exports.ListDimensionsResponse = exports.ListDimensionsRequest = exports.ListDetectMitigationActionsTasksResponse = exports.ListDetectMitigationActionsTasksRequest = exports.ListDetectMitigationActionsExecutionsResponse = exports.DetectMitigationActionExecution = exports.DetectMitigationActionExecutionStatus = exports.ListDetectMitigationActionsExecutionsRequest = exports.ListCustomMetricsResponse = exports.ListCustomMetricsRequest = exports.ListCertificatesByCAResponse = exports.ListCertificatesByCARequest = exports.ListCertificatesResponse = exports.Certificate = exports.ListCertificatesRequest = exports.ListCACertificatesResponse = exports.CACertificate = exports.ListCACertificatesRequest = exports.ListBillingGroupsResponse = exports.ListBillingGroupsRequest = void 0;
exports.ReportType = exports.ListThingPrincipalsResponse = exports.ListThingPrincipalsRequest = exports.ListThingGroupsForThingResponse = exports.ListThingGroupsForThingRequest = exports.ListThingGroupsResponse = exports.ListThingGroupsRequest = exports.ListTargetsForSecurityProfileResponse = exports.ListTargetsForSecurityProfileRequest = exports.ListTargetsForPolicyResponse = exports.ListTargetsForPolicyRequest = exports.ListTagsForResourceResponse = exports.ListTagsForResourceRequest = exports.ListStreamsResponse = exports.StreamSummary = exports.ListStreamsRequest = exports.ListSecurityProfilesForTargetResponse = exports.SecurityProfileTargetMapping = exports.SecurityProfileTarget = exports.ListSecurityProfilesForTargetRequest = exports.ListSecurityProfilesResponse = exports.SecurityProfileIdentifier = exports.ListSecurityProfilesRequest = exports.ListScheduledAuditsResponse = exports.ScheduledAuditMetadata = exports.ListScheduledAuditsRequest = exports.ListRoleAliasesResponse = exports.ListRoleAliasesRequest = exports.ListProvisioningTemplateVersionsResponse = exports.ProvisioningTemplateVersionSummary = exports.ListProvisioningTemplateVersionsRequest = exports.ListProvisioningTemplatesResponse = exports.ProvisioningTemplateSummary = exports.ListProvisioningTemplatesRequest = exports.ListPrincipalThingsResponse = exports.ListPrincipalThingsRequest = exports.ListPrincipalPoliciesResponse = exports.ListPrincipalPoliciesRequest = exports.ListPolicyVersionsResponse = exports.PolicyVersion = exports.ListPolicyVersionsRequest = exports.ListPolicyPrincipalsResponse = exports.ListPolicyPrincipalsRequest = exports.ListPoliciesResponse = exports.ListPoliciesRequest = exports.ListOutgoingCertificatesResponse = exports.OutgoingCertificate = exports.ListOutgoingCertificatesRequest = void 0;
var DeleteScheduledAuditRequest;
(function (DeleteScheduledAuditRequest) {
    DeleteScheduledAuditRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteScheduledAuditRequest = exports.DeleteScheduledAuditRequest || (exports.DeleteScheduledAuditRequest = {}));
var DeleteScheduledAuditResponse;
(function (DeleteScheduledAuditResponse) {
    DeleteScheduledAuditResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteScheduledAuditResponse = exports.DeleteScheduledAuditResponse || (exports.DeleteScheduledAuditResponse = {}));
var DeleteSecurityProfileRequest;
(function (DeleteSecurityProfileRequest) {
    DeleteSecurityProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteSecurityProfileRequest = exports.DeleteSecurityProfileRequest || (exports.DeleteSecurityProfileRequest = {}));
var DeleteSecurityProfileResponse;
(function (DeleteSecurityProfileResponse) {
    DeleteSecurityProfileResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteSecurityProfileResponse = exports.DeleteSecurityProfileResponse || (exports.DeleteSecurityProfileResponse = {}));
var DeleteStreamRequest;
(function (DeleteStreamRequest) {
    DeleteStreamRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteStreamRequest = exports.DeleteStreamRequest || (exports.DeleteStreamRequest = {}));
var DeleteStreamResponse;
(function (DeleteStreamResponse) {
    DeleteStreamResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteStreamResponse = exports.DeleteStreamResponse || (exports.DeleteStreamResponse = {}));
var DeleteThingRequest;
(function (DeleteThingRequest) {
    DeleteThingRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteThingRequest = exports.DeleteThingRequest || (exports.DeleteThingRequest = {}));
var DeleteThingResponse;
(function (DeleteThingResponse) {
    DeleteThingResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteThingResponse = exports.DeleteThingResponse || (exports.DeleteThingResponse = {}));
var DeleteThingGroupRequest;
(function (DeleteThingGroupRequest) {
    DeleteThingGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteThingGroupRequest = exports.DeleteThingGroupRequest || (exports.DeleteThingGroupRequest = {}));
var DeleteThingGroupResponse;
(function (DeleteThingGroupResponse) {
    DeleteThingGroupResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteThingGroupResponse = exports.DeleteThingGroupResponse || (exports.DeleteThingGroupResponse = {}));
var DeleteThingTypeRequest;
(function (DeleteThingTypeRequest) {
    DeleteThingTypeRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteThingTypeRequest = exports.DeleteThingTypeRequest || (exports.DeleteThingTypeRequest = {}));
var DeleteThingTypeResponse;
(function (DeleteThingTypeResponse) {
    DeleteThingTypeResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteThingTypeResponse = exports.DeleteThingTypeResponse || (exports.DeleteThingTypeResponse = {}));
var DeleteTopicRuleRequest;
(function (DeleteTopicRuleRequest) {
    DeleteTopicRuleRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteTopicRuleRequest = exports.DeleteTopicRuleRequest || (exports.DeleteTopicRuleRequest = {}));
var DeleteTopicRuleDestinationRequest;
(function (DeleteTopicRuleDestinationRequest) {
    DeleteTopicRuleDestinationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteTopicRuleDestinationRequest = exports.DeleteTopicRuleDestinationRequest || (exports.DeleteTopicRuleDestinationRequest = {}));
var DeleteTopicRuleDestinationResponse;
(function (DeleteTopicRuleDestinationResponse) {
    DeleteTopicRuleDestinationResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteTopicRuleDestinationResponse = exports.DeleteTopicRuleDestinationResponse || (exports.DeleteTopicRuleDestinationResponse = {}));
var LogTargetType;
(function (LogTargetType) {
    LogTargetType["DEFAULT"] = "DEFAULT";
    LogTargetType["THING_GROUP"] = "THING_GROUP";
})(LogTargetType = exports.LogTargetType || (exports.LogTargetType = {}));
var DeleteV2LoggingLevelRequest;
(function (DeleteV2LoggingLevelRequest) {
    DeleteV2LoggingLevelRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteV2LoggingLevelRequest = exports.DeleteV2LoggingLevelRequest || (exports.DeleteV2LoggingLevelRequest = {}));
var DeprecateThingTypeRequest;
(function (DeprecateThingTypeRequest) {
    DeprecateThingTypeRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeprecateThingTypeRequest = exports.DeprecateThingTypeRequest || (exports.DeprecateThingTypeRequest = {}));
var DeprecateThingTypeResponse;
(function (DeprecateThingTypeResponse) {
    DeprecateThingTypeResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeprecateThingTypeResponse = exports.DeprecateThingTypeResponse || (exports.DeprecateThingTypeResponse = {}));
var DescribeAccountAuditConfigurationRequest;
(function (DescribeAccountAuditConfigurationRequest) {
    DescribeAccountAuditConfigurationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAccountAuditConfigurationRequest = exports.DescribeAccountAuditConfigurationRequest || (exports.DescribeAccountAuditConfigurationRequest = {}));
var DescribeAccountAuditConfigurationResponse;
(function (DescribeAccountAuditConfigurationResponse) {
    DescribeAccountAuditConfigurationResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAccountAuditConfigurationResponse = exports.DescribeAccountAuditConfigurationResponse || (exports.DescribeAccountAuditConfigurationResponse = {}));
var DescribeAuditFindingRequest;
(function (DescribeAuditFindingRequest) {
    DescribeAuditFindingRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAuditFindingRequest = exports.DescribeAuditFindingRequest || (exports.DescribeAuditFindingRequest = {}));
var DescribeAuditFindingResponse;
(function (DescribeAuditFindingResponse) {
    DescribeAuditFindingResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAuditFindingResponse = exports.DescribeAuditFindingResponse || (exports.DescribeAuditFindingResponse = {}));
var DescribeAuditMitigationActionsTaskRequest;
(function (DescribeAuditMitigationActionsTaskRequest) {
    DescribeAuditMitigationActionsTaskRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAuditMitigationActionsTaskRequest = exports.DescribeAuditMitigationActionsTaskRequest || (exports.DescribeAuditMitigationActionsTaskRequest = {}));
var MitigationAction;
(function (MitigationAction) {
    MitigationAction.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(MitigationAction = exports.MitigationAction || (exports.MitigationAction = {}));
var DescribeAuditMitigationActionsTaskResponse;
(function (DescribeAuditMitigationActionsTaskResponse) {
    DescribeAuditMitigationActionsTaskResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAuditMitigationActionsTaskResponse = exports.DescribeAuditMitigationActionsTaskResponse || (exports.DescribeAuditMitigationActionsTaskResponse = {}));
var DescribeAuditSuppressionRequest;
(function (DescribeAuditSuppressionRequest) {
    DescribeAuditSuppressionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAuditSuppressionRequest = exports.DescribeAuditSuppressionRequest || (exports.DescribeAuditSuppressionRequest = {}));
var DescribeAuditSuppressionResponse;
(function (DescribeAuditSuppressionResponse) {
    DescribeAuditSuppressionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAuditSuppressionResponse = exports.DescribeAuditSuppressionResponse || (exports.DescribeAuditSuppressionResponse = {}));
var DescribeAuditTaskRequest;
(function (DescribeAuditTaskRequest) {
    DescribeAuditTaskRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAuditTaskRequest = exports.DescribeAuditTaskRequest || (exports.DescribeAuditTaskRequest = {}));
var TaskStatistics;
(function (TaskStatistics) {
    TaskStatistics.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TaskStatistics = exports.TaskStatistics || (exports.TaskStatistics = {}));
var DescribeAuditTaskResponse;
(function (DescribeAuditTaskResponse) {
    DescribeAuditTaskResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAuditTaskResponse = exports.DescribeAuditTaskResponse || (exports.DescribeAuditTaskResponse = {}));
var DescribeAuthorizerRequest;
(function (DescribeAuthorizerRequest) {
    DescribeAuthorizerRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAuthorizerRequest = exports.DescribeAuthorizerRequest || (exports.DescribeAuthorizerRequest = {}));
var DescribeAuthorizerResponse;
(function (DescribeAuthorizerResponse) {
    DescribeAuthorizerResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAuthorizerResponse = exports.DescribeAuthorizerResponse || (exports.DescribeAuthorizerResponse = {}));
var DescribeBillingGroupRequest;
(function (DescribeBillingGroupRequest) {
    DescribeBillingGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeBillingGroupRequest = exports.DescribeBillingGroupRequest || (exports.DescribeBillingGroupRequest = {}));
var BillingGroupMetadata;
(function (BillingGroupMetadata) {
    BillingGroupMetadata.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(BillingGroupMetadata = exports.BillingGroupMetadata || (exports.BillingGroupMetadata = {}));
var DescribeBillingGroupResponse;
(function (DescribeBillingGroupResponse) {
    DescribeBillingGroupResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeBillingGroupResponse = exports.DescribeBillingGroupResponse || (exports.DescribeBillingGroupResponse = {}));
var DescribeCACertificateRequest;
(function (DescribeCACertificateRequest) {
    DescribeCACertificateRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeCACertificateRequest = exports.DescribeCACertificateRequest || (exports.DescribeCACertificateRequest = {}));
var CACertificateStatus;
(function (CACertificateStatus) {
    CACertificateStatus["ACTIVE"] = "ACTIVE";
    CACertificateStatus["INACTIVE"] = "INACTIVE";
})(CACertificateStatus = exports.CACertificateStatus || (exports.CACertificateStatus = {}));
var CertificateValidity;
(function (CertificateValidity) {
    CertificateValidity.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CertificateValidity = exports.CertificateValidity || (exports.CertificateValidity = {}));
var CACertificateDescription;
(function (CACertificateDescription) {
    CACertificateDescription.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CACertificateDescription = exports.CACertificateDescription || (exports.CACertificateDescription = {}));
var RegistrationConfig;
(function (RegistrationConfig) {
    RegistrationConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RegistrationConfig = exports.RegistrationConfig || (exports.RegistrationConfig = {}));
var DescribeCACertificateResponse;
(function (DescribeCACertificateResponse) {
    DescribeCACertificateResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeCACertificateResponse = exports.DescribeCACertificateResponse || (exports.DescribeCACertificateResponse = {}));
var DescribeCertificateRequest;
(function (DescribeCertificateRequest) {
    DescribeCertificateRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeCertificateRequest = exports.DescribeCertificateRequest || (exports.DescribeCertificateRequest = {}));
var CertificateMode;
(function (CertificateMode) {
    CertificateMode["DEFAULT"] = "DEFAULT";
    CertificateMode["SNI_ONLY"] = "SNI_ONLY";
})(CertificateMode = exports.CertificateMode || (exports.CertificateMode = {}));
var CertificateStatus;
(function (CertificateStatus) {
    CertificateStatus["ACTIVE"] = "ACTIVE";
    CertificateStatus["INACTIVE"] = "INACTIVE";
    CertificateStatus["PENDING_ACTIVATION"] = "PENDING_ACTIVATION";
    CertificateStatus["PENDING_TRANSFER"] = "PENDING_TRANSFER";
    CertificateStatus["REGISTER_INACTIVE"] = "REGISTER_INACTIVE";
    CertificateStatus["REVOKED"] = "REVOKED";
})(CertificateStatus = exports.CertificateStatus || (exports.CertificateStatus = {}));
var TransferData;
(function (TransferData) {
    TransferData.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TransferData = exports.TransferData || (exports.TransferData = {}));
var CertificateDescription;
(function (CertificateDescription) {
    CertificateDescription.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CertificateDescription = exports.CertificateDescription || (exports.CertificateDescription = {}));
var DescribeCertificateResponse;
(function (DescribeCertificateResponse) {
    DescribeCertificateResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeCertificateResponse = exports.DescribeCertificateResponse || (exports.DescribeCertificateResponse = {}));
var DescribeCustomMetricRequest;
(function (DescribeCustomMetricRequest) {
    DescribeCustomMetricRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeCustomMetricRequest = exports.DescribeCustomMetricRequest || (exports.DescribeCustomMetricRequest = {}));
var DescribeCustomMetricResponse;
(function (DescribeCustomMetricResponse) {
    DescribeCustomMetricResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeCustomMetricResponse = exports.DescribeCustomMetricResponse || (exports.DescribeCustomMetricResponse = {}));
var DescribeDefaultAuthorizerRequest;
(function (DescribeDefaultAuthorizerRequest) {
    DescribeDefaultAuthorizerRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDefaultAuthorizerRequest = exports.DescribeDefaultAuthorizerRequest || (exports.DescribeDefaultAuthorizerRequest = {}));
var DescribeDefaultAuthorizerResponse;
(function (DescribeDefaultAuthorizerResponse) {
    DescribeDefaultAuthorizerResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDefaultAuthorizerResponse = exports.DescribeDefaultAuthorizerResponse || (exports.DescribeDefaultAuthorizerResponse = {}));
var DescribeDetectMitigationActionsTaskRequest;
(function (DescribeDetectMitigationActionsTaskRequest) {
    DescribeDetectMitigationActionsTaskRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDetectMitigationActionsTaskRequest = exports.DescribeDetectMitigationActionsTaskRequest || (exports.DescribeDetectMitigationActionsTaskRequest = {}));
var DetectMitigationActionsTaskTarget;
(function (DetectMitigationActionsTaskTarget) {
    DetectMitigationActionsTaskTarget.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetectMitigationActionsTaskTarget = exports.DetectMitigationActionsTaskTarget || (exports.DetectMitigationActionsTaskTarget = {}));
var DetectMitigationActionsTaskStatistics;
(function (DetectMitigationActionsTaskStatistics) {
    DetectMitigationActionsTaskStatistics.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetectMitigationActionsTaskStatistics = exports.DetectMitigationActionsTaskStatistics || (exports.DetectMitigationActionsTaskStatistics = {}));
var DetectMitigationActionsTaskStatus;
(function (DetectMitigationActionsTaskStatus) {
    DetectMitigationActionsTaskStatus["CANCELED"] = "CANCELED";
    DetectMitigationActionsTaskStatus["FAILED"] = "FAILED";
    DetectMitigationActionsTaskStatus["IN_PROGRESS"] = "IN_PROGRESS";
    DetectMitigationActionsTaskStatus["SUCCESSFUL"] = "SUCCESSFUL";
})(DetectMitigationActionsTaskStatus = exports.DetectMitigationActionsTaskStatus || (exports.DetectMitigationActionsTaskStatus = {}));
var ViolationEventOccurrenceRange;
(function (ViolationEventOccurrenceRange) {
    ViolationEventOccurrenceRange.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ViolationEventOccurrenceRange = exports.ViolationEventOccurrenceRange || (exports.ViolationEventOccurrenceRange = {}));
var DetectMitigationActionsTaskSummary;
(function (DetectMitigationActionsTaskSummary) {
    DetectMitigationActionsTaskSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetectMitigationActionsTaskSummary = exports.DetectMitigationActionsTaskSummary || (exports.DetectMitigationActionsTaskSummary = {}));
var DescribeDetectMitigationActionsTaskResponse;
(function (DescribeDetectMitigationActionsTaskResponse) {
    DescribeDetectMitigationActionsTaskResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDetectMitigationActionsTaskResponse = exports.DescribeDetectMitigationActionsTaskResponse || (exports.DescribeDetectMitigationActionsTaskResponse = {}));
var DescribeDimensionRequest;
(function (DescribeDimensionRequest) {
    DescribeDimensionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDimensionRequest = exports.DescribeDimensionRequest || (exports.DescribeDimensionRequest = {}));
var DescribeDimensionResponse;
(function (DescribeDimensionResponse) {
    DescribeDimensionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDimensionResponse = exports.DescribeDimensionResponse || (exports.DescribeDimensionResponse = {}));
var DescribeDomainConfigurationRequest;
(function (DescribeDomainConfigurationRequest) {
    DescribeDomainConfigurationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDomainConfigurationRequest = exports.DescribeDomainConfigurationRequest || (exports.DescribeDomainConfigurationRequest = {}));
var DomainConfigurationStatus;
(function (DomainConfigurationStatus) {
    DomainConfigurationStatus["DISABLED"] = "DISABLED";
    DomainConfigurationStatus["ENABLED"] = "ENABLED";
})(DomainConfigurationStatus = exports.DomainConfigurationStatus || (exports.DomainConfigurationStatus = {}));
var DomainType;
(function (DomainType) {
    DomainType["AWS_MANAGED"] = "AWS_MANAGED";
    DomainType["CUSTOMER_MANAGED"] = "CUSTOMER_MANAGED";
    DomainType["ENDPOINT"] = "ENDPOINT";
})(DomainType = exports.DomainType || (exports.DomainType = {}));
var ServerCertificateStatus;
(function (ServerCertificateStatus) {
    ServerCertificateStatus["INVALID"] = "INVALID";
    ServerCertificateStatus["VALID"] = "VALID";
})(ServerCertificateStatus = exports.ServerCertificateStatus || (exports.ServerCertificateStatus = {}));
var ServerCertificateSummary;
(function (ServerCertificateSummary) {
    ServerCertificateSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ServerCertificateSummary = exports.ServerCertificateSummary || (exports.ServerCertificateSummary = {}));
var DescribeDomainConfigurationResponse;
(function (DescribeDomainConfigurationResponse) {
    DescribeDomainConfigurationResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeDomainConfigurationResponse = exports.DescribeDomainConfigurationResponse || (exports.DescribeDomainConfigurationResponse = {}));
var DescribeEndpointRequest;
(function (DescribeEndpointRequest) {
    DescribeEndpointRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeEndpointRequest = exports.DescribeEndpointRequest || (exports.DescribeEndpointRequest = {}));
var DescribeEndpointResponse;
(function (DescribeEndpointResponse) {
    DescribeEndpointResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeEndpointResponse = exports.DescribeEndpointResponse || (exports.DescribeEndpointResponse = {}));
var DescribeEventConfigurationsRequest;
(function (DescribeEventConfigurationsRequest) {
    DescribeEventConfigurationsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeEventConfigurationsRequest = exports.DescribeEventConfigurationsRequest || (exports.DescribeEventConfigurationsRequest = {}));
var EventType;
(function (EventType) {
    EventType["CA_CERTIFICATE"] = "CA_CERTIFICATE";
    EventType["CERTIFICATE"] = "CERTIFICATE";
    EventType["JOB"] = "JOB";
    EventType["JOB_EXECUTION"] = "JOB_EXECUTION";
    EventType["POLICY"] = "POLICY";
    EventType["THING"] = "THING";
    EventType["THING_GROUP"] = "THING_GROUP";
    EventType["THING_GROUP_HIERARCHY"] = "THING_GROUP_HIERARCHY";
    EventType["THING_GROUP_MEMBERSHIP"] = "THING_GROUP_MEMBERSHIP";
    EventType["THING_TYPE"] = "THING_TYPE";
    EventType["THING_TYPE_ASSOCIATION"] = "THING_TYPE_ASSOCIATION";
})(EventType = exports.EventType || (exports.EventType = {}));
var Configuration;
(function (Configuration) {
    Configuration.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Configuration = exports.Configuration || (exports.Configuration = {}));
var DescribeEventConfigurationsResponse;
(function (DescribeEventConfigurationsResponse) {
    DescribeEventConfigurationsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeEventConfigurationsResponse = exports.DescribeEventConfigurationsResponse || (exports.DescribeEventConfigurationsResponse = {}));
var DescribeFleetMetricRequest;
(function (DescribeFleetMetricRequest) {
    DescribeFleetMetricRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeFleetMetricRequest = exports.DescribeFleetMetricRequest || (exports.DescribeFleetMetricRequest = {}));
var DescribeFleetMetricResponse;
(function (DescribeFleetMetricResponse) {
    DescribeFleetMetricResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeFleetMetricResponse = exports.DescribeFleetMetricResponse || (exports.DescribeFleetMetricResponse = {}));
var DescribeIndexRequest;
(function (DescribeIndexRequest) {
    DescribeIndexRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeIndexRequest = exports.DescribeIndexRequest || (exports.DescribeIndexRequest = {}));
var IndexStatus;
(function (IndexStatus) {
    IndexStatus["ACTIVE"] = "ACTIVE";
    IndexStatus["BUILDING"] = "BUILDING";
    IndexStatus["REBUILDING"] = "REBUILDING";
})(IndexStatus = exports.IndexStatus || (exports.IndexStatus = {}));
var DescribeIndexResponse;
(function (DescribeIndexResponse) {
    DescribeIndexResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeIndexResponse = exports.DescribeIndexResponse || (exports.DescribeIndexResponse = {}));
var DescribeJobRequest;
(function (DescribeJobRequest) {
    DescribeJobRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeJobRequest = exports.DescribeJobRequest || (exports.DescribeJobRequest = {}));
var JobProcessDetails;
(function (JobProcessDetails) {
    JobProcessDetails.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(JobProcessDetails = exports.JobProcessDetails || (exports.JobProcessDetails = {}));
var JobStatus;
(function (JobStatus) {
    JobStatus["CANCELED"] = "CANCELED";
    JobStatus["COMPLETED"] = "COMPLETED";
    JobStatus["DELETION_IN_PROGRESS"] = "DELETION_IN_PROGRESS";
    JobStatus["IN_PROGRESS"] = "IN_PROGRESS";
})(JobStatus = exports.JobStatus || (exports.JobStatus = {}));
var Job;
(function (Job) {
    Job.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Job = exports.Job || (exports.Job = {}));
var DescribeJobResponse;
(function (DescribeJobResponse) {
    DescribeJobResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeJobResponse = exports.DescribeJobResponse || (exports.DescribeJobResponse = {}));
var DescribeJobExecutionRequest;
(function (DescribeJobExecutionRequest) {
    DescribeJobExecutionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeJobExecutionRequest = exports.DescribeJobExecutionRequest || (exports.DescribeJobExecutionRequest = {}));
var JobExecutionStatus;
(function (JobExecutionStatus) {
    JobExecutionStatus["CANCELED"] = "CANCELED";
    JobExecutionStatus["FAILED"] = "FAILED";
    JobExecutionStatus["IN_PROGRESS"] = "IN_PROGRESS";
    JobExecutionStatus["QUEUED"] = "QUEUED";
    JobExecutionStatus["REJECTED"] = "REJECTED";
    JobExecutionStatus["REMOVED"] = "REMOVED";
    JobExecutionStatus["SUCCEEDED"] = "SUCCEEDED";
    JobExecutionStatus["TIMED_OUT"] = "TIMED_OUT";
})(JobExecutionStatus = exports.JobExecutionStatus || (exports.JobExecutionStatus = {}));
var JobExecutionStatusDetails;
(function (JobExecutionStatusDetails) {
    JobExecutionStatusDetails.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(JobExecutionStatusDetails = exports.JobExecutionStatusDetails || (exports.JobExecutionStatusDetails = {}));
var JobExecution;
(function (JobExecution) {
    JobExecution.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(JobExecution = exports.JobExecution || (exports.JobExecution = {}));
var DescribeJobExecutionResponse;
(function (DescribeJobExecutionResponse) {
    DescribeJobExecutionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeJobExecutionResponse = exports.DescribeJobExecutionResponse || (exports.DescribeJobExecutionResponse = {}));
var DescribeJobTemplateRequest;
(function (DescribeJobTemplateRequest) {
    DescribeJobTemplateRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeJobTemplateRequest = exports.DescribeJobTemplateRequest || (exports.DescribeJobTemplateRequest = {}));
var DescribeJobTemplateResponse;
(function (DescribeJobTemplateResponse) {
    DescribeJobTemplateResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeJobTemplateResponse = exports.DescribeJobTemplateResponse || (exports.DescribeJobTemplateResponse = {}));
var DescribeManagedJobTemplateRequest;
(function (DescribeManagedJobTemplateRequest) {
    DescribeManagedJobTemplateRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeManagedJobTemplateRequest = exports.DescribeManagedJobTemplateRequest || (exports.DescribeManagedJobTemplateRequest = {}));
var DocumentParameter;
(function (DocumentParameter) {
    DocumentParameter.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DocumentParameter = exports.DocumentParameter || (exports.DocumentParameter = {}));
var DescribeManagedJobTemplateResponse;
(function (DescribeManagedJobTemplateResponse) {
    DescribeManagedJobTemplateResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeManagedJobTemplateResponse = exports.DescribeManagedJobTemplateResponse || (exports.DescribeManagedJobTemplateResponse = {}));
var DescribeMitigationActionRequest;
(function (DescribeMitigationActionRequest) {
    DescribeMitigationActionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeMitigationActionRequest = exports.DescribeMitigationActionRequest || (exports.DescribeMitigationActionRequest = {}));
var MitigationActionType;
(function (MitigationActionType) {
    MitigationActionType["ADD_THINGS_TO_THING_GROUP"] = "ADD_THINGS_TO_THING_GROUP";
    MitigationActionType["ENABLE_IOT_LOGGING"] = "ENABLE_IOT_LOGGING";
    MitigationActionType["PUBLISH_FINDING_TO_SNS"] = "PUBLISH_FINDING_TO_SNS";
    MitigationActionType["REPLACE_DEFAULT_POLICY_VERSION"] = "REPLACE_DEFAULT_POLICY_VERSION";
    MitigationActionType["UPDATE_CA_CERTIFICATE"] = "UPDATE_CA_CERTIFICATE";
    MitigationActionType["UPDATE_DEVICE_CERTIFICATE"] = "UPDATE_DEVICE_CERTIFICATE";
})(MitigationActionType = exports.MitigationActionType || (exports.MitigationActionType = {}));
var DescribeMitigationActionResponse;
(function (DescribeMitigationActionResponse) {
    DescribeMitigationActionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeMitigationActionResponse = exports.DescribeMitigationActionResponse || (exports.DescribeMitigationActionResponse = {}));
var DescribeProvisioningTemplateRequest;
(function (DescribeProvisioningTemplateRequest) {
    DescribeProvisioningTemplateRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeProvisioningTemplateRequest = exports.DescribeProvisioningTemplateRequest || (exports.DescribeProvisioningTemplateRequest = {}));
var DescribeProvisioningTemplateResponse;
(function (DescribeProvisioningTemplateResponse) {
    DescribeProvisioningTemplateResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeProvisioningTemplateResponse = exports.DescribeProvisioningTemplateResponse || (exports.DescribeProvisioningTemplateResponse = {}));
var DescribeProvisioningTemplateVersionRequest;
(function (DescribeProvisioningTemplateVersionRequest) {
    DescribeProvisioningTemplateVersionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeProvisioningTemplateVersionRequest = exports.DescribeProvisioningTemplateVersionRequest || (exports.DescribeProvisioningTemplateVersionRequest = {}));
var DescribeProvisioningTemplateVersionResponse;
(function (DescribeProvisioningTemplateVersionResponse) {
    DescribeProvisioningTemplateVersionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeProvisioningTemplateVersionResponse = exports.DescribeProvisioningTemplateVersionResponse || (exports.DescribeProvisioningTemplateVersionResponse = {}));
var DescribeRoleAliasRequest;
(function (DescribeRoleAliasRequest) {
    DescribeRoleAliasRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeRoleAliasRequest = exports.DescribeRoleAliasRequest || (exports.DescribeRoleAliasRequest = {}));
var RoleAliasDescription;
(function (RoleAliasDescription) {
    RoleAliasDescription.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RoleAliasDescription = exports.RoleAliasDescription || (exports.RoleAliasDescription = {}));
var DescribeRoleAliasResponse;
(function (DescribeRoleAliasResponse) {
    DescribeRoleAliasResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeRoleAliasResponse = exports.DescribeRoleAliasResponse || (exports.DescribeRoleAliasResponse = {}));
var DescribeScheduledAuditRequest;
(function (DescribeScheduledAuditRequest) {
    DescribeScheduledAuditRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeScheduledAuditRequest = exports.DescribeScheduledAuditRequest || (exports.DescribeScheduledAuditRequest = {}));
var DescribeScheduledAuditResponse;
(function (DescribeScheduledAuditResponse) {
    DescribeScheduledAuditResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeScheduledAuditResponse = exports.DescribeScheduledAuditResponse || (exports.DescribeScheduledAuditResponse = {}));
var DescribeSecurityProfileRequest;
(function (DescribeSecurityProfileRequest) {
    DescribeSecurityProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeSecurityProfileRequest = exports.DescribeSecurityProfileRequest || (exports.DescribeSecurityProfileRequest = {}));
var DescribeSecurityProfileResponse;
(function (DescribeSecurityProfileResponse) {
    DescribeSecurityProfileResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeSecurityProfileResponse = exports.DescribeSecurityProfileResponse || (exports.DescribeSecurityProfileResponse = {}));
var DescribeStreamRequest;
(function (DescribeStreamRequest) {
    DescribeStreamRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStreamRequest = exports.DescribeStreamRequest || (exports.DescribeStreamRequest = {}));
var StreamInfo;
(function (StreamInfo) {
    StreamInfo.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StreamInfo = exports.StreamInfo || (exports.StreamInfo = {}));
var DescribeStreamResponse;
(function (DescribeStreamResponse) {
    DescribeStreamResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStreamResponse = exports.DescribeStreamResponse || (exports.DescribeStreamResponse = {}));
var DescribeThingRequest;
(function (DescribeThingRequest) {
    DescribeThingRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeThingRequest = exports.DescribeThingRequest || (exports.DescribeThingRequest = {}));
var DescribeThingResponse;
(function (DescribeThingResponse) {
    DescribeThingResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeThingResponse = exports.DescribeThingResponse || (exports.DescribeThingResponse = {}));
var DescribeThingGroupRequest;
(function (DescribeThingGroupRequest) {
    DescribeThingGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeThingGroupRequest = exports.DescribeThingGroupRequest || (exports.DescribeThingGroupRequest = {}));
var DynamicGroupStatus;
(function (DynamicGroupStatus) {
    DynamicGroupStatus["ACTIVE"] = "ACTIVE";
    DynamicGroupStatus["BUILDING"] = "BUILDING";
    DynamicGroupStatus["REBUILDING"] = "REBUILDING";
})(DynamicGroupStatus = exports.DynamicGroupStatus || (exports.DynamicGroupStatus = {}));
var GroupNameAndArn;
(function (GroupNameAndArn) {
    GroupNameAndArn.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GroupNameAndArn = exports.GroupNameAndArn || (exports.GroupNameAndArn = {}));
var ThingGroupMetadata;
(function (ThingGroupMetadata) {
    ThingGroupMetadata.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ThingGroupMetadata = exports.ThingGroupMetadata || (exports.ThingGroupMetadata = {}));
var DescribeThingGroupResponse;
(function (DescribeThingGroupResponse) {
    DescribeThingGroupResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeThingGroupResponse = exports.DescribeThingGroupResponse || (exports.DescribeThingGroupResponse = {}));
var DescribeThingRegistrationTaskRequest;
(function (DescribeThingRegistrationTaskRequest) {
    DescribeThingRegistrationTaskRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeThingRegistrationTaskRequest = exports.DescribeThingRegistrationTaskRequest || (exports.DescribeThingRegistrationTaskRequest = {}));
var Status;
(function (Status) {
    Status["Cancelled"] = "Cancelled";
    Status["Cancelling"] = "Cancelling";
    Status["Completed"] = "Completed";
    Status["Failed"] = "Failed";
    Status["InProgress"] = "InProgress";
})(Status = exports.Status || (exports.Status = {}));
var DescribeThingRegistrationTaskResponse;
(function (DescribeThingRegistrationTaskResponse) {
    DescribeThingRegistrationTaskResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeThingRegistrationTaskResponse = exports.DescribeThingRegistrationTaskResponse || (exports.DescribeThingRegistrationTaskResponse = {}));
var DescribeThingTypeRequest;
(function (DescribeThingTypeRequest) {
    DescribeThingTypeRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeThingTypeRequest = exports.DescribeThingTypeRequest || (exports.DescribeThingTypeRequest = {}));
var ThingTypeMetadata;
(function (ThingTypeMetadata) {
    ThingTypeMetadata.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ThingTypeMetadata = exports.ThingTypeMetadata || (exports.ThingTypeMetadata = {}));
var DescribeThingTypeResponse;
(function (DescribeThingTypeResponse) {
    DescribeThingTypeResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeThingTypeResponse = exports.DescribeThingTypeResponse || (exports.DescribeThingTypeResponse = {}));
var DetachPolicyRequest;
(function (DetachPolicyRequest) {
    DetachPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetachPolicyRequest = exports.DetachPolicyRequest || (exports.DetachPolicyRequest = {}));
var DetachPrincipalPolicyRequest;
(function (DetachPrincipalPolicyRequest) {
    DetachPrincipalPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetachPrincipalPolicyRequest = exports.DetachPrincipalPolicyRequest || (exports.DetachPrincipalPolicyRequest = {}));
var DetachSecurityProfileRequest;
(function (DetachSecurityProfileRequest) {
    DetachSecurityProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetachSecurityProfileRequest = exports.DetachSecurityProfileRequest || (exports.DetachSecurityProfileRequest = {}));
var DetachSecurityProfileResponse;
(function (DetachSecurityProfileResponse) {
    DetachSecurityProfileResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetachSecurityProfileResponse = exports.DetachSecurityProfileResponse || (exports.DetachSecurityProfileResponse = {}));
var DetachThingPrincipalRequest;
(function (DetachThingPrincipalRequest) {
    DetachThingPrincipalRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetachThingPrincipalRequest = exports.DetachThingPrincipalRequest || (exports.DetachThingPrincipalRequest = {}));
var DetachThingPrincipalResponse;
(function (DetachThingPrincipalResponse) {
    DetachThingPrincipalResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetachThingPrincipalResponse = exports.DetachThingPrincipalResponse || (exports.DetachThingPrincipalResponse = {}));
var DisableTopicRuleRequest;
(function (DisableTopicRuleRequest) {
    DisableTopicRuleRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DisableTopicRuleRequest = exports.DisableTopicRuleRequest || (exports.DisableTopicRuleRequest = {}));
var EnableTopicRuleRequest;
(function (EnableTopicRuleRequest) {
    EnableTopicRuleRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EnableTopicRuleRequest = exports.EnableTopicRuleRequest || (exports.EnableTopicRuleRequest = {}));
var GetBehaviorModelTrainingSummariesRequest;
(function (GetBehaviorModelTrainingSummariesRequest) {
    GetBehaviorModelTrainingSummariesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetBehaviorModelTrainingSummariesRequest = exports.GetBehaviorModelTrainingSummariesRequest || (exports.GetBehaviorModelTrainingSummariesRequest = {}));
var ModelStatus;
(function (ModelStatus) {
    ModelStatus["ACTIVE"] = "ACTIVE";
    ModelStatus["EXPIRED"] = "EXPIRED";
    ModelStatus["PENDING_BUILD"] = "PENDING_BUILD";
})(ModelStatus = exports.ModelStatus || (exports.ModelStatus = {}));
var BehaviorModelTrainingSummary;
(function (BehaviorModelTrainingSummary) {
    BehaviorModelTrainingSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(BehaviorModelTrainingSummary = exports.BehaviorModelTrainingSummary || (exports.BehaviorModelTrainingSummary = {}));
var GetBehaviorModelTrainingSummariesResponse;
(function (GetBehaviorModelTrainingSummariesResponse) {
    GetBehaviorModelTrainingSummariesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetBehaviorModelTrainingSummariesResponse = exports.GetBehaviorModelTrainingSummariesResponse || (exports.GetBehaviorModelTrainingSummariesResponse = {}));
var TermsAggregation;
(function (TermsAggregation) {
    TermsAggregation.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TermsAggregation = exports.TermsAggregation || (exports.TermsAggregation = {}));
var BucketsAggregationType;
(function (BucketsAggregationType) {
    BucketsAggregationType.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(BucketsAggregationType = exports.BucketsAggregationType || (exports.BucketsAggregationType = {}));
var GetBucketsAggregationRequest;
(function (GetBucketsAggregationRequest) {
    GetBucketsAggregationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetBucketsAggregationRequest = exports.GetBucketsAggregationRequest || (exports.GetBucketsAggregationRequest = {}));
var Bucket;
(function (Bucket) {
    Bucket.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Bucket = exports.Bucket || (exports.Bucket = {}));
var GetBucketsAggregationResponse;
(function (GetBucketsAggregationResponse) {
    GetBucketsAggregationResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetBucketsAggregationResponse = exports.GetBucketsAggregationResponse || (exports.GetBucketsAggregationResponse = {}));
var GetCardinalityRequest;
(function (GetCardinalityRequest) {
    GetCardinalityRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCardinalityRequest = exports.GetCardinalityRequest || (exports.GetCardinalityRequest = {}));
var GetCardinalityResponse;
(function (GetCardinalityResponse) {
    GetCardinalityResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetCardinalityResponse = exports.GetCardinalityResponse || (exports.GetCardinalityResponse = {}));
var GetEffectivePoliciesRequest;
(function (GetEffectivePoliciesRequest) {
    GetEffectivePoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetEffectivePoliciesRequest = exports.GetEffectivePoliciesRequest || (exports.GetEffectivePoliciesRequest = {}));
var EffectivePolicy;
(function (EffectivePolicy) {
    EffectivePolicy.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EffectivePolicy = exports.EffectivePolicy || (exports.EffectivePolicy = {}));
var GetEffectivePoliciesResponse;
(function (GetEffectivePoliciesResponse) {
    GetEffectivePoliciesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetEffectivePoliciesResponse = exports.GetEffectivePoliciesResponse || (exports.GetEffectivePoliciesResponse = {}));
var GetIndexingConfigurationRequest;
(function (GetIndexingConfigurationRequest) {
    GetIndexingConfigurationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetIndexingConfigurationRequest = exports.GetIndexingConfigurationRequest || (exports.GetIndexingConfigurationRequest = {}));
var FieldType;
(function (FieldType) {
    FieldType["BOOLEAN"] = "Boolean";
    FieldType["NUMBER"] = "Number";
    FieldType["STRING"] = "String";
})(FieldType = exports.FieldType || (exports.FieldType = {}));
var Field;
(function (Field) {
    Field.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Field = exports.Field || (exports.Field = {}));
var ThingGroupIndexingMode;
(function (ThingGroupIndexingMode) {
    ThingGroupIndexingMode["OFF"] = "OFF";
    ThingGroupIndexingMode["ON"] = "ON";
})(ThingGroupIndexingMode = exports.ThingGroupIndexingMode || (exports.ThingGroupIndexingMode = {}));
var ThingGroupIndexingConfiguration;
(function (ThingGroupIndexingConfiguration) {
    ThingGroupIndexingConfiguration.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ThingGroupIndexingConfiguration = exports.ThingGroupIndexingConfiguration || (exports.ThingGroupIndexingConfiguration = {}));
var DeviceDefenderIndexingMode;
(function (DeviceDefenderIndexingMode) {
    DeviceDefenderIndexingMode["OFF"] = "OFF";
    DeviceDefenderIndexingMode["VIOLATIONS"] = "VIOLATIONS";
})(DeviceDefenderIndexingMode = exports.DeviceDefenderIndexingMode || (exports.DeviceDefenderIndexingMode = {}));
var NamedShadowIndexingMode;
(function (NamedShadowIndexingMode) {
    NamedShadowIndexingMode["OFF"] = "OFF";
    NamedShadowIndexingMode["ON"] = "ON";
})(NamedShadowIndexingMode = exports.NamedShadowIndexingMode || (exports.NamedShadowIndexingMode = {}));
var ThingConnectivityIndexingMode;
(function (ThingConnectivityIndexingMode) {
    ThingConnectivityIndexingMode["OFF"] = "OFF";
    ThingConnectivityIndexingMode["STATUS"] = "STATUS";
})(ThingConnectivityIndexingMode = exports.ThingConnectivityIndexingMode || (exports.ThingConnectivityIndexingMode = {}));
var ThingIndexingMode;
(function (ThingIndexingMode) {
    ThingIndexingMode["OFF"] = "OFF";
    ThingIndexingMode["REGISTRY"] = "REGISTRY";
    ThingIndexingMode["REGISTRY_AND_SHADOW"] = "REGISTRY_AND_SHADOW";
})(ThingIndexingMode = exports.ThingIndexingMode || (exports.ThingIndexingMode = {}));
var ThingIndexingConfiguration;
(function (ThingIndexingConfiguration) {
    ThingIndexingConfiguration.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ThingIndexingConfiguration = exports.ThingIndexingConfiguration || (exports.ThingIndexingConfiguration = {}));
var GetIndexingConfigurationResponse;
(function (GetIndexingConfigurationResponse) {
    GetIndexingConfigurationResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetIndexingConfigurationResponse = exports.GetIndexingConfigurationResponse || (exports.GetIndexingConfigurationResponse = {}));
var GetJobDocumentRequest;
(function (GetJobDocumentRequest) {
    GetJobDocumentRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetJobDocumentRequest = exports.GetJobDocumentRequest || (exports.GetJobDocumentRequest = {}));
var GetJobDocumentResponse;
(function (GetJobDocumentResponse) {
    GetJobDocumentResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetJobDocumentResponse = exports.GetJobDocumentResponse || (exports.GetJobDocumentResponse = {}));
var GetLoggingOptionsRequest;
(function (GetLoggingOptionsRequest) {
    GetLoggingOptionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetLoggingOptionsRequest = exports.GetLoggingOptionsRequest || (exports.GetLoggingOptionsRequest = {}));
var GetLoggingOptionsResponse;
(function (GetLoggingOptionsResponse) {
    GetLoggingOptionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetLoggingOptionsResponse = exports.GetLoggingOptionsResponse || (exports.GetLoggingOptionsResponse = {}));
var GetOTAUpdateRequest;
(function (GetOTAUpdateRequest) {
    GetOTAUpdateRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetOTAUpdateRequest = exports.GetOTAUpdateRequest || (exports.GetOTAUpdateRequest = {}));
var ErrorInfo;
(function (ErrorInfo) {
    ErrorInfo.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ErrorInfo = exports.ErrorInfo || (exports.ErrorInfo = {}));
var OTAUpdateInfo;
(function (OTAUpdateInfo) {
    OTAUpdateInfo.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OTAUpdateInfo = exports.OTAUpdateInfo || (exports.OTAUpdateInfo = {}));
var GetOTAUpdateResponse;
(function (GetOTAUpdateResponse) {
    GetOTAUpdateResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetOTAUpdateResponse = exports.GetOTAUpdateResponse || (exports.GetOTAUpdateResponse = {}));
var GetPercentilesRequest;
(function (GetPercentilesRequest) {
    GetPercentilesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPercentilesRequest = exports.GetPercentilesRequest || (exports.GetPercentilesRequest = {}));
var PercentPair;
(function (PercentPair) {
    PercentPair.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PercentPair = exports.PercentPair || (exports.PercentPair = {}));
var GetPercentilesResponse;
(function (GetPercentilesResponse) {
    GetPercentilesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPercentilesResponse = exports.GetPercentilesResponse || (exports.GetPercentilesResponse = {}));
var GetPolicyRequest;
(function (GetPolicyRequest) {
    GetPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPolicyRequest = exports.GetPolicyRequest || (exports.GetPolicyRequest = {}));
var GetPolicyResponse;
(function (GetPolicyResponse) {
    GetPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPolicyResponse = exports.GetPolicyResponse || (exports.GetPolicyResponse = {}));
var GetPolicyVersionRequest;
(function (GetPolicyVersionRequest) {
    GetPolicyVersionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPolicyVersionRequest = exports.GetPolicyVersionRequest || (exports.GetPolicyVersionRequest = {}));
var GetPolicyVersionResponse;
(function (GetPolicyVersionResponse) {
    GetPolicyVersionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPolicyVersionResponse = exports.GetPolicyVersionResponse || (exports.GetPolicyVersionResponse = {}));
var GetRegistrationCodeRequest;
(function (GetRegistrationCodeRequest) {
    GetRegistrationCodeRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetRegistrationCodeRequest = exports.GetRegistrationCodeRequest || (exports.GetRegistrationCodeRequest = {}));
var GetRegistrationCodeResponse;
(function (GetRegistrationCodeResponse) {
    GetRegistrationCodeResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetRegistrationCodeResponse = exports.GetRegistrationCodeResponse || (exports.GetRegistrationCodeResponse = {}));
var GetStatisticsRequest;
(function (GetStatisticsRequest) {
    GetStatisticsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetStatisticsRequest = exports.GetStatisticsRequest || (exports.GetStatisticsRequest = {}));
var Statistics;
(function (Statistics) {
    Statistics.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Statistics = exports.Statistics || (exports.Statistics = {}));
var GetStatisticsResponse;
(function (GetStatisticsResponse) {
    GetStatisticsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetStatisticsResponse = exports.GetStatisticsResponse || (exports.GetStatisticsResponse = {}));
var GetTopicRuleRequest;
(function (GetTopicRuleRequest) {
    GetTopicRuleRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetTopicRuleRequest = exports.GetTopicRuleRequest || (exports.GetTopicRuleRequest = {}));
var TopicRule;
(function (TopicRule) {
    TopicRule.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TopicRule = exports.TopicRule || (exports.TopicRule = {}));
var GetTopicRuleResponse;
(function (GetTopicRuleResponse) {
    GetTopicRuleResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetTopicRuleResponse = exports.GetTopicRuleResponse || (exports.GetTopicRuleResponse = {}));
var GetTopicRuleDestinationRequest;
(function (GetTopicRuleDestinationRequest) {
    GetTopicRuleDestinationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetTopicRuleDestinationRequest = exports.GetTopicRuleDestinationRequest || (exports.GetTopicRuleDestinationRequest = {}));
var GetTopicRuleDestinationResponse;
(function (GetTopicRuleDestinationResponse) {
    GetTopicRuleDestinationResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetTopicRuleDestinationResponse = exports.GetTopicRuleDestinationResponse || (exports.GetTopicRuleDestinationResponse = {}));
var GetV2LoggingOptionsRequest;
(function (GetV2LoggingOptionsRequest) {
    GetV2LoggingOptionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetV2LoggingOptionsRequest = exports.GetV2LoggingOptionsRequest || (exports.GetV2LoggingOptionsRequest = {}));
var GetV2LoggingOptionsResponse;
(function (GetV2LoggingOptionsResponse) {
    GetV2LoggingOptionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetV2LoggingOptionsResponse = exports.GetV2LoggingOptionsResponse || (exports.GetV2LoggingOptionsResponse = {}));
var BehaviorCriteriaType;
(function (BehaviorCriteriaType) {
    BehaviorCriteriaType["MACHINE_LEARNING"] = "MACHINE_LEARNING";
    BehaviorCriteriaType["STATIC"] = "STATIC";
    BehaviorCriteriaType["STATISTICAL"] = "STATISTICAL";
})(BehaviorCriteriaType = exports.BehaviorCriteriaType || (exports.BehaviorCriteriaType = {}));
var ListActiveViolationsRequest;
(function (ListActiveViolationsRequest) {
    ListActiveViolationsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListActiveViolationsRequest = exports.ListActiveViolationsRequest || (exports.ListActiveViolationsRequest = {}));
var ListActiveViolationsResponse;
(function (ListActiveViolationsResponse) {
    ListActiveViolationsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListActiveViolationsResponse = exports.ListActiveViolationsResponse || (exports.ListActiveViolationsResponse = {}));
var ListAttachedPoliciesRequest;
(function (ListAttachedPoliciesRequest) {
    ListAttachedPoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAttachedPoliciesRequest = exports.ListAttachedPoliciesRequest || (exports.ListAttachedPoliciesRequest = {}));
var ListAttachedPoliciesResponse;
(function (ListAttachedPoliciesResponse) {
    ListAttachedPoliciesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAttachedPoliciesResponse = exports.ListAttachedPoliciesResponse || (exports.ListAttachedPoliciesResponse = {}));
var ListAuditFindingsRequest;
(function (ListAuditFindingsRequest) {
    ListAuditFindingsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAuditFindingsRequest = exports.ListAuditFindingsRequest || (exports.ListAuditFindingsRequest = {}));
var ListAuditFindingsResponse;
(function (ListAuditFindingsResponse) {
    ListAuditFindingsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAuditFindingsResponse = exports.ListAuditFindingsResponse || (exports.ListAuditFindingsResponse = {}));
var ListAuditMitigationActionsExecutionsRequest;
(function (ListAuditMitigationActionsExecutionsRequest) {
    ListAuditMitigationActionsExecutionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAuditMitigationActionsExecutionsRequest = exports.ListAuditMitigationActionsExecutionsRequest || (exports.ListAuditMitigationActionsExecutionsRequest = {}));
var ListAuditMitigationActionsExecutionsResponse;
(function (ListAuditMitigationActionsExecutionsResponse) {
    ListAuditMitigationActionsExecutionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAuditMitigationActionsExecutionsResponse = exports.ListAuditMitigationActionsExecutionsResponse || (exports.ListAuditMitigationActionsExecutionsResponse = {}));
var ListAuditMitigationActionsTasksRequest;
(function (ListAuditMitigationActionsTasksRequest) {
    ListAuditMitigationActionsTasksRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAuditMitigationActionsTasksRequest = exports.ListAuditMitigationActionsTasksRequest || (exports.ListAuditMitigationActionsTasksRequest = {}));
var ListAuditMitigationActionsTasksResponse;
(function (ListAuditMitigationActionsTasksResponse) {
    ListAuditMitigationActionsTasksResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAuditMitigationActionsTasksResponse = exports.ListAuditMitigationActionsTasksResponse || (exports.ListAuditMitigationActionsTasksResponse = {}));
var ListAuditSuppressionsRequest;
(function (ListAuditSuppressionsRequest) {
    ListAuditSuppressionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAuditSuppressionsRequest = exports.ListAuditSuppressionsRequest || (exports.ListAuditSuppressionsRequest = {}));
var ListAuditSuppressionsResponse;
(function (ListAuditSuppressionsResponse) {
    ListAuditSuppressionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAuditSuppressionsResponse = exports.ListAuditSuppressionsResponse || (exports.ListAuditSuppressionsResponse = {}));
var ListAuditTasksRequest;
(function (ListAuditTasksRequest) {
    ListAuditTasksRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAuditTasksRequest = exports.ListAuditTasksRequest || (exports.ListAuditTasksRequest = {}));
var ListAuditTasksResponse;
(function (ListAuditTasksResponse) {
    ListAuditTasksResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAuditTasksResponse = exports.ListAuditTasksResponse || (exports.ListAuditTasksResponse = {}));
var ListAuthorizersRequest;
(function (ListAuthorizersRequest) {
    ListAuthorizersRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAuthorizersRequest = exports.ListAuthorizersRequest || (exports.ListAuthorizersRequest = {}));
var ListAuthorizersResponse;
(function (ListAuthorizersResponse) {
    ListAuthorizersResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAuthorizersResponse = exports.ListAuthorizersResponse || (exports.ListAuthorizersResponse = {}));
var ListBillingGroupsRequest;
(function (ListBillingGroupsRequest) {
    ListBillingGroupsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListBillingGroupsRequest = exports.ListBillingGroupsRequest || (exports.ListBillingGroupsRequest = {}));
var ListBillingGroupsResponse;
(function (ListBillingGroupsResponse) {
    ListBillingGroupsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListBillingGroupsResponse = exports.ListBillingGroupsResponse || (exports.ListBillingGroupsResponse = {}));
var ListCACertificatesRequest;
(function (ListCACertificatesRequest) {
    ListCACertificatesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCACertificatesRequest = exports.ListCACertificatesRequest || (exports.ListCACertificatesRequest = {}));
var CACertificate;
(function (CACertificate) {
    CACertificate.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CACertificate = exports.CACertificate || (exports.CACertificate = {}));
var ListCACertificatesResponse;
(function (ListCACertificatesResponse) {
    ListCACertificatesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCACertificatesResponse = exports.ListCACertificatesResponse || (exports.ListCACertificatesResponse = {}));
var ListCertificatesRequest;
(function (ListCertificatesRequest) {
    ListCertificatesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCertificatesRequest = exports.ListCertificatesRequest || (exports.ListCertificatesRequest = {}));
var Certificate;
(function (Certificate) {
    Certificate.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Certificate = exports.Certificate || (exports.Certificate = {}));
var ListCertificatesResponse;
(function (ListCertificatesResponse) {
    ListCertificatesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCertificatesResponse = exports.ListCertificatesResponse || (exports.ListCertificatesResponse = {}));
var ListCertificatesByCARequest;
(function (ListCertificatesByCARequest) {
    ListCertificatesByCARequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCertificatesByCARequest = exports.ListCertificatesByCARequest || (exports.ListCertificatesByCARequest = {}));
var ListCertificatesByCAResponse;
(function (ListCertificatesByCAResponse) {
    ListCertificatesByCAResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCertificatesByCAResponse = exports.ListCertificatesByCAResponse || (exports.ListCertificatesByCAResponse = {}));
var ListCustomMetricsRequest;
(function (ListCustomMetricsRequest) {
    ListCustomMetricsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCustomMetricsRequest = exports.ListCustomMetricsRequest || (exports.ListCustomMetricsRequest = {}));
var ListCustomMetricsResponse;
(function (ListCustomMetricsResponse) {
    ListCustomMetricsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListCustomMetricsResponse = exports.ListCustomMetricsResponse || (exports.ListCustomMetricsResponse = {}));
var ListDetectMitigationActionsExecutionsRequest;
(function (ListDetectMitigationActionsExecutionsRequest) {
    ListDetectMitigationActionsExecutionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDetectMitigationActionsExecutionsRequest = exports.ListDetectMitigationActionsExecutionsRequest || (exports.ListDetectMitigationActionsExecutionsRequest = {}));
var DetectMitigationActionExecutionStatus;
(function (DetectMitigationActionExecutionStatus) {
    DetectMitigationActionExecutionStatus["FAILED"] = "FAILED";
    DetectMitigationActionExecutionStatus["IN_PROGRESS"] = "IN_PROGRESS";
    DetectMitigationActionExecutionStatus["SKIPPED"] = "SKIPPED";
    DetectMitigationActionExecutionStatus["SUCCESSFUL"] = "SUCCESSFUL";
})(DetectMitigationActionExecutionStatus = exports.DetectMitigationActionExecutionStatus || (exports.DetectMitigationActionExecutionStatus = {}));
var DetectMitigationActionExecution;
(function (DetectMitigationActionExecution) {
    DetectMitigationActionExecution.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetectMitigationActionExecution = exports.DetectMitigationActionExecution || (exports.DetectMitigationActionExecution = {}));
var ListDetectMitigationActionsExecutionsResponse;
(function (ListDetectMitigationActionsExecutionsResponse) {
    ListDetectMitigationActionsExecutionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDetectMitigationActionsExecutionsResponse = exports.ListDetectMitigationActionsExecutionsResponse || (exports.ListDetectMitigationActionsExecutionsResponse = {}));
var ListDetectMitigationActionsTasksRequest;
(function (ListDetectMitigationActionsTasksRequest) {
    ListDetectMitigationActionsTasksRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDetectMitigationActionsTasksRequest = exports.ListDetectMitigationActionsTasksRequest || (exports.ListDetectMitigationActionsTasksRequest = {}));
var ListDetectMitigationActionsTasksResponse;
(function (ListDetectMitigationActionsTasksResponse) {
    ListDetectMitigationActionsTasksResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDetectMitigationActionsTasksResponse = exports.ListDetectMitigationActionsTasksResponse || (exports.ListDetectMitigationActionsTasksResponse = {}));
var ListDimensionsRequest;
(function (ListDimensionsRequest) {
    ListDimensionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDimensionsRequest = exports.ListDimensionsRequest || (exports.ListDimensionsRequest = {}));
var ListDimensionsResponse;
(function (ListDimensionsResponse) {
    ListDimensionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDimensionsResponse = exports.ListDimensionsResponse || (exports.ListDimensionsResponse = {}));
var ListDomainConfigurationsRequest;
(function (ListDomainConfigurationsRequest) {
    ListDomainConfigurationsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDomainConfigurationsRequest = exports.ListDomainConfigurationsRequest || (exports.ListDomainConfigurationsRequest = {}));
var DomainConfigurationSummary;
(function (DomainConfigurationSummary) {
    DomainConfigurationSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DomainConfigurationSummary = exports.DomainConfigurationSummary || (exports.DomainConfigurationSummary = {}));
var ListDomainConfigurationsResponse;
(function (ListDomainConfigurationsResponse) {
    ListDomainConfigurationsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListDomainConfigurationsResponse = exports.ListDomainConfigurationsResponse || (exports.ListDomainConfigurationsResponse = {}));
var ListFleetMetricsRequest;
(function (ListFleetMetricsRequest) {
    ListFleetMetricsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFleetMetricsRequest = exports.ListFleetMetricsRequest || (exports.ListFleetMetricsRequest = {}));
var FleetMetricNameAndArn;
(function (FleetMetricNameAndArn) {
    FleetMetricNameAndArn.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(FleetMetricNameAndArn = exports.FleetMetricNameAndArn || (exports.FleetMetricNameAndArn = {}));
var ListFleetMetricsResponse;
(function (ListFleetMetricsResponse) {
    ListFleetMetricsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListFleetMetricsResponse = exports.ListFleetMetricsResponse || (exports.ListFleetMetricsResponse = {}));
var ListIndicesRequest;
(function (ListIndicesRequest) {
    ListIndicesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListIndicesRequest = exports.ListIndicesRequest || (exports.ListIndicesRequest = {}));
var ListIndicesResponse;
(function (ListIndicesResponse) {
    ListIndicesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListIndicesResponse = exports.ListIndicesResponse || (exports.ListIndicesResponse = {}));
var ListJobExecutionsForJobRequest;
(function (ListJobExecutionsForJobRequest) {
    ListJobExecutionsForJobRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListJobExecutionsForJobRequest = exports.ListJobExecutionsForJobRequest || (exports.ListJobExecutionsForJobRequest = {}));
var JobExecutionSummary;
(function (JobExecutionSummary) {
    JobExecutionSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(JobExecutionSummary = exports.JobExecutionSummary || (exports.JobExecutionSummary = {}));
var JobExecutionSummaryForJob;
(function (JobExecutionSummaryForJob) {
    JobExecutionSummaryForJob.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(JobExecutionSummaryForJob = exports.JobExecutionSummaryForJob || (exports.JobExecutionSummaryForJob = {}));
var ListJobExecutionsForJobResponse;
(function (ListJobExecutionsForJobResponse) {
    ListJobExecutionsForJobResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListJobExecutionsForJobResponse = exports.ListJobExecutionsForJobResponse || (exports.ListJobExecutionsForJobResponse = {}));
var ListJobExecutionsForThingRequest;
(function (ListJobExecutionsForThingRequest) {
    ListJobExecutionsForThingRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListJobExecutionsForThingRequest = exports.ListJobExecutionsForThingRequest || (exports.ListJobExecutionsForThingRequest = {}));
var JobExecutionSummaryForThing;
(function (JobExecutionSummaryForThing) {
    JobExecutionSummaryForThing.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(JobExecutionSummaryForThing = exports.JobExecutionSummaryForThing || (exports.JobExecutionSummaryForThing = {}));
var ListJobExecutionsForThingResponse;
(function (ListJobExecutionsForThingResponse) {
    ListJobExecutionsForThingResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListJobExecutionsForThingResponse = exports.ListJobExecutionsForThingResponse || (exports.ListJobExecutionsForThingResponse = {}));
var ListJobsRequest;
(function (ListJobsRequest) {
    ListJobsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListJobsRequest = exports.ListJobsRequest || (exports.ListJobsRequest = {}));
var JobSummary;
(function (JobSummary) {
    JobSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(JobSummary = exports.JobSummary || (exports.JobSummary = {}));
var ListJobsResponse;
(function (ListJobsResponse) {
    ListJobsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListJobsResponse = exports.ListJobsResponse || (exports.ListJobsResponse = {}));
var ListJobTemplatesRequest;
(function (ListJobTemplatesRequest) {
    ListJobTemplatesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListJobTemplatesRequest = exports.ListJobTemplatesRequest || (exports.ListJobTemplatesRequest = {}));
var JobTemplateSummary;
(function (JobTemplateSummary) {
    JobTemplateSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(JobTemplateSummary = exports.JobTemplateSummary || (exports.JobTemplateSummary = {}));
var ListJobTemplatesResponse;
(function (ListJobTemplatesResponse) {
    ListJobTemplatesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListJobTemplatesResponse = exports.ListJobTemplatesResponse || (exports.ListJobTemplatesResponse = {}));
var ListManagedJobTemplatesRequest;
(function (ListManagedJobTemplatesRequest) {
    ListManagedJobTemplatesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListManagedJobTemplatesRequest = exports.ListManagedJobTemplatesRequest || (exports.ListManagedJobTemplatesRequest = {}));
var ManagedJobTemplateSummary;
(function (ManagedJobTemplateSummary) {
    ManagedJobTemplateSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ManagedJobTemplateSummary = exports.ManagedJobTemplateSummary || (exports.ManagedJobTemplateSummary = {}));
var ListManagedJobTemplatesResponse;
(function (ListManagedJobTemplatesResponse) {
    ListManagedJobTemplatesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListManagedJobTemplatesResponse = exports.ListManagedJobTemplatesResponse || (exports.ListManagedJobTemplatesResponse = {}));
var ListMitigationActionsRequest;
(function (ListMitigationActionsRequest) {
    ListMitigationActionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListMitigationActionsRequest = exports.ListMitigationActionsRequest || (exports.ListMitigationActionsRequest = {}));
var MitigationActionIdentifier;
(function (MitigationActionIdentifier) {
    MitigationActionIdentifier.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(MitigationActionIdentifier = exports.MitigationActionIdentifier || (exports.MitigationActionIdentifier = {}));
var ListMitigationActionsResponse;
(function (ListMitigationActionsResponse) {
    ListMitigationActionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListMitigationActionsResponse = exports.ListMitigationActionsResponse || (exports.ListMitigationActionsResponse = {}));
var ListOTAUpdatesRequest;
(function (ListOTAUpdatesRequest) {
    ListOTAUpdatesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListOTAUpdatesRequest = exports.ListOTAUpdatesRequest || (exports.ListOTAUpdatesRequest = {}));
var OTAUpdateSummary;
(function (OTAUpdateSummary) {
    OTAUpdateSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OTAUpdateSummary = exports.OTAUpdateSummary || (exports.OTAUpdateSummary = {}));
var ListOTAUpdatesResponse;
(function (ListOTAUpdatesResponse) {
    ListOTAUpdatesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListOTAUpdatesResponse = exports.ListOTAUpdatesResponse || (exports.ListOTAUpdatesResponse = {}));
var ListOutgoingCertificatesRequest;
(function (ListOutgoingCertificatesRequest) {
    ListOutgoingCertificatesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListOutgoingCertificatesRequest = exports.ListOutgoingCertificatesRequest || (exports.ListOutgoingCertificatesRequest = {}));
var OutgoingCertificate;
(function (OutgoingCertificate) {
    OutgoingCertificate.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OutgoingCertificate = exports.OutgoingCertificate || (exports.OutgoingCertificate = {}));
var ListOutgoingCertificatesResponse;
(function (ListOutgoingCertificatesResponse) {
    ListOutgoingCertificatesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListOutgoingCertificatesResponse = exports.ListOutgoingCertificatesResponse || (exports.ListOutgoingCertificatesResponse = {}));
var ListPoliciesRequest;
(function (ListPoliciesRequest) {
    ListPoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPoliciesRequest = exports.ListPoliciesRequest || (exports.ListPoliciesRequest = {}));
var ListPoliciesResponse;
(function (ListPoliciesResponse) {
    ListPoliciesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPoliciesResponse = exports.ListPoliciesResponse || (exports.ListPoliciesResponse = {}));
var ListPolicyPrincipalsRequest;
(function (ListPolicyPrincipalsRequest) {
    ListPolicyPrincipalsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPolicyPrincipalsRequest = exports.ListPolicyPrincipalsRequest || (exports.ListPolicyPrincipalsRequest = {}));
var ListPolicyPrincipalsResponse;
(function (ListPolicyPrincipalsResponse) {
    ListPolicyPrincipalsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPolicyPrincipalsResponse = exports.ListPolicyPrincipalsResponse || (exports.ListPolicyPrincipalsResponse = {}));
var ListPolicyVersionsRequest;
(function (ListPolicyVersionsRequest) {
    ListPolicyVersionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPolicyVersionsRequest = exports.ListPolicyVersionsRequest || (exports.ListPolicyVersionsRequest = {}));
var PolicyVersion;
(function (PolicyVersion) {
    PolicyVersion.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PolicyVersion = exports.PolicyVersion || (exports.PolicyVersion = {}));
var ListPolicyVersionsResponse;
(function (ListPolicyVersionsResponse) {
    ListPolicyVersionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPolicyVersionsResponse = exports.ListPolicyVersionsResponse || (exports.ListPolicyVersionsResponse = {}));
var ListPrincipalPoliciesRequest;
(function (ListPrincipalPoliciesRequest) {
    ListPrincipalPoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPrincipalPoliciesRequest = exports.ListPrincipalPoliciesRequest || (exports.ListPrincipalPoliciesRequest = {}));
var ListPrincipalPoliciesResponse;
(function (ListPrincipalPoliciesResponse) {
    ListPrincipalPoliciesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPrincipalPoliciesResponse = exports.ListPrincipalPoliciesResponse || (exports.ListPrincipalPoliciesResponse = {}));
var ListPrincipalThingsRequest;
(function (ListPrincipalThingsRequest) {
    ListPrincipalThingsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPrincipalThingsRequest = exports.ListPrincipalThingsRequest || (exports.ListPrincipalThingsRequest = {}));
var ListPrincipalThingsResponse;
(function (ListPrincipalThingsResponse) {
    ListPrincipalThingsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListPrincipalThingsResponse = exports.ListPrincipalThingsResponse || (exports.ListPrincipalThingsResponse = {}));
var ListProvisioningTemplatesRequest;
(function (ListProvisioningTemplatesRequest) {
    ListProvisioningTemplatesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListProvisioningTemplatesRequest = exports.ListProvisioningTemplatesRequest || (exports.ListProvisioningTemplatesRequest = {}));
var ProvisioningTemplateSummary;
(function (ProvisioningTemplateSummary) {
    ProvisioningTemplateSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ProvisioningTemplateSummary = exports.ProvisioningTemplateSummary || (exports.ProvisioningTemplateSummary = {}));
var ListProvisioningTemplatesResponse;
(function (ListProvisioningTemplatesResponse) {
    ListProvisioningTemplatesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListProvisioningTemplatesResponse = exports.ListProvisioningTemplatesResponse || (exports.ListProvisioningTemplatesResponse = {}));
var ListProvisioningTemplateVersionsRequest;
(function (ListProvisioningTemplateVersionsRequest) {
    ListProvisioningTemplateVersionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListProvisioningTemplateVersionsRequest = exports.ListProvisioningTemplateVersionsRequest || (exports.ListProvisioningTemplateVersionsRequest = {}));
var ProvisioningTemplateVersionSummary;
(function (ProvisioningTemplateVersionSummary) {
    ProvisioningTemplateVersionSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ProvisioningTemplateVersionSummary = exports.ProvisioningTemplateVersionSummary || (exports.ProvisioningTemplateVersionSummary = {}));
var ListProvisioningTemplateVersionsResponse;
(function (ListProvisioningTemplateVersionsResponse) {
    ListProvisioningTemplateVersionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListProvisioningTemplateVersionsResponse = exports.ListProvisioningTemplateVersionsResponse || (exports.ListProvisioningTemplateVersionsResponse = {}));
var ListRoleAliasesRequest;
(function (ListRoleAliasesRequest) {
    ListRoleAliasesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListRoleAliasesRequest = exports.ListRoleAliasesRequest || (exports.ListRoleAliasesRequest = {}));
var ListRoleAliasesResponse;
(function (ListRoleAliasesResponse) {
    ListRoleAliasesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListRoleAliasesResponse = exports.ListRoleAliasesResponse || (exports.ListRoleAliasesResponse = {}));
var ListScheduledAuditsRequest;
(function (ListScheduledAuditsRequest) {
    ListScheduledAuditsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListScheduledAuditsRequest = exports.ListScheduledAuditsRequest || (exports.ListScheduledAuditsRequest = {}));
var ScheduledAuditMetadata;
(function (ScheduledAuditMetadata) {
    ScheduledAuditMetadata.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ScheduledAuditMetadata = exports.ScheduledAuditMetadata || (exports.ScheduledAuditMetadata = {}));
var ListScheduledAuditsResponse;
(function (ListScheduledAuditsResponse) {
    ListScheduledAuditsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListScheduledAuditsResponse = exports.ListScheduledAuditsResponse || (exports.ListScheduledAuditsResponse = {}));
var ListSecurityProfilesRequest;
(function (ListSecurityProfilesRequest) {
    ListSecurityProfilesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListSecurityProfilesRequest = exports.ListSecurityProfilesRequest || (exports.ListSecurityProfilesRequest = {}));
var SecurityProfileIdentifier;
(function (SecurityProfileIdentifier) {
    SecurityProfileIdentifier.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SecurityProfileIdentifier = exports.SecurityProfileIdentifier || (exports.SecurityProfileIdentifier = {}));
var ListSecurityProfilesResponse;
(function (ListSecurityProfilesResponse) {
    ListSecurityProfilesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListSecurityProfilesResponse = exports.ListSecurityProfilesResponse || (exports.ListSecurityProfilesResponse = {}));
var ListSecurityProfilesForTargetRequest;
(function (ListSecurityProfilesForTargetRequest) {
    ListSecurityProfilesForTargetRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListSecurityProfilesForTargetRequest = exports.ListSecurityProfilesForTargetRequest || (exports.ListSecurityProfilesForTargetRequest = {}));
var SecurityProfileTarget;
(function (SecurityProfileTarget) {
    SecurityProfileTarget.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SecurityProfileTarget = exports.SecurityProfileTarget || (exports.SecurityProfileTarget = {}));
var SecurityProfileTargetMapping;
(function (SecurityProfileTargetMapping) {
    SecurityProfileTargetMapping.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SecurityProfileTargetMapping = exports.SecurityProfileTargetMapping || (exports.SecurityProfileTargetMapping = {}));
var ListSecurityProfilesForTargetResponse;
(function (ListSecurityProfilesForTargetResponse) {
    ListSecurityProfilesForTargetResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListSecurityProfilesForTargetResponse = exports.ListSecurityProfilesForTargetResponse || (exports.ListSecurityProfilesForTargetResponse = {}));
var ListStreamsRequest;
(function (ListStreamsRequest) {
    ListStreamsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStreamsRequest = exports.ListStreamsRequest || (exports.ListStreamsRequest = {}));
var StreamSummary;
(function (StreamSummary) {
    StreamSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StreamSummary = exports.StreamSummary || (exports.StreamSummary = {}));
var ListStreamsResponse;
(function (ListStreamsResponse) {
    ListStreamsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStreamsResponse = exports.ListStreamsResponse || (exports.ListStreamsResponse = {}));
var ListTagsForResourceRequest;
(function (ListTagsForResourceRequest) {
    ListTagsForResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTagsForResourceRequest = exports.ListTagsForResourceRequest || (exports.ListTagsForResourceRequest = {}));
var ListTagsForResourceResponse;
(function (ListTagsForResourceResponse) {
    ListTagsForResourceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTagsForResourceResponse = exports.ListTagsForResourceResponse || (exports.ListTagsForResourceResponse = {}));
var ListTargetsForPolicyRequest;
(function (ListTargetsForPolicyRequest) {
    ListTargetsForPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTargetsForPolicyRequest = exports.ListTargetsForPolicyRequest || (exports.ListTargetsForPolicyRequest = {}));
var ListTargetsForPolicyResponse;
(function (ListTargetsForPolicyResponse) {
    ListTargetsForPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTargetsForPolicyResponse = exports.ListTargetsForPolicyResponse || (exports.ListTargetsForPolicyResponse = {}));
var ListTargetsForSecurityProfileRequest;
(function (ListTargetsForSecurityProfileRequest) {
    ListTargetsForSecurityProfileRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTargetsForSecurityProfileRequest = exports.ListTargetsForSecurityProfileRequest || (exports.ListTargetsForSecurityProfileRequest = {}));
var ListTargetsForSecurityProfileResponse;
(function (ListTargetsForSecurityProfileResponse) {
    ListTargetsForSecurityProfileResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTargetsForSecurityProfileResponse = exports.ListTargetsForSecurityProfileResponse || (exports.ListTargetsForSecurityProfileResponse = {}));
var ListThingGroupsRequest;
(function (ListThingGroupsRequest) {
    ListThingGroupsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListThingGroupsRequest = exports.ListThingGroupsRequest || (exports.ListThingGroupsRequest = {}));
var ListThingGroupsResponse;
(function (ListThingGroupsResponse) {
    ListThingGroupsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListThingGroupsResponse = exports.ListThingGroupsResponse || (exports.ListThingGroupsResponse = {}));
var ListThingGroupsForThingRequest;
(function (ListThingGroupsForThingRequest) {
    ListThingGroupsForThingRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListThingGroupsForThingRequest = exports.ListThingGroupsForThingRequest || (exports.ListThingGroupsForThingRequest = {}));
var ListThingGroupsForThingResponse;
(function (ListThingGroupsForThingResponse) {
    ListThingGroupsForThingResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListThingGroupsForThingResponse = exports.ListThingGroupsForThingResponse || (exports.ListThingGroupsForThingResponse = {}));
var ListThingPrincipalsRequest;
(function (ListThingPrincipalsRequest) {
    ListThingPrincipalsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListThingPrincipalsRequest = exports.ListThingPrincipalsRequest || (exports.ListThingPrincipalsRequest = {}));
var ListThingPrincipalsResponse;
(function (ListThingPrincipalsResponse) {
    ListThingPrincipalsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListThingPrincipalsResponse = exports.ListThingPrincipalsResponse || (exports.ListThingPrincipalsResponse = {}));
var ReportType;
(function (ReportType) {
    ReportType["ERRORS"] = "ERRORS";
    ReportType["RESULTS"] = "RESULTS";
})(ReportType = exports.ReportType || (exports.ReportType = {}));
