// @aws-sdk/client-iot/dist-es/commands/AssociateTargetsWithJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AssociateTargetsWithJobRequest, AssociateTargetsWithJobResponse } from "../models/models_0";
import { deserializeAws_restJson1AssociateTargetsWithJobCommand, serializeAws_restJson1AssociateTargetsWithJobCommand, } from "../protocols/Aws_restJson1";
var AssociateTargetsWithJobCommand = (function (_super) {
    __extends(AssociateTargetsWithJobCommand, _super);
    function AssociateTargetsWithJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AssociateTargetsWithJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "AssociateTargetsWithJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AssociateTargetsWithJobRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AssociateTargetsWithJobResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AssociateTargetsWithJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1AssociateTargetsWithJobCommand(input, context);
    };
    AssociateTargetsWithJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1AssociateTargetsWithJobCommand(output, context);
    };
    return AssociateTargetsWithJobCommand;
}($Command));
export { AssociateTargetsWithJobCommand };
