// @aws-sdk/client-iot/dist-es/commands/AttachThingPrincipalCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AttachThingPrincipalRequest, AttachThingPrincipalResponse } from "../models/models_0";
import { deserializeAws_restJson1AttachThingPrincipalCommand, serializeAws_restJson1AttachThingPrincipalCommand, } from "../protocols/Aws_restJson1";
var AttachThingPrincipalCommand = (function (_super) {
    __extends(AttachThingPrincipalCommand, _super);
    function AttachThingPrincipalCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AttachThingPrincipalCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "AttachThingPrincipalCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AttachThingPrincipalRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AttachThingPrincipalResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AttachThingPrincipalCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1AttachThingPrincipalCommand(input, context);
    };
    AttachThingPrincipalCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1AttachThingPrincipalCommand(output, context);
    };
    return AttachThingPrincipalCommand;
}($Command));
export { AttachThingPrincipalCommand };
