// @aws-sdk/client-iot/dist-es/commands/CancelDetectMitigationActionsTaskCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CancelDetectMitigationActionsTaskRequest, CancelDetectMitigationActionsTaskResponse, } from "../models/models_0";
import { deserializeAws_restJson1CancelDetectMitigationActionsTaskCommand, serializeAws_restJson1CancelDetectMitigationActionsTaskCommand, } from "../protocols/Aws_restJson1";
var CancelDetectMitigationActionsTaskCommand = (function (_super) {
    __extends(CancelDetectMitigationActionsTaskCommand, _super);
    function CancelDetectMitigationActionsTaskCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CancelDetectMitigationActionsTaskCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "CancelDetectMitigationActionsTaskCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CancelDetectMitigationActionsTaskRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CancelDetectMitigationActionsTaskResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CancelDetectMitigationActionsTaskCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CancelDetectMitigationActionsTaskCommand(input, context);
    };
    CancelDetectMitigationActionsTaskCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CancelDetectMitigationActionsTaskCommand(output, context);
    };
    return CancelDetectMitigationActionsTaskCommand;
}($Command));
export { CancelDetectMitigationActionsTaskCommand };
