// @aws-sdk/client-iot/dist-es/commands/CancelJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CancelJobRequest, CancelJobResponse } from "../models/models_0";
import { deserializeAws_restJson1CancelJobCommand, serializeAws_restJson1CancelJobCommand, } from "../protocols/Aws_restJson1";
var CancelJobCommand = (function (_super) {
    __extends(CancelJobCommand, _super);
    function CancelJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CancelJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "CancelJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CancelJobRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CancelJobResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CancelJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CancelJobCommand(input, context);
    };
    CancelJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CancelJobCommand(output, context);
    };
    return CancelJobCommand;
}($Command));
export { CancelJobCommand };
