// @aws-sdk/client-iot/dist-es/commands/ConfirmTopicRuleDestinationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ConfirmTopicRuleDestinationRequest, ConfirmTopicRuleDestinationResponse } from "../models/models_0";
import { deserializeAws_restJson1ConfirmTopicRuleDestinationCommand, serializeAws_restJson1ConfirmTopicRuleDestinationCommand, } from "../protocols/Aws_restJson1";
var ConfirmTopicRuleDestinationCommand = (function (_super) {
    __extends(ConfirmTopicRuleDestinationCommand, _super);
    function ConfirmTopicRuleDestinationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ConfirmTopicRuleDestinationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ConfirmTopicRuleDestinationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ConfirmTopicRuleDestinationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ConfirmTopicRuleDestinationResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ConfirmTopicRuleDestinationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ConfirmTopicRuleDestinationCommand(input, context);
    };
    ConfirmTopicRuleDestinationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ConfirmTopicRuleDestinationCommand(output, context);
    };
    return ConfirmTopicRuleDestinationCommand;
}($Command));
export { ConfirmTopicRuleDestinationCommand };
