// @aws-sdk/client-iot/dist-es/commands/CreateBillingGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateBillingGroupRequest, CreateBillingGroupResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateBillingGroupCommand, serializeAws_restJson1CreateBillingGroupCommand, } from "../protocols/Aws_restJson1";
var CreateBillingGroupCommand = (function (_super) {
    __extends(CreateBillingGroupCommand, _super);
    function CreateBillingGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateBillingGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "CreateBillingGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateBillingGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateBillingGroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateBillingGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateBillingGroupCommand(input, context);
    };
    CreateBillingGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateBillingGroupCommand(output, context);
    };
    return CreateBillingGroupCommand;
}($Command));
export { CreateBillingGroupCommand };
