// @aws-sdk/client-iot/dist-es/commands/CreateMitigationActionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateMitigationActionRequest, CreateMitigationActionResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateMitigationActionCommand, serializeAws_restJson1CreateMitigationActionCommand, } from "../protocols/Aws_restJson1";
var CreateMitigationActionCommand = (function (_super) {
    __extends(CreateMitigationActionCommand, _super);
    function CreateMitigationActionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateMitigationActionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "CreateMitigationActionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateMitigationActionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateMitigationActionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateMitigationActionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateMitigationActionCommand(input, context);
    };
    CreateMitigationActionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateMitigationActionCommand(output, context);
    };
    return CreateMitigationActionCommand;
}($Command));
export { CreateMitigationActionCommand };
