// @aws-sdk/client-iot/dist-es/commands/CreateThingTypeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateThingTypeRequest, CreateThingTypeResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateThingTypeCommand, serializeAws_restJson1CreateThingTypeCommand, } from "../protocols/Aws_restJson1";
var CreateThingTypeCommand = (function (_super) {
    __extends(CreateThingTypeCommand, _super);
    function CreateThingTypeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateThingTypeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "CreateThingTypeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateThingTypeRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateThingTypeResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateThingTypeCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateThingTypeCommand(input, context);
    };
    CreateThingTypeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateThingTypeCommand(output, context);
    };
    return CreateThingTypeCommand;
}($Command));
export { CreateThingTypeCommand };
