// @aws-sdk/client-iot/dist-es/commands/CreateJobTemplateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateJobTemplateRequest, CreateJobTemplateResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateJobTemplateCommand, serializeAws_restJson1CreateJobTemplateCommand, } from "../protocols/Aws_restJson1";
var CreateJobTemplateCommand = (function (_super) {
    __extends(CreateJobTemplateCommand, _super);
    function CreateJobTemplateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateJobTemplateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "CreateJobTemplateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateJobTemplateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateJobTemplateResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateJobTemplateCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateJobTemplateCommand(input, context);
    };
    CreateJobTemplateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateJobTemplateCommand(output, context);
    };
    return CreateJobTemplateCommand;
}($Command));
export { CreateJobTemplateCommand };
