// @aws-sdk/client-iot/dist-es/commands/CreateTopicRuleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateTopicRuleRequest } from "../models/models_0";
import { deserializeAws_restJson1CreateTopicRuleCommand, serializeAws_restJson1CreateTopicRuleCommand, } from "../protocols/Aws_restJson1";
var CreateTopicRuleCommand = (function (_super) {
    __extends(CreateTopicRuleCommand, _super);
    function CreateTopicRuleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateTopicRuleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "CreateTopicRuleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateTopicRuleRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateTopicRuleCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateTopicRuleCommand(input, context);
    };
    CreateTopicRuleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateTopicRuleCommand(output, context);
    };
    return CreateTopicRuleCommand;
}($Command));
export { CreateTopicRuleCommand };
