// @aws-sdk/client-iot/dist-es/commands/CreateProvisioningClaimCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateProvisioningClaimRequest, CreateProvisioningClaimResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateProvisioningClaimCommand, serializeAws_restJson1CreateProvisioningClaimCommand, } from "../protocols/Aws_restJson1";
var CreateProvisioningClaimCommand = (function (_super) {
    __extends(CreateProvisioningClaimCommand, _super);
    function CreateProvisioningClaimCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateProvisioningClaimCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "CreateProvisioningClaimCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateProvisioningClaimRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateProvisioningClaimResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateProvisioningClaimCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateProvisioningClaimCommand(input, context);
    };
    CreateProvisioningClaimCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateProvisioningClaimCommand(output, context);
    };
    return CreateProvisioningClaimCommand;
}($Command));
export { CreateProvisioningClaimCommand };
