// @aws-sdk/client-iot/dist-es/commands/CreateCustomMetricCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateCustomMetricRequest, CreateCustomMetricResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateCustomMetricCommand, serializeAws_restJson1CreateCustomMetricCommand, } from "../protocols/Aws_restJson1";
var CreateCustomMetricCommand = (function (_super) {
    __extends(CreateCustomMetricCommand, _super);
    function CreateCustomMetricCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateCustomMetricCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "CreateCustomMetricCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateCustomMetricRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateCustomMetricResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateCustomMetricCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateCustomMetricCommand(input, context);
    };
    CreateCustomMetricCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateCustomMetricCommand(output, context);
    };
    return CreateCustomMetricCommand;
}($Command));
export { CreateCustomMetricCommand };
