// @aws-sdk/client-iot/dist-es/commands/CreateAuditSuppressionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateAuditSuppressionRequest, CreateAuditSuppressionResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateAuditSuppressionCommand, serializeAws_restJson1CreateAuditSuppressionCommand, } from "../protocols/Aws_restJson1";
var CreateAuditSuppressionCommand = (function (_super) {
    __extends(CreateAuditSuppressionCommand, _super);
    function CreateAuditSuppressionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateAuditSuppressionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "CreateAuditSuppressionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateAuditSuppressionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateAuditSuppressionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateAuditSuppressionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateAuditSuppressionCommand(input, context);
    };
    CreateAuditSuppressionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateAuditSuppressionCommand(output, context);
    };
    return CreateAuditSuppressionCommand;
}($Command));
export { CreateAuditSuppressionCommand };
