// @aws-sdk/client-iot/dist-es/commands/CreateThingGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateThingGroupRequest, CreateThingGroupResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateThingGroupCommand, serializeAws_restJson1CreateThingGroupCommand, } from "../protocols/Aws_restJson1";
var CreateThingGroupCommand = (function (_super) {
    __extends(CreateThingGroupCommand, _super);
    function CreateThingGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateThingGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "CreateThingGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateThingGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateThingGroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateThingGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateThingGroupCommand(input, context);
    };
    CreateThingGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateThingGroupCommand(output, context);
    };
    return CreateThingGroupCommand;
}($Command));
export { CreateThingGroupCommand };
