// @aws-sdk/client-iot/dist-es/commands/CreateThingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateThingRequest, CreateThingResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateThingCommand, serializeAws_restJson1CreateThingCommand, } from "../protocols/Aws_restJson1";
var CreateThingCommand = (function (_super) {
    __extends(CreateThingCommand, _super);
    function CreateThingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateThingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "CreateThingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateThingRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateThingResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateThingCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateThingCommand(input, context);
    };
    CreateThingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateThingCommand(output, context);
    };
    return CreateThingCommand;
}($Command));
export { CreateThingCommand };
