// @aws-sdk/client-iot/dist-es/commands/CreateFleetMetricCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateFleetMetricRequest, CreateFleetMetricResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateFleetMetricCommand, serializeAws_restJson1CreateFleetMetricCommand, } from "../protocols/Aws_restJson1";
var CreateFleetMetricCommand = (function (_super) {
    __extends(CreateFleetMetricCommand, _super);
    function CreateFleetMetricCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateFleetMetricCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "CreateFleetMetricCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateFleetMetricRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateFleetMetricResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateFleetMetricCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateFleetMetricCommand(input, context);
    };
    CreateFleetMetricCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateFleetMetricCommand(output, context);
    };
    return CreateFleetMetricCommand;
}($Command));
export { CreateFleetMetricCommand };
