// @aws-sdk/client-iot/dist-es/commands/DeleteRegistrationCodeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteRegistrationCodeRequest, DeleteRegistrationCodeResponse } from "../models/models_0";
import { deserializeAws_restJson1DeleteRegistrationCodeCommand, serializeAws_restJson1DeleteRegistrationCodeCommand, } from "../protocols/Aws_restJson1";
var DeleteRegistrationCodeCommand = (function (_super) {
    __extends(DeleteRegistrationCodeCommand, _super);
    function DeleteRegistrationCodeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteRegistrationCodeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DeleteRegistrationCodeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteRegistrationCodeRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteRegistrationCodeResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteRegistrationCodeCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteRegistrationCodeCommand(input, context);
    };
    DeleteRegistrationCodeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteRegistrationCodeCommand(output, context);
    };
    return DeleteRegistrationCodeCommand;
}($Command));
export { DeleteRegistrationCodeCommand };
