// @aws-sdk/client-iot/dist-es/commands/DeleteCACertificateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteCACertificateRequest, DeleteCACertificateResponse } from "../models/models_0";
import { deserializeAws_restJson1DeleteCACertificateCommand, serializeAws_restJson1DeleteCACertificateCommand, } from "../protocols/Aws_restJson1";
var DeleteCACertificateCommand = (function (_super) {
    __extends(DeleteCACertificateCommand, _super);
    function DeleteCACertificateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteCACertificateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DeleteCACertificateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteCACertificateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteCACertificateResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteCACertificateCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteCACertificateCommand(input, context);
    };
    DeleteCACertificateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteCACertificateCommand(output, context);
    };
    return DeleteCACertificateCommand;
}($Command));
export { DeleteCACertificateCommand };
