// @aws-sdk/client-iot/dist-es/commands/DeleteThingTypeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteThingTypeRequest, DeleteThingTypeResponse } from "../models/models_1";
import { deserializeAws_restJson1DeleteThingTypeCommand, serializeAws_restJson1DeleteThingTypeCommand, } from "../protocols/Aws_restJson1";
var DeleteThingTypeCommand = (function (_super) {
    __extends(DeleteThingTypeCommand, _super);
    function DeleteThingTypeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteThingTypeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DeleteThingTypeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteThingTypeRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteThingTypeResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteThingTypeCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteThingTypeCommand(input, context);
    };
    DeleteThingTypeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteThingTypeCommand(output, context);
    };
    return DeleteThingTypeCommand;
}($Command));
export { DeleteThingTypeCommand };
