// @aws-sdk/client-iot/dist-es/commands/DeleteThingGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteThingGroupRequest, DeleteThingGroupResponse } from "../models/models_1";
import { deserializeAws_restJson1DeleteThingGroupCommand, serializeAws_restJson1DeleteThingGroupCommand, } from "../protocols/Aws_restJson1";
var DeleteThingGroupCommand = (function (_super) {
    __extends(DeleteThingGroupCommand, _super);
    function DeleteThingGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteThingGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DeleteThingGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteThingGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteThingGroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteThingGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteThingGroupCommand(input, context);
    };
    DeleteThingGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteThingGroupCommand(output, context);
    };
    return DeleteThingGroupCommand;
}($Command));
export { DeleteThingGroupCommand };
