// @aws-sdk/client-iot/dist-es/commands/DeleteAccountAuditConfigurationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteAccountAuditConfigurationRequest, DeleteAccountAuditConfigurationResponse } from "../models/models_0";
import { deserializeAws_restJson1DeleteAccountAuditConfigurationCommand, serializeAws_restJson1DeleteAccountAuditConfigurationCommand, } from "../protocols/Aws_restJson1";
var DeleteAccountAuditConfigurationCommand = (function (_super) {
    __extends(DeleteAccountAuditConfigurationCommand, _super);
    function DeleteAccountAuditConfigurationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteAccountAuditConfigurationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DeleteAccountAuditConfigurationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteAccountAuditConfigurationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteAccountAuditConfigurationResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteAccountAuditConfigurationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteAccountAuditConfigurationCommand(input, context);
    };
    DeleteAccountAuditConfigurationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteAccountAuditConfigurationCommand(output, context);
    };
    return DeleteAccountAuditConfigurationCommand;
}($Command));
export { DeleteAccountAuditConfigurationCommand };
