// @aws-sdk/client-iot/dist-es/commands/DeleteFleetMetricCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteFleetMetricRequest } from "../models/models_0";
import { deserializeAws_restJson1DeleteFleetMetricCommand, serializeAws_restJson1DeleteFleetMetricCommand, } from "../protocols/Aws_restJson1";
var DeleteFleetMetricCommand = (function (_super) {
    __extends(DeleteFleetMetricCommand, _super);
    function DeleteFleetMetricCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteFleetMetricCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DeleteFleetMetricCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteFleetMetricRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteFleetMetricCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteFleetMetricCommand(input, context);
    };
    DeleteFleetMetricCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteFleetMetricCommand(output, context);
    };
    return DeleteFleetMetricCommand;
}($Command));
export { DeleteFleetMetricCommand };
