// @aws-sdk/client-iot/dist-es/commands/DeleteCustomMetricCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteCustomMetricRequest, DeleteCustomMetricResponse } from "../models/models_0";
import { deserializeAws_restJson1DeleteCustomMetricCommand, serializeAws_restJson1DeleteCustomMetricCommand, } from "../protocols/Aws_restJson1";
var DeleteCustomMetricCommand = (function (_super) {
    __extends(DeleteCustomMetricCommand, _super);
    function DeleteCustomMetricCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteCustomMetricCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DeleteCustomMetricCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteCustomMetricRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteCustomMetricResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteCustomMetricCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteCustomMetricCommand(input, context);
    };
    DeleteCustomMetricCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteCustomMetricCommand(output, context);
    };
    return DeleteCustomMetricCommand;
}($Command));
export { DeleteCustomMetricCommand };
