// @aws-sdk/client-iot/dist-es/commands/DeleteJobExecutionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteJobExecutionRequest } from "../models/models_0";
import { deserializeAws_restJson1DeleteJobExecutionCommand, serializeAws_restJson1DeleteJobExecutionCommand, } from "../protocols/Aws_restJson1";
var DeleteJobExecutionCommand = (function (_super) {
    __extends(DeleteJobExecutionCommand, _super);
    function DeleteJobExecutionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteJobExecutionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DeleteJobExecutionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteJobExecutionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteJobExecutionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteJobExecutionCommand(input, context);
    };
    DeleteJobExecutionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteJobExecutionCommand(output, context);
    };
    return DeleteJobExecutionCommand;
}($Command));
export { DeleteJobExecutionCommand };
