// @aws-sdk/client-iot/dist-es/commands/DescribeManagedJobTemplateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeManagedJobTemplateRequest, DescribeManagedJobTemplateResponse } from "../models/models_1";
import { deserializeAws_restJson1DescribeManagedJobTemplateCommand, serializeAws_restJson1DescribeManagedJobTemplateCommand, } from "../protocols/Aws_restJson1";
var DescribeManagedJobTemplateCommand = (function (_super) {
    __extends(DescribeManagedJobTemplateCommand, _super);
    function DescribeManagedJobTemplateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeManagedJobTemplateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DescribeManagedJobTemplateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeManagedJobTemplateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeManagedJobTemplateResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeManagedJobTemplateCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeManagedJobTemplateCommand(input, context);
    };
    DescribeManagedJobTemplateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeManagedJobTemplateCommand(output, context);
    };
    return DescribeManagedJobTemplateCommand;
}($Command));
export { DescribeManagedJobTemplateCommand };
