// @aws-sdk/client-iot/dist-es/commands/DescribeMitigationActionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeMitigationActionRequest, DescribeMitigationActionResponse } from "../models/models_1";
import { deserializeAws_restJson1DescribeMitigationActionCommand, serializeAws_restJson1DescribeMitigationActionCommand, } from "../protocols/Aws_restJson1";
var DescribeMitigationActionCommand = (function (_super) {
    __extends(DescribeMitigationActionCommand, _super);
    function DescribeMitigationActionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeMitigationActionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DescribeMitigationActionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeMitigationActionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeMitigationActionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeMitigationActionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeMitigationActionCommand(input, context);
    };
    DescribeMitigationActionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeMitigationActionCommand(output, context);
    };
    return DescribeMitigationActionCommand;
}($Command));
export { DescribeMitigationActionCommand };
