// @aws-sdk/client-iot/dist-es/commands/DescribeThingTypeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeThingTypeRequest, DescribeThingTypeResponse } from "../models/models_1";
import { deserializeAws_restJson1DescribeThingTypeCommand, serializeAws_restJson1DescribeThingTypeCommand, } from "../protocols/Aws_restJson1";
var DescribeThingTypeCommand = (function (_super) {
    __extends(DescribeThingTypeCommand, _super);
    function DescribeThingTypeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeThingTypeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DescribeThingTypeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeThingTypeRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeThingTypeResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeThingTypeCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeThingTypeCommand(input, context);
    };
    DescribeThingTypeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeThingTypeCommand(output, context);
    };
    return DescribeThingTypeCommand;
}($Command));
export { DescribeThingTypeCommand };
