// @aws-sdk/client-iot/dist-es/commands/DescribeEventConfigurationsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeEventConfigurationsRequest, DescribeEventConfigurationsResponse } from "../models/models_1";
import { deserializeAws_restJson1DescribeEventConfigurationsCommand, serializeAws_restJson1DescribeEventConfigurationsCommand, } from "../protocols/Aws_restJson1";
var DescribeEventConfigurationsCommand = (function (_super) {
    __extends(DescribeEventConfigurationsCommand, _super);
    function DescribeEventConfigurationsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeEventConfigurationsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DescribeEventConfigurationsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeEventConfigurationsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeEventConfigurationsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeEventConfigurationsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeEventConfigurationsCommand(input, context);
    };
    DescribeEventConfigurationsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeEventConfigurationsCommand(output, context);
    };
    return DescribeEventConfigurationsCommand;
}($Command));
export { DescribeEventConfigurationsCommand };
