// @aws-sdk/client-iot/dist-es/commands/DescribeIndexCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeIndexRequest, DescribeIndexResponse } from "../models/models_1";
import { deserializeAws_restJson1DescribeIndexCommand, serializeAws_restJson1DescribeIndexCommand, } from "../protocols/Aws_restJson1";
var DescribeIndexCommand = (function (_super) {
    __extends(DescribeIndexCommand, _super);
    function DescribeIndexCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeIndexCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DescribeIndexCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeIndexRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeIndexResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeIndexCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeIndexCommand(input, context);
    };
    DescribeIndexCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeIndexCommand(output, context);
    };
    return DescribeIndexCommand;
}($Command));
export { DescribeIndexCommand };
