// @aws-sdk/client-iot/dist-es/commands/DescribeAuditFindingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeAuditFindingRequest, DescribeAuditFindingResponse } from "../models/models_1";
import { deserializeAws_restJson1DescribeAuditFindingCommand, serializeAws_restJson1DescribeAuditFindingCommand, } from "../protocols/Aws_restJson1";
var DescribeAuditFindingCommand = (function (_super) {
    __extends(DescribeAuditFindingCommand, _super);
    function DescribeAuditFindingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeAuditFindingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DescribeAuditFindingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeAuditFindingRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeAuditFindingResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeAuditFindingCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeAuditFindingCommand(input, context);
    };
    DescribeAuditFindingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeAuditFindingCommand(output, context);
    };
    return DescribeAuditFindingCommand;
}($Command));
export { DescribeAuditFindingCommand };
