// @aws-sdk/client-iot/dist-es/commands/DescribeFleetMetricCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeFleetMetricRequest, DescribeFleetMetricResponse } from "../models/models_1";
import { deserializeAws_restJson1DescribeFleetMetricCommand, serializeAws_restJson1DescribeFleetMetricCommand, } from "../protocols/Aws_restJson1";
var DescribeFleetMetricCommand = (function (_super) {
    __extends(DescribeFleetMetricCommand, _super);
    function DescribeFleetMetricCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeFleetMetricCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DescribeFleetMetricCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeFleetMetricRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeFleetMetricResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeFleetMetricCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeFleetMetricCommand(input, context);
    };
    DescribeFleetMetricCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeFleetMetricCommand(output, context);
    };
    return DescribeFleetMetricCommand;
}($Command));
export { DescribeFleetMetricCommand };
