// @aws-sdk/client-iot/dist-es/commands/DescribeCustomMetricCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeCustomMetricRequest, DescribeCustomMetricResponse } from "../models/models_1";
import { deserializeAws_restJson1DescribeCustomMetricCommand, serializeAws_restJson1DescribeCustomMetricCommand, } from "../protocols/Aws_restJson1";
var DescribeCustomMetricCommand = (function (_super) {
    __extends(DescribeCustomMetricCommand, _super);
    function DescribeCustomMetricCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeCustomMetricCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DescribeCustomMetricCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeCustomMetricRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeCustomMetricResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeCustomMetricCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeCustomMetricCommand(input, context);
    };
    DescribeCustomMetricCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeCustomMetricCommand(output, context);
    };
    return DescribeCustomMetricCommand;
}($Command));
export { DescribeCustomMetricCommand };
