// @aws-sdk/client-iot/dist-es/commands/DescribeDefaultAuthorizerCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeDefaultAuthorizerRequest, DescribeDefaultAuthorizerResponse } from "../models/models_1";
import { deserializeAws_restJson1DescribeDefaultAuthorizerCommand, serializeAws_restJson1DescribeDefaultAuthorizerCommand, } from "../protocols/Aws_restJson1";
var DescribeDefaultAuthorizerCommand = (function (_super) {
    __extends(DescribeDefaultAuthorizerCommand, _super);
    function DescribeDefaultAuthorizerCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeDefaultAuthorizerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DescribeDefaultAuthorizerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDefaultAuthorizerRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeDefaultAuthorizerResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDefaultAuthorizerCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeDefaultAuthorizerCommand(input, context);
    };
    DescribeDefaultAuthorizerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeDefaultAuthorizerCommand(output, context);
    };
    return DescribeDefaultAuthorizerCommand;
}($Command));
export { DescribeDefaultAuthorizerCommand };
