// @aws-sdk/client-iot/dist-es/commands/DescribeThingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeThingRequest, DescribeThingResponse } from "../models/models_1";
import { deserializeAws_restJson1DescribeThingCommand, serializeAws_restJson1DescribeThingCommand, } from "../protocols/Aws_restJson1";
var DescribeThingCommand = (function (_super) {
    __extends(DescribeThingCommand, _super);
    function DescribeThingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeThingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DescribeThingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeThingRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeThingResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeThingCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeThingCommand(input, context);
    };
    DescribeThingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeThingCommand(output, context);
    };
    return DescribeThingCommand;
}($Command));
export { DescribeThingCommand };
