// @aws-sdk/client-iot/dist-es/commands/DescribeAuthorizerCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeAuthorizerRequest, DescribeAuthorizerResponse } from "../models/models_1";
import { deserializeAws_restJson1DescribeAuthorizerCommand, serializeAws_restJson1DescribeAuthorizerCommand, } from "../protocols/Aws_restJson1";
var DescribeAuthorizerCommand = (function (_super) {
    __extends(DescribeAuthorizerCommand, _super);
    function DescribeAuthorizerCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeAuthorizerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DescribeAuthorizerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeAuthorizerRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeAuthorizerResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeAuthorizerCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeAuthorizerCommand(input, context);
    };
    DescribeAuthorizerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeAuthorizerCommand(output, context);
    };
    return DescribeAuthorizerCommand;
}($Command));
export { DescribeAuthorizerCommand };
