// @aws-sdk/client-iot/dist-es/commands/DescribeProvisioningTemplateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeProvisioningTemplateRequest, DescribeProvisioningTemplateResponse } from "../models/models_1";
import { deserializeAws_restJson1DescribeProvisioningTemplateCommand, serializeAws_restJson1DescribeProvisioningTemplateCommand, } from "../protocols/Aws_restJson1";
var DescribeProvisioningTemplateCommand = (function (_super) {
    __extends(DescribeProvisioningTemplateCommand, _super);
    function DescribeProvisioningTemplateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeProvisioningTemplateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DescribeProvisioningTemplateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeProvisioningTemplateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeProvisioningTemplateResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeProvisioningTemplateCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeProvisioningTemplateCommand(input, context);
    };
    DescribeProvisioningTemplateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeProvisioningTemplateCommand(output, context);
    };
    return DescribeProvisioningTemplateCommand;
}($Command));
export { DescribeProvisioningTemplateCommand };
