// @aws-sdk/client-iot/dist-es/commands/DescribeBillingGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeBillingGroupRequest, DescribeBillingGroupResponse } from "../models/models_1";
import { deserializeAws_restJson1DescribeBillingGroupCommand, serializeAws_restJson1DescribeBillingGroupCommand, } from "../protocols/Aws_restJson1";
var DescribeBillingGroupCommand = (function (_super) {
    __extends(DescribeBillingGroupCommand, _super);
    function DescribeBillingGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeBillingGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DescribeBillingGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeBillingGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeBillingGroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeBillingGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeBillingGroupCommand(input, context);
    };
    DescribeBillingGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeBillingGroupCommand(output, context);
    };
    return DescribeBillingGroupCommand;
}($Command));
export { DescribeBillingGroupCommand };
