// @aws-sdk/client-iot/dist-es/commands/DetachPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DetachPolicyRequest } from "../models/models_1";
import { deserializeAws_restJson1DetachPolicyCommand, serializeAws_restJson1DetachPolicyCommand, } from "../protocols/Aws_restJson1";
var DetachPolicyCommand = (function (_super) {
    __extends(DetachPolicyCommand, _super);
    function DetachPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DetachPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DetachPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DetachPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DetachPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DetachPolicyCommand(input, context);
    };
    DetachPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DetachPolicyCommand(output, context);
    };
    return DetachPolicyCommand;
}($Command));
export { DetachPolicyCommand };
