// @aws-sdk/client-iot/dist-es/commands/DetachThingPrincipalCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DetachThingPrincipalRequest, DetachThingPrincipalResponse } from "../models/models_1";
import { deserializeAws_restJson1DetachThingPrincipalCommand, serializeAws_restJson1DetachThingPrincipalCommand, } from "../protocols/Aws_restJson1";
var DetachThingPrincipalCommand = (function (_super) {
    __extends(DetachThingPrincipalCommand, _super);
    function DetachThingPrincipalCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DetachThingPrincipalCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DetachThingPrincipalCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DetachThingPrincipalRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DetachThingPrincipalResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DetachThingPrincipalCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DetachThingPrincipalCommand(input, context);
    };
    DetachThingPrincipalCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DetachThingPrincipalCommand(output, context);
    };
    return DetachThingPrincipalCommand;
}($Command));
export { DetachThingPrincipalCommand };
