// @aws-sdk/client-iot/dist-es/commands/DetachSecurityProfileCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DetachSecurityProfileRequest, DetachSecurityProfileResponse } from "../models/models_1";
import { deserializeAws_restJson1DetachSecurityProfileCommand, serializeAws_restJson1DetachSecurityProfileCommand, } from "../protocols/Aws_restJson1";
var DetachSecurityProfileCommand = (function (_super) {
    __extends(DetachSecurityProfileCommand, _super);
    function DetachSecurityProfileCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DetachSecurityProfileCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DetachSecurityProfileCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DetachSecurityProfileRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DetachSecurityProfileResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DetachSecurityProfileCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DetachSecurityProfileCommand(input, context);
    };
    DetachSecurityProfileCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DetachSecurityProfileCommand(output, context);
    };
    return DetachSecurityProfileCommand;
}($Command));
export { DetachSecurityProfileCommand };
