// @aws-sdk/client-iot/dist-es/commands/DetachPrincipalPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DetachPrincipalPolicyRequest } from "../models/models_1";
import { deserializeAws_restJson1DetachPrincipalPolicyCommand, serializeAws_restJson1DetachPrincipalPolicyCommand, } from "../protocols/Aws_restJson1";
var DetachPrincipalPolicyCommand = (function (_super) {
    __extends(DetachPrincipalPolicyCommand, _super);
    function DetachPrincipalPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DetachPrincipalPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "DetachPrincipalPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DetachPrincipalPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DetachPrincipalPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DetachPrincipalPolicyCommand(input, context);
    };
    DetachPrincipalPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DetachPrincipalPolicyCommand(output, context);
    };
    return DetachPrincipalPolicyCommand;
}($Command));
export { DetachPrincipalPolicyCommand };
