// @aws-sdk/client-iot/dist-es/commands/GetBehaviorModelTrainingSummariesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetBehaviorModelTrainingSummariesRequest, GetBehaviorModelTrainingSummariesResponse, } from "../models/models_1";
import { deserializeAws_restJson1GetBehaviorModelTrainingSummariesCommand, serializeAws_restJson1GetBehaviorModelTrainingSummariesCommand, } from "../protocols/Aws_restJson1";
var GetBehaviorModelTrainingSummariesCommand = (function (_super) {
    __extends(GetBehaviorModelTrainingSummariesCommand, _super);
    function GetBehaviorModelTrainingSummariesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetBehaviorModelTrainingSummariesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "GetBehaviorModelTrainingSummariesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetBehaviorModelTrainingSummariesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetBehaviorModelTrainingSummariesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetBehaviorModelTrainingSummariesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetBehaviorModelTrainingSummariesCommand(input, context);
    };
    GetBehaviorModelTrainingSummariesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetBehaviorModelTrainingSummariesCommand(output, context);
    };
    return GetBehaviorModelTrainingSummariesCommand;
}($Command));
export { GetBehaviorModelTrainingSummariesCommand };
