// @aws-sdk/client-iot/dist-es/commands/GetCardinalityCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetCardinalityRequest, GetCardinalityResponse } from "../models/models_1";
import { deserializeAws_restJson1GetCardinalityCommand, serializeAws_restJson1GetCardinalityCommand, } from "../protocols/Aws_restJson1";
var GetCardinalityCommand = (function (_super) {
    __extends(GetCardinalityCommand, _super);
    function GetCardinalityCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetCardinalityCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "GetCardinalityCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetCardinalityRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetCardinalityResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetCardinalityCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetCardinalityCommand(input, context);
    };
    GetCardinalityCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetCardinalityCommand(output, context);
    };
    return GetCardinalityCommand;
}($Command));
export { GetCardinalityCommand };
