// @aws-sdk/client-iot/dist-es/commands/GetEffectivePoliciesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetEffectivePoliciesRequest, GetEffectivePoliciesResponse } from "../models/models_1";
import { deserializeAws_restJson1GetEffectivePoliciesCommand, serializeAws_restJson1GetEffectivePoliciesCommand, } from "../protocols/Aws_restJson1";
var GetEffectivePoliciesCommand = (function (_super) {
    __extends(GetEffectivePoliciesCommand, _super);
    function GetEffectivePoliciesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetEffectivePoliciesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "GetEffectivePoliciesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetEffectivePoliciesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetEffectivePoliciesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetEffectivePoliciesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetEffectivePoliciesCommand(input, context);
    };
    GetEffectivePoliciesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetEffectivePoliciesCommand(output, context);
    };
    return GetEffectivePoliciesCommand;
}($Command));
export { GetEffectivePoliciesCommand };
