// @aws-sdk/client-iot/dist-es/commands/GetIndexingConfigurationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetIndexingConfigurationRequest, GetIndexingConfigurationResponse } from "../models/models_1";
import { deserializeAws_restJson1GetIndexingConfigurationCommand, serializeAws_restJson1GetIndexingConfigurationCommand, } from "../protocols/Aws_restJson1";
var GetIndexingConfigurationCommand = (function (_super) {
    __extends(GetIndexingConfigurationCommand, _super);
    function GetIndexingConfigurationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetIndexingConfigurationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "GetIndexingConfigurationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetIndexingConfigurationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetIndexingConfigurationResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetIndexingConfigurationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetIndexingConfigurationCommand(input, context);
    };
    GetIndexingConfigurationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetIndexingConfigurationCommand(output, context);
    };
    return GetIndexingConfigurationCommand;
}($Command));
export { GetIndexingConfigurationCommand };
