// @aws-sdk/client-iot/dist-es/commands/GetOTAUpdateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetOTAUpdateRequest, GetOTAUpdateResponse } from "../models/models_1";
import { deserializeAws_restJson1GetOTAUpdateCommand, serializeAws_restJson1GetOTAUpdateCommand, } from "../protocols/Aws_restJson1";
var GetOTAUpdateCommand = (function (_super) {
    __extends(GetOTAUpdateCommand, _super);
    function GetOTAUpdateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetOTAUpdateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "GetOTAUpdateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetOTAUpdateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetOTAUpdateResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetOTAUpdateCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetOTAUpdateCommand(input, context);
    };
    GetOTAUpdateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetOTAUpdateCommand(output, context);
    };
    return GetOTAUpdateCommand;
}($Command));
export { GetOTAUpdateCommand };
