// @aws-sdk/client-iot/dist-es/commands/GetPercentilesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetPercentilesRequest, GetPercentilesResponse } from "../models/models_1";
import { deserializeAws_restJson1GetPercentilesCommand, serializeAws_restJson1GetPercentilesCommand, } from "../protocols/Aws_restJson1";
var GetPercentilesCommand = (function (_super) {
    __extends(GetPercentilesCommand, _super);
    function GetPercentilesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetPercentilesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "GetPercentilesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetPercentilesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetPercentilesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetPercentilesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetPercentilesCommand(input, context);
    };
    GetPercentilesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetPercentilesCommand(output, context);
    };
    return GetPercentilesCommand;
}($Command));
export { GetPercentilesCommand };
