// @aws-sdk/client-iot/dist-es/commands/GetTopicRuleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetTopicRuleRequest, GetTopicRuleResponse } from "../models/models_1";
import { deserializeAws_restJson1GetTopicRuleCommand, serializeAws_restJson1GetTopicRuleCommand, } from "../protocols/Aws_restJson1";
var GetTopicRuleCommand = (function (_super) {
    __extends(GetTopicRuleCommand, _super);
    function GetTopicRuleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetTopicRuleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "GetTopicRuleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetTopicRuleRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetTopicRuleResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetTopicRuleCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetTopicRuleCommand(input, context);
    };
    GetTopicRuleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetTopicRuleCommand(output, context);
    };
    return GetTopicRuleCommand;
}($Command));
export { GetTopicRuleCommand };
