// @aws-sdk/client-iot/dist-es/commands/GetTopicRuleDestinationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetTopicRuleDestinationRequest, GetTopicRuleDestinationResponse } from "../models/models_1";
import { deserializeAws_restJson1GetTopicRuleDestinationCommand, serializeAws_restJson1GetTopicRuleDestinationCommand, } from "../protocols/Aws_restJson1";
var GetTopicRuleDestinationCommand = (function (_super) {
    __extends(GetTopicRuleDestinationCommand, _super);
    function GetTopicRuleDestinationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetTopicRuleDestinationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "GetTopicRuleDestinationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetTopicRuleDestinationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetTopicRuleDestinationResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetTopicRuleDestinationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetTopicRuleDestinationCommand(input, context);
    };
    GetTopicRuleDestinationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetTopicRuleDestinationCommand(output, context);
    };
    return GetTopicRuleDestinationCommand;
}($Command));
export { GetTopicRuleDestinationCommand };
