// @aws-sdk/client-iot/dist-es/commands/ListFleetMetricsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListFleetMetricsRequest, ListFleetMetricsResponse } from "../models/models_1";
import { deserializeAws_restJson1ListFleetMetricsCommand, serializeAws_restJson1ListFleetMetricsCommand, } from "../protocols/Aws_restJson1";
var ListFleetMetricsCommand = (function (_super) {
    __extends(ListFleetMetricsCommand, _super);
    function ListFleetMetricsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListFleetMetricsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListFleetMetricsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListFleetMetricsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListFleetMetricsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListFleetMetricsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListFleetMetricsCommand(input, context);
    };
    ListFleetMetricsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListFleetMetricsCommand(output, context);
    };
    return ListFleetMetricsCommand;
}($Command));
export { ListFleetMetricsCommand };
