// @aws-sdk/client-iot/dist-es/commands/ListCustomMetricsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListCustomMetricsRequest, ListCustomMetricsResponse } from "../models/models_1";
import { deserializeAws_restJson1ListCustomMetricsCommand, serializeAws_restJson1ListCustomMetricsCommand, } from "../protocols/Aws_restJson1";
var ListCustomMetricsCommand = (function (_super) {
    __extends(ListCustomMetricsCommand, _super);
    function ListCustomMetricsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListCustomMetricsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListCustomMetricsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListCustomMetricsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListCustomMetricsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListCustomMetricsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListCustomMetricsCommand(input, context);
    };
    ListCustomMetricsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListCustomMetricsCommand(output, context);
    };
    return ListCustomMetricsCommand;
}($Command));
export { ListCustomMetricsCommand };
