// @aws-sdk/client-iot/dist-es/commands/ListProvisioningTemplateVersionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListProvisioningTemplateVersionsRequest, ListProvisioningTemplateVersionsResponse } from "../models/models_1";
import { deserializeAws_restJson1ListProvisioningTemplateVersionsCommand, serializeAws_restJson1ListProvisioningTemplateVersionsCommand, } from "../protocols/Aws_restJson1";
var ListProvisioningTemplateVersionsCommand = (function (_super) {
    __extends(ListProvisioningTemplateVersionsCommand, _super);
    function ListProvisioningTemplateVersionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListProvisioningTemplateVersionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListProvisioningTemplateVersionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListProvisioningTemplateVersionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListProvisioningTemplateVersionsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListProvisioningTemplateVersionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListProvisioningTemplateVersionsCommand(input, context);
    };
    ListProvisioningTemplateVersionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListProvisioningTemplateVersionsCommand(output, context);
    };
    return ListProvisioningTemplateVersionsCommand;
}($Command));
export { ListProvisioningTemplateVersionsCommand };
