// @aws-sdk/client-iot/dist-es/commands/ListBillingGroupsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListBillingGroupsRequest, ListBillingGroupsResponse } from "../models/models_1";
import { deserializeAws_restJson1ListBillingGroupsCommand, serializeAws_restJson1ListBillingGroupsCommand, } from "../protocols/Aws_restJson1";
var ListBillingGroupsCommand = (function (_super) {
    __extends(ListBillingGroupsCommand, _super);
    function ListBillingGroupsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListBillingGroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListBillingGroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListBillingGroupsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListBillingGroupsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListBillingGroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListBillingGroupsCommand(input, context);
    };
    ListBillingGroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListBillingGroupsCommand(output, context);
    };
    return ListBillingGroupsCommand;
}($Command));
export { ListBillingGroupsCommand };
