// @aws-sdk/client-iot/dist-es/commands/ListActiveViolationsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListActiveViolationsRequest, ListActiveViolationsResponse } from "../models/models_1";
import { deserializeAws_restJson1ListActiveViolationsCommand, serializeAws_restJson1ListActiveViolationsCommand, } from "../protocols/Aws_restJson1";
var ListActiveViolationsCommand = (function (_super) {
    __extends(ListActiveViolationsCommand, _super);
    function ListActiveViolationsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListActiveViolationsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListActiveViolationsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListActiveViolationsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListActiveViolationsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListActiveViolationsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListActiveViolationsCommand(input, context);
    };
    ListActiveViolationsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListActiveViolationsCommand(output, context);
    };
    return ListActiveViolationsCommand;
}($Command));
export { ListActiveViolationsCommand };
