// @aws-sdk/client-iot/dist-es/commands/ListCertificatesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListCertificatesRequest, ListCertificatesResponse } from "../models/models_1";
import { deserializeAws_restJson1ListCertificatesCommand, serializeAws_restJson1ListCertificatesCommand, } from "../protocols/Aws_restJson1";
var ListCertificatesCommand = (function (_super) {
    __extends(ListCertificatesCommand, _super);
    function ListCertificatesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListCertificatesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListCertificatesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListCertificatesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListCertificatesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListCertificatesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListCertificatesCommand(input, context);
    };
    ListCertificatesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListCertificatesCommand(output, context);
    };
    return ListCertificatesCommand;
}($Command));
export { ListCertificatesCommand };
