// @aws-sdk/client-iot/dist-es/commands/ListJobExecutionsForThingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListJobExecutionsForThingRequest, ListJobExecutionsForThingResponse } from "../models/models_1";
import { deserializeAws_restJson1ListJobExecutionsForThingCommand, serializeAws_restJson1ListJobExecutionsForThingCommand, } from "../protocols/Aws_restJson1";
var ListJobExecutionsForThingCommand = (function (_super) {
    __extends(ListJobExecutionsForThingCommand, _super);
    function ListJobExecutionsForThingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListJobExecutionsForThingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListJobExecutionsForThingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListJobExecutionsForThingRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListJobExecutionsForThingResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListJobExecutionsForThingCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListJobExecutionsForThingCommand(input, context);
    };
    ListJobExecutionsForThingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListJobExecutionsForThingCommand(output, context);
    };
    return ListJobExecutionsForThingCommand;
}($Command));
export { ListJobExecutionsForThingCommand };
