// @aws-sdk/client-iot/dist-es/commands/ListThingsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListThingsRequest, ListThingsResponse } from "../models/models_2";
import { deserializeAws_restJson1ListThingsCommand, serializeAws_restJson1ListThingsCommand, } from "../protocols/Aws_restJson1";
var ListThingsCommand = (function (_super) {
    __extends(ListThingsCommand, _super);
    function ListThingsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListThingsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListThingsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListThingsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListThingsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListThingsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListThingsCommand(input, context);
    };
    ListThingsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListThingsCommand(output, context);
    };
    return ListThingsCommand;
}($Command));
export { ListThingsCommand };
