// @aws-sdk/client-iot/dist-es/commands/ListThingGroupsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListThingGroupsRequest, ListThingGroupsResponse } from "../models/models_1";
import { deserializeAws_restJson1ListThingGroupsCommand, serializeAws_restJson1ListThingGroupsCommand, } from "../protocols/Aws_restJson1";
var ListThingGroupsCommand = (function (_super) {
    __extends(ListThingGroupsCommand, _super);
    function ListThingGroupsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListThingGroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListThingGroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListThingGroupsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListThingGroupsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListThingGroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListThingGroupsCommand(input, context);
    };
    ListThingGroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListThingGroupsCommand(output, context);
    };
    return ListThingGroupsCommand;
}($Command));
export { ListThingGroupsCommand };
