// @aws-sdk/client-iot/dist-es/commands/ListSecurityProfilesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListSecurityProfilesRequest, ListSecurityProfilesResponse } from "../models/models_1";
import { deserializeAws_restJson1ListSecurityProfilesCommand, serializeAws_restJson1ListSecurityProfilesCommand, } from "../protocols/Aws_restJson1";
var ListSecurityProfilesCommand = (function (_super) {
    __extends(ListSecurityProfilesCommand, _super);
    function ListSecurityProfilesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListSecurityProfilesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListSecurityProfilesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListSecurityProfilesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListSecurityProfilesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListSecurityProfilesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListSecurityProfilesCommand(input, context);
    };
    ListSecurityProfilesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListSecurityProfilesCommand(output, context);
    };
    return ListSecurityProfilesCommand;
}($Command));
export { ListSecurityProfilesCommand };
