// @aws-sdk/client-iot/dist-es/commands/ListScheduledAuditsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListScheduledAuditsRequest, ListScheduledAuditsResponse } from "../models/models_1";
import { deserializeAws_restJson1ListScheduledAuditsCommand, serializeAws_restJson1ListScheduledAuditsCommand, } from "../protocols/Aws_restJson1";
var ListScheduledAuditsCommand = (function (_super) {
    __extends(ListScheduledAuditsCommand, _super);
    function ListScheduledAuditsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListScheduledAuditsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListScheduledAuditsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListScheduledAuditsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListScheduledAuditsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListScheduledAuditsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListScheduledAuditsCommand(input, context);
    };
    ListScheduledAuditsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListScheduledAuditsCommand(output, context);
    };
    return ListScheduledAuditsCommand;
}($Command));
export { ListScheduledAuditsCommand };
