// @aws-sdk/client-iot/dist-es/commands/ListPrincipalPoliciesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListPrincipalPoliciesRequest, ListPrincipalPoliciesResponse } from "../models/models_1";
import { deserializeAws_restJson1ListPrincipalPoliciesCommand, serializeAws_restJson1ListPrincipalPoliciesCommand, } from "../protocols/Aws_restJson1";
var ListPrincipalPoliciesCommand = (function (_super) {
    __extends(ListPrincipalPoliciesCommand, _super);
    function ListPrincipalPoliciesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListPrincipalPoliciesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListPrincipalPoliciesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListPrincipalPoliciesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListPrincipalPoliciesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListPrincipalPoliciesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListPrincipalPoliciesCommand(input, context);
    };
    ListPrincipalPoliciesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListPrincipalPoliciesCommand(output, context);
    };
    return ListPrincipalPoliciesCommand;
}($Command));
export { ListPrincipalPoliciesCommand };
