// @aws-sdk/client-iot/dist-es/commands/ListManagedJobTemplatesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListManagedJobTemplatesRequest, ListManagedJobTemplatesResponse } from "../models/models_1";
import { deserializeAws_restJson1ListManagedJobTemplatesCommand, serializeAws_restJson1ListManagedJobTemplatesCommand, } from "../protocols/Aws_restJson1";
var ListManagedJobTemplatesCommand = (function (_super) {
    __extends(ListManagedJobTemplatesCommand, _super);
    function ListManagedJobTemplatesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListManagedJobTemplatesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListManagedJobTemplatesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListManagedJobTemplatesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListManagedJobTemplatesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListManagedJobTemplatesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListManagedJobTemplatesCommand(input, context);
    };
    ListManagedJobTemplatesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListManagedJobTemplatesCommand(output, context);
    };
    return ListManagedJobTemplatesCommand;
}($Command));
export { ListManagedJobTemplatesCommand };
