// @aws-sdk/client-iot/dist-es/commands/ListThingsInBillingGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListThingsInBillingGroupRequest, ListThingsInBillingGroupResponse } from "../models/models_2";
import { deserializeAws_restJson1ListThingsInBillingGroupCommand, serializeAws_restJson1ListThingsInBillingGroupCommand, } from "../protocols/Aws_restJson1";
var ListThingsInBillingGroupCommand = (function (_super) {
    __extends(ListThingsInBillingGroupCommand, _super);
    function ListThingsInBillingGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListThingsInBillingGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "ListThingsInBillingGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListThingsInBillingGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListThingsInBillingGroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListThingsInBillingGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListThingsInBillingGroupCommand(input, context);
    };
    ListThingsInBillingGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListThingsInBillingGroupCommand(output, context);
    };
    return ListThingsInBillingGroupCommand;
}($Command));
export { ListThingsInBillingGroupCommand };
