// @aws-sdk/client-iot/dist-es/commands/PutVerificationStateOnViolationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PutVerificationStateOnViolationRequest, PutVerificationStateOnViolationResponse } from "../models/models_2";
import { deserializeAws_restJson1PutVerificationStateOnViolationCommand, serializeAws_restJson1PutVerificationStateOnViolationCommand, } from "../protocols/Aws_restJson1";
var PutVerificationStateOnViolationCommand = (function (_super) {
    __extends(PutVerificationStateOnViolationCommand, _super);
    function PutVerificationStateOnViolationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PutVerificationStateOnViolationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "PutVerificationStateOnViolationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutVerificationStateOnViolationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: PutVerificationStateOnViolationResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutVerificationStateOnViolationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1PutVerificationStateOnViolationCommand(input, context);
    };
    PutVerificationStateOnViolationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1PutVerificationStateOnViolationCommand(output, context);
    };
    return PutVerificationStateOnViolationCommand;
}($Command));
export { PutVerificationStateOnViolationCommand };
