// @aws-sdk/client-iot/dist-es/commands/RegisterCertificateWithoutCACommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RegisterCertificateWithoutCARequest, RegisterCertificateWithoutCAResponse } from "../models/models_2";
import { deserializeAws_restJson1RegisterCertificateWithoutCACommand, serializeAws_restJson1RegisterCertificateWithoutCACommand, } from "../protocols/Aws_restJson1";
var RegisterCertificateWithoutCACommand = (function (_super) {
    __extends(RegisterCertificateWithoutCACommand, _super);
    function RegisterCertificateWithoutCACommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RegisterCertificateWithoutCACommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "RegisterCertificateWithoutCACommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RegisterCertificateWithoutCARequest.filterSensitiveLog,
            outputFilterSensitiveLog: RegisterCertificateWithoutCAResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RegisterCertificateWithoutCACommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1RegisterCertificateWithoutCACommand(input, context);
    };
    RegisterCertificateWithoutCACommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1RegisterCertificateWithoutCACommand(output, context);
    };
    return RegisterCertificateWithoutCACommand;
}($Command));
export { RegisterCertificateWithoutCACommand };
