// @aws-sdk/client-iot/dist-es/commands/RemoveThingFromBillingGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RemoveThingFromBillingGroupRequest, RemoveThingFromBillingGroupResponse } from "../models/models_2";
import { deserializeAws_restJson1RemoveThingFromBillingGroupCommand, serializeAws_restJson1RemoveThingFromBillingGroupCommand, } from "../protocols/Aws_restJson1";
var RemoveThingFromBillingGroupCommand = (function (_super) {
    __extends(RemoveThingFromBillingGroupCommand, _super);
    function RemoveThingFromBillingGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RemoveThingFromBillingGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "RemoveThingFromBillingGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RemoveThingFromBillingGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: RemoveThingFromBillingGroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RemoveThingFromBillingGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1RemoveThingFromBillingGroupCommand(input, context);
    };
    RemoveThingFromBillingGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1RemoveThingFromBillingGroupCommand(output, context);
    };
    return RemoveThingFromBillingGroupCommand;
}($Command));
export { RemoveThingFromBillingGroupCommand };
