// @aws-sdk/client-iot/dist-es/commands/SearchIndexCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SearchIndexRequest, SearchIndexResponse } from "../models/models_2";
import { deserializeAws_restJson1SearchIndexCommand, serializeAws_restJson1SearchIndexCommand, } from "../protocols/Aws_restJson1";
var SearchIndexCommand = (function (_super) {
    __extends(SearchIndexCommand, _super);
    function SearchIndexCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    SearchIndexCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "SearchIndexCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SearchIndexRequest.filterSensitiveLog,
            outputFilterSensitiveLog: SearchIndexResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SearchIndexCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1SearchIndexCommand(input, context);
    };
    SearchIndexCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1SearchIndexCommand(output, context);
    };
    return SearchIndexCommand;
}($Command));
export { SearchIndexCommand };
