// @aws-sdk/client-iot/dist-es/commands/SetDefaultAuthorizerCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SetDefaultAuthorizerRequest, SetDefaultAuthorizerResponse } from "../models/models_2";
import { deserializeAws_restJson1SetDefaultAuthorizerCommand, serializeAws_restJson1SetDefaultAuthorizerCommand, } from "../protocols/Aws_restJson1";
var SetDefaultAuthorizerCommand = (function (_super) {
    __extends(SetDefaultAuthorizerCommand, _super);
    function SetDefaultAuthorizerCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    SetDefaultAuthorizerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "SetDefaultAuthorizerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SetDefaultAuthorizerRequest.filterSensitiveLog,
            outputFilterSensitiveLog: SetDefaultAuthorizerResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SetDefaultAuthorizerCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1SetDefaultAuthorizerCommand(input, context);
    };
    SetDefaultAuthorizerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1SetDefaultAuthorizerCommand(output, context);
    };
    return SetDefaultAuthorizerCommand;
}($Command));
export { SetDefaultAuthorizerCommand };
