// @aws-sdk/client-iot/dist-es/commands/SetV2LoggingOptionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SetV2LoggingOptionsRequest } from "../models/models_2";
import { deserializeAws_restJson1SetV2LoggingOptionsCommand, serializeAws_restJson1SetV2LoggingOptionsCommand, } from "../protocols/Aws_restJson1";
var SetV2LoggingOptionsCommand = (function (_super) {
    __extends(SetV2LoggingOptionsCommand, _super);
    function SetV2LoggingOptionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    SetV2LoggingOptionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "SetV2LoggingOptionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SetV2LoggingOptionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SetV2LoggingOptionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1SetV2LoggingOptionsCommand(input, context);
    };
    SetV2LoggingOptionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1SetV2LoggingOptionsCommand(output, context);
    };
    return SetV2LoggingOptionsCommand;
}($Command));
export { SetV2LoggingOptionsCommand };
