// @aws-sdk/client-iot/dist-es/commands/StartDetectMitigationActionsTaskCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartDetectMitigationActionsTaskRequest, StartDetectMitigationActionsTaskResponse } from "../models/models_2";
import { deserializeAws_restJson1StartDetectMitigationActionsTaskCommand, serializeAws_restJson1StartDetectMitigationActionsTaskCommand, } from "../protocols/Aws_restJson1";
var StartDetectMitigationActionsTaskCommand = (function (_super) {
    __extends(StartDetectMitigationActionsTaskCommand, _super);
    function StartDetectMitigationActionsTaskCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartDetectMitigationActionsTaskCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "StartDetectMitigationActionsTaskCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartDetectMitigationActionsTaskRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StartDetectMitigationActionsTaskResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartDetectMitigationActionsTaskCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1StartDetectMitigationActionsTaskCommand(input, context);
    };
    StartDetectMitigationActionsTaskCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1StartDetectMitigationActionsTaskCommand(output, context);
    };
    return StartDetectMitigationActionsTaskCommand;
}($Command));
export { StartDetectMitigationActionsTaskCommand };
