// @aws-sdk/client-iot/dist-es/commands/TransferCertificateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { TransferCertificateRequest, TransferCertificateResponse } from "../models/models_2";
import { deserializeAws_restJson1TransferCertificateCommand, serializeAws_restJson1TransferCertificateCommand, } from "../protocols/Aws_restJson1";
var TransferCertificateCommand = (function (_super) {
    __extends(TransferCertificateCommand, _super);
    function TransferCertificateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    TransferCertificateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "TransferCertificateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: TransferCertificateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: TransferCertificateResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    TransferCertificateCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1TransferCertificateCommand(input, context);
    };
    TransferCertificateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1TransferCertificateCommand(output, context);
    };
    return TransferCertificateCommand;
}($Command));
export { TransferCertificateCommand };
