// @aws-sdk/client-iot/dist-es/commands/UpdateThingGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateThingGroupRequest, UpdateThingGroupResponse } from "../models/models_2";
import { deserializeAws_restJson1UpdateThingGroupCommand, serializeAws_restJson1UpdateThingGroupCommand, } from "../protocols/Aws_restJson1";
var UpdateThingGroupCommand = (function (_super) {
    __extends(UpdateThingGroupCommand, _super);
    function UpdateThingGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateThingGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "UpdateThingGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateThingGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateThingGroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateThingGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateThingGroupCommand(input, context);
    };
    UpdateThingGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateThingGroupCommand(output, context);
    };
    return UpdateThingGroupCommand;
}($Command));
export { UpdateThingGroupCommand };
