// @aws-sdk/client-iot/dist-es/commands/UpdateJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateJobRequest } from "../models/models_2";
import { deserializeAws_restJson1UpdateJobCommand, serializeAws_restJson1UpdateJobCommand, } from "../protocols/Aws_restJson1";
var UpdateJobCommand = (function (_super) {
    __extends(UpdateJobCommand, _super);
    function UpdateJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "UpdateJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateJobRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateJobCommand(input, context);
    };
    UpdateJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateJobCommand(output, context);
    };
    return UpdateJobCommand;
}($Command));
export { UpdateJobCommand };
