// @aws-sdk/client-iot/dist-es/commands/UpdateIndexingConfigurationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateIndexingConfigurationRequest, UpdateIndexingConfigurationResponse } from "../models/models_2";
import { deserializeAws_restJson1UpdateIndexingConfigurationCommand, serializeAws_restJson1UpdateIndexingConfigurationCommand, } from "../protocols/Aws_restJson1";
var UpdateIndexingConfigurationCommand = (function (_super) {
    __extends(UpdateIndexingConfigurationCommand, _super);
    function UpdateIndexingConfigurationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateIndexingConfigurationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "UpdateIndexingConfigurationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateIndexingConfigurationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateIndexingConfigurationResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateIndexingConfigurationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateIndexingConfigurationCommand(input, context);
    };
    UpdateIndexingConfigurationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateIndexingConfigurationCommand(output, context);
    };
    return UpdateIndexingConfigurationCommand;
}($Command));
export { UpdateIndexingConfigurationCommand };
