// @aws-sdk/client-iot/dist-es/commands/UpdateBillingGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateBillingGroupRequest, UpdateBillingGroupResponse } from "../models/models_2";
import { deserializeAws_restJson1UpdateBillingGroupCommand, serializeAws_restJson1UpdateBillingGroupCommand, } from "../protocols/Aws_restJson1";
var UpdateBillingGroupCommand = (function (_super) {
    __extends(UpdateBillingGroupCommand, _super);
    function UpdateBillingGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateBillingGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "UpdateBillingGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateBillingGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateBillingGroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateBillingGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateBillingGroupCommand(input, context);
    };
    UpdateBillingGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateBillingGroupCommand(output, context);
    };
    return UpdateBillingGroupCommand;
}($Command));
export { UpdateBillingGroupCommand };
