// @aws-sdk/client-iot/dist-es/commands/UpdateCACertificateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateCACertificateRequest } from "../models/models_2";
import { deserializeAws_restJson1UpdateCACertificateCommand, serializeAws_restJson1UpdateCACertificateCommand, } from "../protocols/Aws_restJson1";
var UpdateCACertificateCommand = (function (_super) {
    __extends(UpdateCACertificateCommand, _super);
    function UpdateCACertificateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateCACertificateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "UpdateCACertificateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateCACertificateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateCACertificateCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateCACertificateCommand(input, context);
    };
    UpdateCACertificateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateCACertificateCommand(output, context);
    };
    return UpdateCACertificateCommand;
}($Command));
export { UpdateCACertificateCommand };
