// @aws-sdk/client-iot/dist-es/commands/UpdateCertificateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateCertificateRequest } from "../models/models_2";
import { deserializeAws_restJson1UpdateCertificateCommand, serializeAws_restJson1UpdateCertificateCommand, } from "../protocols/Aws_restJson1";
var UpdateCertificateCommand = (function (_super) {
    __extends(UpdateCertificateCommand, _super);
    function UpdateCertificateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateCertificateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "UpdateCertificateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateCertificateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateCertificateCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateCertificateCommand(input, context);
    };
    UpdateCertificateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateCertificateCommand(output, context);
    };
    return UpdateCertificateCommand;
}($Command));
export { UpdateCertificateCommand };
