// @aws-sdk/client-iot/dist-es/commands/UpdateCustomMetricCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateCustomMetricRequest, UpdateCustomMetricResponse } from "../models/models_2";
import { deserializeAws_restJson1UpdateCustomMetricCommand, serializeAws_restJson1UpdateCustomMetricCommand, } from "../protocols/Aws_restJson1";
var UpdateCustomMetricCommand = (function (_super) {
    __extends(UpdateCustomMetricCommand, _super);
    function UpdateCustomMetricCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateCustomMetricCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "UpdateCustomMetricCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateCustomMetricRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateCustomMetricResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateCustomMetricCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateCustomMetricCommand(input, context);
    };
    UpdateCustomMetricCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateCustomMetricCommand(output, context);
    };
    return UpdateCustomMetricCommand;
}($Command));
export { UpdateCustomMetricCommand };
