// @aws-sdk/client-iot/dist-es/commands/UpdateDomainConfigurationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateDomainConfigurationRequest, UpdateDomainConfigurationResponse } from "../models/models_2";
import { deserializeAws_restJson1UpdateDomainConfigurationCommand, serializeAws_restJson1UpdateDomainConfigurationCommand, } from "../protocols/Aws_restJson1";
var UpdateDomainConfigurationCommand = (function (_super) {
    __extends(UpdateDomainConfigurationCommand, _super);
    function UpdateDomainConfigurationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateDomainConfigurationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "UpdateDomainConfigurationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateDomainConfigurationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateDomainConfigurationResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateDomainConfigurationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateDomainConfigurationCommand(input, context);
    };
    UpdateDomainConfigurationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateDomainConfigurationCommand(output, context);
    };
    return UpdateDomainConfigurationCommand;
}($Command));
export { UpdateDomainConfigurationCommand };
