// @aws-sdk/client-iot/dist-es/commands/UpdateThingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateThingRequest, UpdateThingResponse } from "../models/models_2";
import { deserializeAws_restJson1UpdateThingCommand, serializeAws_restJson1UpdateThingCommand, } from "../protocols/Aws_restJson1";
var UpdateThingCommand = (function (_super) {
    __extends(UpdateThingCommand, _super);
    function UpdateThingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateThingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "UpdateThingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateThingRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateThingResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateThingCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateThingCommand(input, context);
    };
    UpdateThingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateThingCommand(output, context);
    };
    return UpdateThingCommand;
}($Command));
export { UpdateThingCommand };
