// @aws-sdk/client-iot/dist-es/commands/UpdateEventConfigurationsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateEventConfigurationsRequest, UpdateEventConfigurationsResponse } from "../models/models_2";
import { deserializeAws_restJson1UpdateEventConfigurationsCommand, serializeAws_restJson1UpdateEventConfigurationsCommand, } from "../protocols/Aws_restJson1";
var UpdateEventConfigurationsCommand = (function (_super) {
    __extends(UpdateEventConfigurationsCommand, _super);
    function UpdateEventConfigurationsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateEventConfigurationsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IoTClient";
        var commandName = "UpdateEventConfigurationsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateEventConfigurationsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateEventConfigurationsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateEventConfigurationsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateEventConfigurationsCommand(input, context);
    };
    UpdateEventConfigurationsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateEventConfigurationsCommand(output, context);
    };
    return UpdateEventConfigurationsCommand;
}($Command));
export { UpdateEventConfigurationsCommand };
