// @aws-sdk/client-iot/dist-es/models/models_1.js
import { __assign } from "tslib";
export var DeleteScheduledAuditRequest;
(function (DeleteScheduledAuditRequest) {
    DeleteScheduledAuditRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteScheduledAuditRequest || (DeleteScheduledAuditRequest = {}));
export var DeleteScheduledAuditResponse;
(function (DeleteScheduledAuditResponse) {
    DeleteScheduledAuditResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteScheduledAuditResponse || (DeleteScheduledAuditResponse = {}));
export var DeleteSecurityProfileRequest;
(function (DeleteSecurityProfileRequest) {
    DeleteSecurityProfileRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteSecurityProfileRequest || (DeleteSecurityProfileRequest = {}));
export var DeleteSecurityProfileResponse;
(function (DeleteSecurityProfileResponse) {
    DeleteSecurityProfileResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteSecurityProfileResponse || (DeleteSecurityProfileResponse = {}));
export var DeleteStreamRequest;
(function (DeleteStreamRequest) {
    DeleteStreamRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteStreamRequest || (DeleteStreamRequest = {}));
export var DeleteStreamResponse;
(function (DeleteStreamResponse) {
    DeleteStreamResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteStreamResponse || (DeleteStreamResponse = {}));
export var DeleteThingRequest;
(function (DeleteThingRequest) {
    DeleteThingRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteThingRequest || (DeleteThingRequest = {}));
export var DeleteThingResponse;
(function (DeleteThingResponse) {
    DeleteThingResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteThingResponse || (DeleteThingResponse = {}));
export var DeleteThingGroupRequest;
(function (DeleteThingGroupRequest) {
    DeleteThingGroupRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteThingGroupRequest || (DeleteThingGroupRequest = {}));
export var DeleteThingGroupResponse;
(function (DeleteThingGroupResponse) {
    DeleteThingGroupResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteThingGroupResponse || (DeleteThingGroupResponse = {}));
export var DeleteThingTypeRequest;
(function (DeleteThingTypeRequest) {
    DeleteThingTypeRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteThingTypeRequest || (DeleteThingTypeRequest = {}));
export var DeleteThingTypeResponse;
(function (DeleteThingTypeResponse) {
    DeleteThingTypeResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteThingTypeResponse || (DeleteThingTypeResponse = {}));
export var DeleteTopicRuleRequest;
(function (DeleteTopicRuleRequest) {
    DeleteTopicRuleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteTopicRuleRequest || (DeleteTopicRuleRequest = {}));
export var DeleteTopicRuleDestinationRequest;
(function (DeleteTopicRuleDestinationRequest) {
    DeleteTopicRuleDestinationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteTopicRuleDestinationRequest || (DeleteTopicRuleDestinationRequest = {}));
export var DeleteTopicRuleDestinationResponse;
(function (DeleteTopicRuleDestinationResponse) {
    DeleteTopicRuleDestinationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteTopicRuleDestinationResponse || (DeleteTopicRuleDestinationResponse = {}));
export var LogTargetType;
(function (LogTargetType) {
    LogTargetType["DEFAULT"] = "DEFAULT";
    LogTargetType["THING_GROUP"] = "THING_GROUP";
})(LogTargetType || (LogTargetType = {}));
export var DeleteV2LoggingLevelRequest;
(function (DeleteV2LoggingLevelRequest) {
    DeleteV2LoggingLevelRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteV2LoggingLevelRequest || (DeleteV2LoggingLevelRequest = {}));
export var DeprecateThingTypeRequest;
(function (DeprecateThingTypeRequest) {
    DeprecateThingTypeRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeprecateThingTypeRequest || (DeprecateThingTypeRequest = {}));
export var DeprecateThingTypeResponse;
(function (DeprecateThingTypeResponse) {
    DeprecateThingTypeResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeprecateThingTypeResponse || (DeprecateThingTypeResponse = {}));
export var DescribeAccountAuditConfigurationRequest;
(function (DescribeAccountAuditConfigurationRequest) {
    DescribeAccountAuditConfigurationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAccountAuditConfigurationRequest || (DescribeAccountAuditConfigurationRequest = {}));
export var DescribeAccountAuditConfigurationResponse;
(function (DescribeAccountAuditConfigurationResponse) {
    DescribeAccountAuditConfigurationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAccountAuditConfigurationResponse || (DescribeAccountAuditConfigurationResponse = {}));
export var DescribeAuditFindingRequest;
(function (DescribeAuditFindingRequest) {
    DescribeAuditFindingRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAuditFindingRequest || (DescribeAuditFindingRequest = {}));
export var DescribeAuditFindingResponse;
(function (DescribeAuditFindingResponse) {
    DescribeAuditFindingResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAuditFindingResponse || (DescribeAuditFindingResponse = {}));
export var DescribeAuditMitigationActionsTaskRequest;
(function (DescribeAuditMitigationActionsTaskRequest) {
    DescribeAuditMitigationActionsTaskRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAuditMitigationActionsTaskRequest || (DescribeAuditMitigationActionsTaskRequest = {}));
export var MitigationAction;
(function (MitigationAction) {
    MitigationAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MitigationAction || (MitigationAction = {}));
export var DescribeAuditMitigationActionsTaskResponse;
(function (DescribeAuditMitigationActionsTaskResponse) {
    DescribeAuditMitigationActionsTaskResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAuditMitigationActionsTaskResponse || (DescribeAuditMitigationActionsTaskResponse = {}));
export var DescribeAuditSuppressionRequest;
(function (DescribeAuditSuppressionRequest) {
    DescribeAuditSuppressionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAuditSuppressionRequest || (DescribeAuditSuppressionRequest = {}));
export var DescribeAuditSuppressionResponse;
(function (DescribeAuditSuppressionResponse) {
    DescribeAuditSuppressionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAuditSuppressionResponse || (DescribeAuditSuppressionResponse = {}));
export var DescribeAuditTaskRequest;
(function (DescribeAuditTaskRequest) {
    DescribeAuditTaskRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAuditTaskRequest || (DescribeAuditTaskRequest = {}));
export var TaskStatistics;
(function (TaskStatistics) {
    TaskStatistics.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TaskStatistics || (TaskStatistics = {}));
export var DescribeAuditTaskResponse;
(function (DescribeAuditTaskResponse) {
    DescribeAuditTaskResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAuditTaskResponse || (DescribeAuditTaskResponse = {}));
export var DescribeAuthorizerRequest;
(function (DescribeAuthorizerRequest) {
    DescribeAuthorizerRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAuthorizerRequest || (DescribeAuthorizerRequest = {}));
export var DescribeAuthorizerResponse;
(function (DescribeAuthorizerResponse) {
    DescribeAuthorizerResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeAuthorizerResponse || (DescribeAuthorizerResponse = {}));
export var DescribeBillingGroupRequest;
(function (DescribeBillingGroupRequest) {
    DescribeBillingGroupRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeBillingGroupRequest || (DescribeBillingGroupRequest = {}));
export var BillingGroupMetadata;
(function (BillingGroupMetadata) {
    BillingGroupMetadata.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BillingGroupMetadata || (BillingGroupMetadata = {}));
export var DescribeBillingGroupResponse;
(function (DescribeBillingGroupResponse) {
    DescribeBillingGroupResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeBillingGroupResponse || (DescribeBillingGroupResponse = {}));
export var DescribeCACertificateRequest;
(function (DescribeCACertificateRequest) {
    DescribeCACertificateRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCACertificateRequest || (DescribeCACertificateRequest = {}));
export var CACertificateStatus;
(function (CACertificateStatus) {
    CACertificateStatus["ACTIVE"] = "ACTIVE";
    CACertificateStatus["INACTIVE"] = "INACTIVE";
})(CACertificateStatus || (CACertificateStatus = {}));
export var CertificateValidity;
(function (CertificateValidity) {
    CertificateValidity.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CertificateValidity || (CertificateValidity = {}));
export var CACertificateDescription;
(function (CACertificateDescription) {
    CACertificateDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CACertificateDescription || (CACertificateDescription = {}));
export var RegistrationConfig;
(function (RegistrationConfig) {
    RegistrationConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RegistrationConfig || (RegistrationConfig = {}));
export var DescribeCACertificateResponse;
(function (DescribeCACertificateResponse) {
    DescribeCACertificateResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCACertificateResponse || (DescribeCACertificateResponse = {}));
export var DescribeCertificateRequest;
(function (DescribeCertificateRequest) {
    DescribeCertificateRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCertificateRequest || (DescribeCertificateRequest = {}));
export var CertificateMode;
(function (CertificateMode) {
    CertificateMode["DEFAULT"] = "DEFAULT";
    CertificateMode["SNI_ONLY"] = "SNI_ONLY";
})(CertificateMode || (CertificateMode = {}));
export var CertificateStatus;
(function (CertificateStatus) {
    CertificateStatus["ACTIVE"] = "ACTIVE";
    CertificateStatus["INACTIVE"] = "INACTIVE";
    CertificateStatus["PENDING_ACTIVATION"] = "PENDING_ACTIVATION";
    CertificateStatus["PENDING_TRANSFER"] = "PENDING_TRANSFER";
    CertificateStatus["REGISTER_INACTIVE"] = "REGISTER_INACTIVE";
    CertificateStatus["REVOKED"] = "REVOKED";
})(CertificateStatus || (CertificateStatus = {}));
export var TransferData;
(function (TransferData) {
    TransferData.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TransferData || (TransferData = {}));
export var CertificateDescription;
(function (CertificateDescription) {
    CertificateDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CertificateDescription || (CertificateDescription = {}));
export var DescribeCertificateResponse;
(function (DescribeCertificateResponse) {
    DescribeCertificateResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCertificateResponse || (DescribeCertificateResponse = {}));
export var DescribeCustomMetricRequest;
(function (DescribeCustomMetricRequest) {
    DescribeCustomMetricRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCustomMetricRequest || (DescribeCustomMetricRequest = {}));
export var DescribeCustomMetricResponse;
(function (DescribeCustomMetricResponse) {
    DescribeCustomMetricResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCustomMetricResponse || (DescribeCustomMetricResponse = {}));
export var DescribeDefaultAuthorizerRequest;
(function (DescribeDefaultAuthorizerRequest) {
    DescribeDefaultAuthorizerRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDefaultAuthorizerRequest || (DescribeDefaultAuthorizerRequest = {}));
export var DescribeDefaultAuthorizerResponse;
(function (DescribeDefaultAuthorizerResponse) {
    DescribeDefaultAuthorizerResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDefaultAuthorizerResponse || (DescribeDefaultAuthorizerResponse = {}));
export var DescribeDetectMitigationActionsTaskRequest;
(function (DescribeDetectMitigationActionsTaskRequest) {
    DescribeDetectMitigationActionsTaskRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDetectMitigationActionsTaskRequest || (DescribeDetectMitigationActionsTaskRequest = {}));
export var DetectMitigationActionsTaskTarget;
(function (DetectMitigationActionsTaskTarget) {
    DetectMitigationActionsTaskTarget.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetectMitigationActionsTaskTarget || (DetectMitigationActionsTaskTarget = {}));
export var DetectMitigationActionsTaskStatistics;
(function (DetectMitigationActionsTaskStatistics) {
    DetectMitigationActionsTaskStatistics.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetectMitigationActionsTaskStatistics || (DetectMitigationActionsTaskStatistics = {}));
export var DetectMitigationActionsTaskStatus;
(function (DetectMitigationActionsTaskStatus) {
    DetectMitigationActionsTaskStatus["CANCELED"] = "CANCELED";
    DetectMitigationActionsTaskStatus["FAILED"] = "FAILED";
    DetectMitigationActionsTaskStatus["IN_PROGRESS"] = "IN_PROGRESS";
    DetectMitigationActionsTaskStatus["SUCCESSFUL"] = "SUCCESSFUL";
})(DetectMitigationActionsTaskStatus || (DetectMitigationActionsTaskStatus = {}));
export var ViolationEventOccurrenceRange;
(function (ViolationEventOccurrenceRange) {
    ViolationEventOccurrenceRange.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ViolationEventOccurrenceRange || (ViolationEventOccurrenceRange = {}));
export var DetectMitigationActionsTaskSummary;
(function (DetectMitigationActionsTaskSummary) {
    DetectMitigationActionsTaskSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetectMitigationActionsTaskSummary || (DetectMitigationActionsTaskSummary = {}));
export var DescribeDetectMitigationActionsTaskResponse;
(function (DescribeDetectMitigationActionsTaskResponse) {
    DescribeDetectMitigationActionsTaskResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDetectMitigationActionsTaskResponse || (DescribeDetectMitigationActionsTaskResponse = {}));
export var DescribeDimensionRequest;
(function (DescribeDimensionRequest) {
    DescribeDimensionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDimensionRequest || (DescribeDimensionRequest = {}));
export var DescribeDimensionResponse;
(function (DescribeDimensionResponse) {
    DescribeDimensionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDimensionResponse || (DescribeDimensionResponse = {}));
export var DescribeDomainConfigurationRequest;
(function (DescribeDomainConfigurationRequest) {
    DescribeDomainConfigurationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDomainConfigurationRequest || (DescribeDomainConfigurationRequest = {}));
export var DomainConfigurationStatus;
(function (DomainConfigurationStatus) {
    DomainConfigurationStatus["DISABLED"] = "DISABLED";
    DomainConfigurationStatus["ENABLED"] = "ENABLED";
})(DomainConfigurationStatus || (DomainConfigurationStatus = {}));
export var DomainType;
(function (DomainType) {
    DomainType["AWS_MANAGED"] = "AWS_MANAGED";
    DomainType["CUSTOMER_MANAGED"] = "CUSTOMER_MANAGED";
    DomainType["ENDPOINT"] = "ENDPOINT";
})(DomainType || (DomainType = {}));
export var ServerCertificateStatus;
(function (ServerCertificateStatus) {
    ServerCertificateStatus["INVALID"] = "INVALID";
    ServerCertificateStatus["VALID"] = "VALID";
})(ServerCertificateStatus || (ServerCertificateStatus = {}));
export var ServerCertificateSummary;
(function (ServerCertificateSummary) {
    ServerCertificateSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ServerCertificateSummary || (ServerCertificateSummary = {}));
export var DescribeDomainConfigurationResponse;
(function (DescribeDomainConfigurationResponse) {
    DescribeDomainConfigurationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDomainConfigurationResponse || (DescribeDomainConfigurationResponse = {}));
export var DescribeEndpointRequest;
(function (DescribeEndpointRequest) {
    DescribeEndpointRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEndpointRequest || (DescribeEndpointRequest = {}));
export var DescribeEndpointResponse;
(function (DescribeEndpointResponse) {
    DescribeEndpointResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEndpointResponse || (DescribeEndpointResponse = {}));
export var DescribeEventConfigurationsRequest;
(function (DescribeEventConfigurationsRequest) {
    DescribeEventConfigurationsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEventConfigurationsRequest || (DescribeEventConfigurationsRequest = {}));
export var EventType;
(function (EventType) {
    EventType["CA_CERTIFICATE"] = "CA_CERTIFICATE";
    EventType["CERTIFICATE"] = "CERTIFICATE";
    EventType["JOB"] = "JOB";
    EventType["JOB_EXECUTION"] = "JOB_EXECUTION";
    EventType["POLICY"] = "POLICY";
    EventType["THING"] = "THING";
    EventType["THING_GROUP"] = "THING_GROUP";
    EventType["THING_GROUP_HIERARCHY"] = "THING_GROUP_HIERARCHY";
    EventType["THING_GROUP_MEMBERSHIP"] = "THING_GROUP_MEMBERSHIP";
    EventType["THING_TYPE"] = "THING_TYPE";
    EventType["THING_TYPE_ASSOCIATION"] = "THING_TYPE_ASSOCIATION";
})(EventType || (EventType = {}));
export var Configuration;
(function (Configuration) {
    Configuration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Configuration || (Configuration = {}));
export var DescribeEventConfigurationsResponse;
(function (DescribeEventConfigurationsResponse) {
    DescribeEventConfigurationsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEventConfigurationsResponse || (DescribeEventConfigurationsResponse = {}));
export var DescribeFleetMetricRequest;
(function (DescribeFleetMetricRequest) {
    DescribeFleetMetricRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeFleetMetricRequest || (DescribeFleetMetricRequest = {}));
export var DescribeFleetMetricResponse;
(function (DescribeFleetMetricResponse) {
    DescribeFleetMetricResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeFleetMetricResponse || (DescribeFleetMetricResponse = {}));
export var DescribeIndexRequest;
(function (DescribeIndexRequest) {
    DescribeIndexRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeIndexRequest || (DescribeIndexRequest = {}));
export var IndexStatus;
(function (IndexStatus) {
    IndexStatus["ACTIVE"] = "ACTIVE";
    IndexStatus["BUILDING"] = "BUILDING";
    IndexStatus["REBUILDING"] = "REBUILDING";
})(IndexStatus || (IndexStatus = {}));
export var DescribeIndexResponse;
(function (DescribeIndexResponse) {
    DescribeIndexResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeIndexResponse || (DescribeIndexResponse = {}));
export var DescribeJobRequest;
(function (DescribeJobRequest) {
    DescribeJobRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeJobRequest || (DescribeJobRequest = {}));
export var JobProcessDetails;
(function (JobProcessDetails) {
    JobProcessDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(JobProcessDetails || (JobProcessDetails = {}));
export var JobStatus;
(function (JobStatus) {
    JobStatus["CANCELED"] = "CANCELED";
    JobStatus["COMPLETED"] = "COMPLETED";
    JobStatus["DELETION_IN_PROGRESS"] = "DELETION_IN_PROGRESS";
    JobStatus["IN_PROGRESS"] = "IN_PROGRESS";
})(JobStatus || (JobStatus = {}));
export var Job;
(function (Job) {
    Job.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Job || (Job = {}));
export var DescribeJobResponse;
(function (DescribeJobResponse) {
    DescribeJobResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeJobResponse || (DescribeJobResponse = {}));
export var DescribeJobExecutionRequest;
(function (DescribeJobExecutionRequest) {
    DescribeJobExecutionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeJobExecutionRequest || (DescribeJobExecutionRequest = {}));
export var JobExecutionStatus;
(function (JobExecutionStatus) {
    JobExecutionStatus["CANCELED"] = "CANCELED";
    JobExecutionStatus["FAILED"] = "FAILED";
    JobExecutionStatus["IN_PROGRESS"] = "IN_PROGRESS";
    JobExecutionStatus["QUEUED"] = "QUEUED";
    JobExecutionStatus["REJECTED"] = "REJECTED";
    JobExecutionStatus["REMOVED"] = "REMOVED";
    JobExecutionStatus["SUCCEEDED"] = "SUCCEEDED";
    JobExecutionStatus["TIMED_OUT"] = "TIMED_OUT";
})(JobExecutionStatus || (JobExecutionStatus = {}));
export var JobExecutionStatusDetails;
(function (JobExecutionStatusDetails) {
    JobExecutionStatusDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(JobExecutionStatusDetails || (JobExecutionStatusDetails = {}));
export var JobExecution;
(function (JobExecution) {
    JobExecution.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(JobExecution || (JobExecution = {}));
export var DescribeJobExecutionResponse;
(function (DescribeJobExecutionResponse) {
    DescribeJobExecutionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeJobExecutionResponse || (DescribeJobExecutionResponse = {}));
export var DescribeJobTemplateRequest;
(function (DescribeJobTemplateRequest) {
    DescribeJobTemplateRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeJobTemplateRequest || (DescribeJobTemplateRequest = {}));
export var DescribeJobTemplateResponse;
(function (DescribeJobTemplateResponse) {
    DescribeJobTemplateResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeJobTemplateResponse || (DescribeJobTemplateResponse = {}));
export var DescribeManagedJobTemplateRequest;
(function (DescribeManagedJobTemplateRequest) {
    DescribeManagedJobTemplateRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeManagedJobTemplateRequest || (DescribeManagedJobTemplateRequest = {}));
export var DocumentParameter;
(function (DocumentParameter) {
    DocumentParameter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DocumentParameter || (DocumentParameter = {}));
export var DescribeManagedJobTemplateResponse;
(function (DescribeManagedJobTemplateResponse) {
    DescribeManagedJobTemplateResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeManagedJobTemplateResponse || (DescribeManagedJobTemplateResponse = {}));
export var DescribeMitigationActionRequest;
(function (DescribeMitigationActionRequest) {
    DescribeMitigationActionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeMitigationActionRequest || (DescribeMitigationActionRequest = {}));
export var MitigationActionType;
(function (MitigationActionType) {
    MitigationActionType["ADD_THINGS_TO_THING_GROUP"] = "ADD_THINGS_TO_THING_GROUP";
    MitigationActionType["ENABLE_IOT_LOGGING"] = "ENABLE_IOT_LOGGING";
    MitigationActionType["PUBLISH_FINDING_TO_SNS"] = "PUBLISH_FINDING_TO_SNS";
    MitigationActionType["REPLACE_DEFAULT_POLICY_VERSION"] = "REPLACE_DEFAULT_POLICY_VERSION";
    MitigationActionType["UPDATE_CA_CERTIFICATE"] = "UPDATE_CA_CERTIFICATE";
    MitigationActionType["UPDATE_DEVICE_CERTIFICATE"] = "UPDATE_DEVICE_CERTIFICATE";
})(MitigationActionType || (MitigationActionType = {}));
export var DescribeMitigationActionResponse;
(function (DescribeMitigationActionResponse) {
    DescribeMitigationActionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeMitigationActionResponse || (DescribeMitigationActionResponse = {}));
export var DescribeProvisioningTemplateRequest;
(function (DescribeProvisioningTemplateRequest) {
    DescribeProvisioningTemplateRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeProvisioningTemplateRequest || (DescribeProvisioningTemplateRequest = {}));
export var DescribeProvisioningTemplateResponse;
(function (DescribeProvisioningTemplateResponse) {
    DescribeProvisioningTemplateResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeProvisioningTemplateResponse || (DescribeProvisioningTemplateResponse = {}));
export var DescribeProvisioningTemplateVersionRequest;
(function (DescribeProvisioningTemplateVersionRequest) {
    DescribeProvisioningTemplateVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeProvisioningTemplateVersionRequest || (DescribeProvisioningTemplateVersionRequest = {}));
export var DescribeProvisioningTemplateVersionResponse;
(function (DescribeProvisioningTemplateVersionResponse) {
    DescribeProvisioningTemplateVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeProvisioningTemplateVersionResponse || (DescribeProvisioningTemplateVersionResponse = {}));
export var DescribeRoleAliasRequest;
(function (DescribeRoleAliasRequest) {
    DescribeRoleAliasRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeRoleAliasRequest || (DescribeRoleAliasRequest = {}));
export var RoleAliasDescription;
(function (RoleAliasDescription) {
    RoleAliasDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RoleAliasDescription || (RoleAliasDescription = {}));
export var DescribeRoleAliasResponse;
(function (DescribeRoleAliasResponse) {
    DescribeRoleAliasResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeRoleAliasResponse || (DescribeRoleAliasResponse = {}));
export var DescribeScheduledAuditRequest;
(function (DescribeScheduledAuditRequest) {
    DescribeScheduledAuditRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeScheduledAuditRequest || (DescribeScheduledAuditRequest = {}));
export var DescribeScheduledAuditResponse;
(function (DescribeScheduledAuditResponse) {
    DescribeScheduledAuditResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeScheduledAuditResponse || (DescribeScheduledAuditResponse = {}));
export var DescribeSecurityProfileRequest;
(function (DescribeSecurityProfileRequest) {
    DescribeSecurityProfileRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeSecurityProfileRequest || (DescribeSecurityProfileRequest = {}));
export var DescribeSecurityProfileResponse;
(function (DescribeSecurityProfileResponse) {
    DescribeSecurityProfileResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeSecurityProfileResponse || (DescribeSecurityProfileResponse = {}));
export var DescribeStreamRequest;
(function (DescribeStreamRequest) {
    DescribeStreamRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeStreamRequest || (DescribeStreamRequest = {}));
export var StreamInfo;
(function (StreamInfo) {
    StreamInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StreamInfo || (StreamInfo = {}));
export var DescribeStreamResponse;
(function (DescribeStreamResponse) {
    DescribeStreamResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeStreamResponse || (DescribeStreamResponse = {}));
export var DescribeThingRequest;
(function (DescribeThingRequest) {
    DescribeThingRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeThingRequest || (DescribeThingRequest = {}));
export var DescribeThingResponse;
(function (DescribeThingResponse) {
    DescribeThingResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeThingResponse || (DescribeThingResponse = {}));
export var DescribeThingGroupRequest;
(function (DescribeThingGroupRequest) {
    DescribeThingGroupRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeThingGroupRequest || (DescribeThingGroupRequest = {}));
export var DynamicGroupStatus;
(function (DynamicGroupStatus) {
    DynamicGroupStatus["ACTIVE"] = "ACTIVE";
    DynamicGroupStatus["BUILDING"] = "BUILDING";
    DynamicGroupStatus["REBUILDING"] = "REBUILDING";
})(DynamicGroupStatus || (DynamicGroupStatus = {}));
export var GroupNameAndArn;
(function (GroupNameAndArn) {
    GroupNameAndArn.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GroupNameAndArn || (GroupNameAndArn = {}));
export var ThingGroupMetadata;
(function (ThingGroupMetadata) {
    ThingGroupMetadata.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ThingGroupMetadata || (ThingGroupMetadata = {}));
export var DescribeThingGroupResponse;
(function (DescribeThingGroupResponse) {
    DescribeThingGroupResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeThingGroupResponse || (DescribeThingGroupResponse = {}));
export var DescribeThingRegistrationTaskRequest;
(function (DescribeThingRegistrationTaskRequest) {
    DescribeThingRegistrationTaskRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeThingRegistrationTaskRequest || (DescribeThingRegistrationTaskRequest = {}));
export var Status;
(function (Status) {
    Status["Cancelled"] = "Cancelled";
    Status["Cancelling"] = "Cancelling";
    Status["Completed"] = "Completed";
    Status["Failed"] = "Failed";
    Status["InProgress"] = "InProgress";
})(Status || (Status = {}));
export var DescribeThingRegistrationTaskResponse;
(function (DescribeThingRegistrationTaskResponse) {
    DescribeThingRegistrationTaskResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeThingRegistrationTaskResponse || (DescribeThingRegistrationTaskResponse = {}));
export var DescribeThingTypeRequest;
(function (DescribeThingTypeRequest) {
    DescribeThingTypeRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeThingTypeRequest || (DescribeThingTypeRequest = {}));
export var ThingTypeMetadata;
(function (ThingTypeMetadata) {
    ThingTypeMetadata.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ThingTypeMetadata || (ThingTypeMetadata = {}));
export var DescribeThingTypeResponse;
(function (DescribeThingTypeResponse) {
    DescribeThingTypeResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeThingTypeResponse || (DescribeThingTypeResponse = {}));
export var DetachPolicyRequest;
(function (DetachPolicyRequest) {
    DetachPolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetachPolicyRequest || (DetachPolicyRequest = {}));
export var DetachPrincipalPolicyRequest;
(function (DetachPrincipalPolicyRequest) {
    DetachPrincipalPolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetachPrincipalPolicyRequest || (DetachPrincipalPolicyRequest = {}));
export var DetachSecurityProfileRequest;
(function (DetachSecurityProfileRequest) {
    DetachSecurityProfileRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetachSecurityProfileRequest || (DetachSecurityProfileRequest = {}));
export var DetachSecurityProfileResponse;
(function (DetachSecurityProfileResponse) {
    DetachSecurityProfileResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetachSecurityProfileResponse || (DetachSecurityProfileResponse = {}));
export var DetachThingPrincipalRequest;
(function (DetachThingPrincipalRequest) {
    DetachThingPrincipalRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetachThingPrincipalRequest || (DetachThingPrincipalRequest = {}));
export var DetachThingPrincipalResponse;
(function (DetachThingPrincipalResponse) {
    DetachThingPrincipalResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetachThingPrincipalResponse || (DetachThingPrincipalResponse = {}));
export var DisableTopicRuleRequest;
(function (DisableTopicRuleRequest) {
    DisableTopicRuleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisableTopicRuleRequest || (DisableTopicRuleRequest = {}));
export var EnableTopicRuleRequest;
(function (EnableTopicRuleRequest) {
    EnableTopicRuleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EnableTopicRuleRequest || (EnableTopicRuleRequest = {}));
export var GetBehaviorModelTrainingSummariesRequest;
(function (GetBehaviorModelTrainingSummariesRequest) {
    GetBehaviorModelTrainingSummariesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetBehaviorModelTrainingSummariesRequest || (GetBehaviorModelTrainingSummariesRequest = {}));
export var ModelStatus;
(function (ModelStatus) {
    ModelStatus["ACTIVE"] = "ACTIVE";
    ModelStatus["EXPIRED"] = "EXPIRED";
    ModelStatus["PENDING_BUILD"] = "PENDING_BUILD";
})(ModelStatus || (ModelStatus = {}));
export var BehaviorModelTrainingSummary;
(function (BehaviorModelTrainingSummary) {
    BehaviorModelTrainingSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BehaviorModelTrainingSummary || (BehaviorModelTrainingSummary = {}));
export var GetBehaviorModelTrainingSummariesResponse;
(function (GetBehaviorModelTrainingSummariesResponse) {
    GetBehaviorModelTrainingSummariesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetBehaviorModelTrainingSummariesResponse || (GetBehaviorModelTrainingSummariesResponse = {}));
export var TermsAggregation;
(function (TermsAggregation) {
    TermsAggregation.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TermsAggregation || (TermsAggregation = {}));
export var BucketsAggregationType;
(function (BucketsAggregationType) {
    BucketsAggregationType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BucketsAggregationType || (BucketsAggregationType = {}));
export var GetBucketsAggregationRequest;
(function (GetBucketsAggregationRequest) {
    GetBucketsAggregationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetBucketsAggregationRequest || (GetBucketsAggregationRequest = {}));
export var Bucket;
(function (Bucket) {
    Bucket.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Bucket || (Bucket = {}));
export var GetBucketsAggregationResponse;
(function (GetBucketsAggregationResponse) {
    GetBucketsAggregationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetBucketsAggregationResponse || (GetBucketsAggregationResponse = {}));
export var GetCardinalityRequest;
(function (GetCardinalityRequest) {
    GetCardinalityRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetCardinalityRequest || (GetCardinalityRequest = {}));
export var GetCardinalityResponse;
(function (GetCardinalityResponse) {
    GetCardinalityResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetCardinalityResponse || (GetCardinalityResponse = {}));
export var GetEffectivePoliciesRequest;
(function (GetEffectivePoliciesRequest) {
    GetEffectivePoliciesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetEffectivePoliciesRequest || (GetEffectivePoliciesRequest = {}));
export var EffectivePolicy;
(function (EffectivePolicy) {
    EffectivePolicy.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EffectivePolicy || (EffectivePolicy = {}));
export var GetEffectivePoliciesResponse;
(function (GetEffectivePoliciesResponse) {
    GetEffectivePoliciesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetEffectivePoliciesResponse || (GetEffectivePoliciesResponse = {}));
export var GetIndexingConfigurationRequest;
(function (GetIndexingConfigurationRequest) {
    GetIndexingConfigurationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetIndexingConfigurationRequest || (GetIndexingConfigurationRequest = {}));
export var FieldType;
(function (FieldType) {
    FieldType["BOOLEAN"] = "Boolean";
    FieldType["NUMBER"] = "Number";
    FieldType["STRING"] = "String";
})(FieldType || (FieldType = {}));
export var Field;
(function (Field) {
    Field.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Field || (Field = {}));
export var ThingGroupIndexingMode;
(function (ThingGroupIndexingMode) {
    ThingGroupIndexingMode["OFF"] = "OFF";
    ThingGroupIndexingMode["ON"] = "ON";
})(ThingGroupIndexingMode || (ThingGroupIndexingMode = {}));
export var ThingGroupIndexingConfiguration;
(function (ThingGroupIndexingConfiguration) {
    ThingGroupIndexingConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ThingGroupIndexingConfiguration || (ThingGroupIndexingConfiguration = {}));
export var DeviceDefenderIndexingMode;
(function (DeviceDefenderIndexingMode) {
    DeviceDefenderIndexingMode["OFF"] = "OFF";
    DeviceDefenderIndexingMode["VIOLATIONS"] = "VIOLATIONS";
})(DeviceDefenderIndexingMode || (DeviceDefenderIndexingMode = {}));
export var NamedShadowIndexingMode;
(function (NamedShadowIndexingMode) {
    NamedShadowIndexingMode["OFF"] = "OFF";
    NamedShadowIndexingMode["ON"] = "ON";
})(NamedShadowIndexingMode || (NamedShadowIndexingMode = {}));
export var ThingConnectivityIndexingMode;
(function (ThingConnectivityIndexingMode) {
    ThingConnectivityIndexingMode["OFF"] = "OFF";
    ThingConnectivityIndexingMode["STATUS"] = "STATUS";
})(ThingConnectivityIndexingMode || (ThingConnectivityIndexingMode = {}));
export var ThingIndexingMode;
(function (ThingIndexingMode) {
    ThingIndexingMode["OFF"] = "OFF";
    ThingIndexingMode["REGISTRY"] = "REGISTRY";
    ThingIndexingMode["REGISTRY_AND_SHADOW"] = "REGISTRY_AND_SHADOW";
})(ThingIndexingMode || (ThingIndexingMode = {}));
export var ThingIndexingConfiguration;
(function (ThingIndexingConfiguration) {
    ThingIndexingConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ThingIndexingConfiguration || (ThingIndexingConfiguration = {}));
export var GetIndexingConfigurationResponse;
(function (GetIndexingConfigurationResponse) {
    GetIndexingConfigurationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetIndexingConfigurationResponse || (GetIndexingConfigurationResponse = {}));
export var GetJobDocumentRequest;
(function (GetJobDocumentRequest) {
    GetJobDocumentRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetJobDocumentRequest || (GetJobDocumentRequest = {}));
export var GetJobDocumentResponse;
(function (GetJobDocumentResponse) {
    GetJobDocumentResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetJobDocumentResponse || (GetJobDocumentResponse = {}));
export var GetLoggingOptionsRequest;
(function (GetLoggingOptionsRequest) {
    GetLoggingOptionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetLoggingOptionsRequest || (GetLoggingOptionsRequest = {}));
export var GetLoggingOptionsResponse;
(function (GetLoggingOptionsResponse) {
    GetLoggingOptionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetLoggingOptionsResponse || (GetLoggingOptionsResponse = {}));
export var GetOTAUpdateRequest;
(function (GetOTAUpdateRequest) {
    GetOTAUpdateRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetOTAUpdateRequest || (GetOTAUpdateRequest = {}));
export var ErrorInfo;
(function (ErrorInfo) {
    ErrorInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ErrorInfo || (ErrorInfo = {}));
export var OTAUpdateInfo;
(function (OTAUpdateInfo) {
    OTAUpdateInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OTAUpdateInfo || (OTAUpdateInfo = {}));
export var GetOTAUpdateResponse;
(function (GetOTAUpdateResponse) {
    GetOTAUpdateResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetOTAUpdateResponse || (GetOTAUpdateResponse = {}));
export var GetPercentilesRequest;
(function (GetPercentilesRequest) {
    GetPercentilesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetPercentilesRequest || (GetPercentilesRequest = {}));
export var PercentPair;
(function (PercentPair) {
    PercentPair.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PercentPair || (PercentPair = {}));
export var GetPercentilesResponse;
(function (GetPercentilesResponse) {
    GetPercentilesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetPercentilesResponse || (GetPercentilesResponse = {}));
export var GetPolicyRequest;
(function (GetPolicyRequest) {
    GetPolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetPolicyRequest || (GetPolicyRequest = {}));
export var GetPolicyResponse;
(function (GetPolicyResponse) {
    GetPolicyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetPolicyResponse || (GetPolicyResponse = {}));
export var GetPolicyVersionRequest;
(function (GetPolicyVersionRequest) {
    GetPolicyVersionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetPolicyVersionRequest || (GetPolicyVersionRequest = {}));
export var GetPolicyVersionResponse;
(function (GetPolicyVersionResponse) {
    GetPolicyVersionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetPolicyVersionResponse || (GetPolicyVersionResponse = {}));
export var GetRegistrationCodeRequest;
(function (GetRegistrationCodeRequest) {
    GetRegistrationCodeRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetRegistrationCodeRequest || (GetRegistrationCodeRequest = {}));
export var GetRegistrationCodeResponse;
(function (GetRegistrationCodeResponse) {
    GetRegistrationCodeResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetRegistrationCodeResponse || (GetRegistrationCodeResponse = {}));
export var GetStatisticsRequest;
(function (GetStatisticsRequest) {
    GetStatisticsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetStatisticsRequest || (GetStatisticsRequest = {}));
export var Statistics;
(function (Statistics) {
    Statistics.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Statistics || (Statistics = {}));
export var GetStatisticsResponse;
(function (GetStatisticsResponse) {
    GetStatisticsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetStatisticsResponse || (GetStatisticsResponse = {}));
export var GetTopicRuleRequest;
(function (GetTopicRuleRequest) {
    GetTopicRuleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetTopicRuleRequest || (GetTopicRuleRequest = {}));
export var TopicRule;
(function (TopicRule) {
    TopicRule.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TopicRule || (TopicRule = {}));
export var GetTopicRuleResponse;
(function (GetTopicRuleResponse) {
    GetTopicRuleResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetTopicRuleResponse || (GetTopicRuleResponse = {}));
export var GetTopicRuleDestinationRequest;
(function (GetTopicRuleDestinationRequest) {
    GetTopicRuleDestinationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetTopicRuleDestinationRequest || (GetTopicRuleDestinationRequest = {}));
export var GetTopicRuleDestinationResponse;
(function (GetTopicRuleDestinationResponse) {
    GetTopicRuleDestinationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetTopicRuleDestinationResponse || (GetTopicRuleDestinationResponse = {}));
export var GetV2LoggingOptionsRequest;
(function (GetV2LoggingOptionsRequest) {
    GetV2LoggingOptionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetV2LoggingOptionsRequest || (GetV2LoggingOptionsRequest = {}));
export var GetV2LoggingOptionsResponse;
(function (GetV2LoggingOptionsResponse) {
    GetV2LoggingOptionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetV2LoggingOptionsResponse || (GetV2LoggingOptionsResponse = {}));
export var BehaviorCriteriaType;
(function (BehaviorCriteriaType) {
    BehaviorCriteriaType["MACHINE_LEARNING"] = "MACHINE_LEARNING";
    BehaviorCriteriaType["STATIC"] = "STATIC";
    BehaviorCriteriaType["STATISTICAL"] = "STATISTICAL";
})(BehaviorCriteriaType || (BehaviorCriteriaType = {}));
export var ListActiveViolationsRequest;
(function (ListActiveViolationsRequest) {
    ListActiveViolationsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListActiveViolationsRequest || (ListActiveViolationsRequest = {}));
export var ListActiveViolationsResponse;
(function (ListActiveViolationsResponse) {
    ListActiveViolationsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListActiveViolationsResponse || (ListActiveViolationsResponse = {}));
export var ListAttachedPoliciesRequest;
(function (ListAttachedPoliciesRequest) {
    ListAttachedPoliciesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAttachedPoliciesRequest || (ListAttachedPoliciesRequest = {}));
export var ListAttachedPoliciesResponse;
(function (ListAttachedPoliciesResponse) {
    ListAttachedPoliciesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAttachedPoliciesResponse || (ListAttachedPoliciesResponse = {}));
export var ListAuditFindingsRequest;
(function (ListAuditFindingsRequest) {
    ListAuditFindingsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAuditFindingsRequest || (ListAuditFindingsRequest = {}));
export var ListAuditFindingsResponse;
(function (ListAuditFindingsResponse) {
    ListAuditFindingsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAuditFindingsResponse || (ListAuditFindingsResponse = {}));
export var ListAuditMitigationActionsExecutionsRequest;
(function (ListAuditMitigationActionsExecutionsRequest) {
    ListAuditMitigationActionsExecutionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAuditMitigationActionsExecutionsRequest || (ListAuditMitigationActionsExecutionsRequest = {}));
export var ListAuditMitigationActionsExecutionsResponse;
(function (ListAuditMitigationActionsExecutionsResponse) {
    ListAuditMitigationActionsExecutionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAuditMitigationActionsExecutionsResponse || (ListAuditMitigationActionsExecutionsResponse = {}));
export var ListAuditMitigationActionsTasksRequest;
(function (ListAuditMitigationActionsTasksRequest) {
    ListAuditMitigationActionsTasksRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAuditMitigationActionsTasksRequest || (ListAuditMitigationActionsTasksRequest = {}));
export var ListAuditMitigationActionsTasksResponse;
(function (ListAuditMitigationActionsTasksResponse) {
    ListAuditMitigationActionsTasksResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAuditMitigationActionsTasksResponse || (ListAuditMitigationActionsTasksResponse = {}));
export var ListAuditSuppressionsRequest;
(function (ListAuditSuppressionsRequest) {
    ListAuditSuppressionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAuditSuppressionsRequest || (ListAuditSuppressionsRequest = {}));
export var ListAuditSuppressionsResponse;
(function (ListAuditSuppressionsResponse) {
    ListAuditSuppressionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAuditSuppressionsResponse || (ListAuditSuppressionsResponse = {}));
export var ListAuditTasksRequest;
(function (ListAuditTasksRequest) {
    ListAuditTasksRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAuditTasksRequest || (ListAuditTasksRequest = {}));
export var ListAuditTasksResponse;
(function (ListAuditTasksResponse) {
    ListAuditTasksResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAuditTasksResponse || (ListAuditTasksResponse = {}));
export var ListAuthorizersRequest;
(function (ListAuthorizersRequest) {
    ListAuthorizersRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAuthorizersRequest || (ListAuthorizersRequest = {}));
export var ListAuthorizersResponse;
(function (ListAuthorizersResponse) {
    ListAuthorizersResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAuthorizersResponse || (ListAuthorizersResponse = {}));
export var ListBillingGroupsRequest;
(function (ListBillingGroupsRequest) {
    ListBillingGroupsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListBillingGroupsRequest || (ListBillingGroupsRequest = {}));
export var ListBillingGroupsResponse;
(function (ListBillingGroupsResponse) {
    ListBillingGroupsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListBillingGroupsResponse || (ListBillingGroupsResponse = {}));
export var ListCACertificatesRequest;
(function (ListCACertificatesRequest) {
    ListCACertificatesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListCACertificatesRequest || (ListCACertificatesRequest = {}));
export var CACertificate;
(function (CACertificate) {
    CACertificate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CACertificate || (CACertificate = {}));
export var ListCACertificatesResponse;
(function (ListCACertificatesResponse) {
    ListCACertificatesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListCACertificatesResponse || (ListCACertificatesResponse = {}));
export var ListCertificatesRequest;
(function (ListCertificatesRequest) {
    ListCertificatesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListCertificatesRequest || (ListCertificatesRequest = {}));
export var Certificate;
(function (Certificate) {
    Certificate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Certificate || (Certificate = {}));
export var ListCertificatesResponse;
(function (ListCertificatesResponse) {
    ListCertificatesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListCertificatesResponse || (ListCertificatesResponse = {}));
export var ListCertificatesByCARequest;
(function (ListCertificatesByCARequest) {
    ListCertificatesByCARequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListCertificatesByCARequest || (ListCertificatesByCARequest = {}));
export var ListCertificatesByCAResponse;
(function (ListCertificatesByCAResponse) {
    ListCertificatesByCAResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListCertificatesByCAResponse || (ListCertificatesByCAResponse = {}));
export var ListCustomMetricsRequest;
(function (ListCustomMetricsRequest) {
    ListCustomMetricsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListCustomMetricsRequest || (ListCustomMetricsRequest = {}));
export var ListCustomMetricsResponse;
(function (ListCustomMetricsResponse) {
    ListCustomMetricsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListCustomMetricsResponse || (ListCustomMetricsResponse = {}));
export var ListDetectMitigationActionsExecutionsRequest;
(function (ListDetectMitigationActionsExecutionsRequest) {
    ListDetectMitigationActionsExecutionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDetectMitigationActionsExecutionsRequest || (ListDetectMitigationActionsExecutionsRequest = {}));
export var DetectMitigationActionExecutionStatus;
(function (DetectMitigationActionExecutionStatus) {
    DetectMitigationActionExecutionStatus["FAILED"] = "FAILED";
    DetectMitigationActionExecutionStatus["IN_PROGRESS"] = "IN_PROGRESS";
    DetectMitigationActionExecutionStatus["SKIPPED"] = "SKIPPED";
    DetectMitigationActionExecutionStatus["SUCCESSFUL"] = "SUCCESSFUL";
})(DetectMitigationActionExecutionStatus || (DetectMitigationActionExecutionStatus = {}));
export var DetectMitigationActionExecution;
(function (DetectMitigationActionExecution) {
    DetectMitigationActionExecution.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DetectMitigationActionExecution || (DetectMitigationActionExecution = {}));
export var ListDetectMitigationActionsExecutionsResponse;
(function (ListDetectMitigationActionsExecutionsResponse) {
    ListDetectMitigationActionsExecutionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDetectMitigationActionsExecutionsResponse || (ListDetectMitigationActionsExecutionsResponse = {}));
export var ListDetectMitigationActionsTasksRequest;
(function (ListDetectMitigationActionsTasksRequest) {
    ListDetectMitigationActionsTasksRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDetectMitigationActionsTasksRequest || (ListDetectMitigationActionsTasksRequest = {}));
export var ListDetectMitigationActionsTasksResponse;
(function (ListDetectMitigationActionsTasksResponse) {
    ListDetectMitigationActionsTasksResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDetectMitigationActionsTasksResponse || (ListDetectMitigationActionsTasksResponse = {}));
export var ListDimensionsRequest;
(function (ListDimensionsRequest) {
    ListDimensionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDimensionsRequest || (ListDimensionsRequest = {}));
export var ListDimensionsResponse;
(function (ListDimensionsResponse) {
    ListDimensionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDimensionsResponse || (ListDimensionsResponse = {}));
export var ListDomainConfigurationsRequest;
(function (ListDomainConfigurationsRequest) {
    ListDomainConfigurationsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDomainConfigurationsRequest || (ListDomainConfigurationsRequest = {}));
export var DomainConfigurationSummary;
(function (DomainConfigurationSummary) {
    DomainConfigurationSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DomainConfigurationSummary || (DomainConfigurationSummary = {}));
export var ListDomainConfigurationsResponse;
(function (ListDomainConfigurationsResponse) {
    ListDomainConfigurationsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDomainConfigurationsResponse || (ListDomainConfigurationsResponse = {}));
export var ListFleetMetricsRequest;
(function (ListFleetMetricsRequest) {
    ListFleetMetricsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListFleetMetricsRequest || (ListFleetMetricsRequest = {}));
export var FleetMetricNameAndArn;
(function (FleetMetricNameAndArn) {
    FleetMetricNameAndArn.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FleetMetricNameAndArn || (FleetMetricNameAndArn = {}));
export var ListFleetMetricsResponse;
(function (ListFleetMetricsResponse) {
    ListFleetMetricsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListFleetMetricsResponse || (ListFleetMetricsResponse = {}));
export var ListIndicesRequest;
(function (ListIndicesRequest) {
    ListIndicesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListIndicesRequest || (ListIndicesRequest = {}));
export var ListIndicesResponse;
(function (ListIndicesResponse) {
    ListIndicesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListIndicesResponse || (ListIndicesResponse = {}));
export var ListJobExecutionsForJobRequest;
(function (ListJobExecutionsForJobRequest) {
    ListJobExecutionsForJobRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListJobExecutionsForJobRequest || (ListJobExecutionsForJobRequest = {}));
export var JobExecutionSummary;
(function (JobExecutionSummary) {
    JobExecutionSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(JobExecutionSummary || (JobExecutionSummary = {}));
export var JobExecutionSummaryForJob;
(function (JobExecutionSummaryForJob) {
    JobExecutionSummaryForJob.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(JobExecutionSummaryForJob || (JobExecutionSummaryForJob = {}));
export var ListJobExecutionsForJobResponse;
(function (ListJobExecutionsForJobResponse) {
    ListJobExecutionsForJobResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListJobExecutionsForJobResponse || (ListJobExecutionsForJobResponse = {}));
export var ListJobExecutionsForThingRequest;
(function (ListJobExecutionsForThingRequest) {
    ListJobExecutionsForThingRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListJobExecutionsForThingRequest || (ListJobExecutionsForThingRequest = {}));
export var JobExecutionSummaryForThing;
(function (JobExecutionSummaryForThing) {
    JobExecutionSummaryForThing.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(JobExecutionSummaryForThing || (JobExecutionSummaryForThing = {}));
export var ListJobExecutionsForThingResponse;
(function (ListJobExecutionsForThingResponse) {
    ListJobExecutionsForThingResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListJobExecutionsForThingResponse || (ListJobExecutionsForThingResponse = {}));
export var ListJobsRequest;
(function (ListJobsRequest) {
    ListJobsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListJobsRequest || (ListJobsRequest = {}));
export var JobSummary;
(function (JobSummary) {
    JobSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(JobSummary || (JobSummary = {}));
export var ListJobsResponse;
(function (ListJobsResponse) {
    ListJobsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListJobsResponse || (ListJobsResponse = {}));
export var ListJobTemplatesRequest;
(function (ListJobTemplatesRequest) {
    ListJobTemplatesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListJobTemplatesRequest || (ListJobTemplatesRequest = {}));
export var JobTemplateSummary;
(function (JobTemplateSummary) {
    JobTemplateSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(JobTemplateSummary || (JobTemplateSummary = {}));
export var ListJobTemplatesResponse;
(function (ListJobTemplatesResponse) {
    ListJobTemplatesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListJobTemplatesResponse || (ListJobTemplatesResponse = {}));
export var ListManagedJobTemplatesRequest;
(function (ListManagedJobTemplatesRequest) {
    ListManagedJobTemplatesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListManagedJobTemplatesRequest || (ListManagedJobTemplatesRequest = {}));
export var ManagedJobTemplateSummary;
(function (ManagedJobTemplateSummary) {
    ManagedJobTemplateSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ManagedJobTemplateSummary || (ManagedJobTemplateSummary = {}));
export var ListManagedJobTemplatesResponse;
(function (ListManagedJobTemplatesResponse) {
    ListManagedJobTemplatesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListManagedJobTemplatesResponse || (ListManagedJobTemplatesResponse = {}));
export var ListMitigationActionsRequest;
(function (ListMitigationActionsRequest) {
    ListMitigationActionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListMitigationActionsRequest || (ListMitigationActionsRequest = {}));
export var MitigationActionIdentifier;
(function (MitigationActionIdentifier) {
    MitigationActionIdentifier.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MitigationActionIdentifier || (MitigationActionIdentifier = {}));
export var ListMitigationActionsResponse;
(function (ListMitigationActionsResponse) {
    ListMitigationActionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListMitigationActionsResponse || (ListMitigationActionsResponse = {}));
export var ListOTAUpdatesRequest;
(function (ListOTAUpdatesRequest) {
    ListOTAUpdatesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListOTAUpdatesRequest || (ListOTAUpdatesRequest = {}));
export var OTAUpdateSummary;
(function (OTAUpdateSummary) {
    OTAUpdateSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OTAUpdateSummary || (OTAUpdateSummary = {}));
export var ListOTAUpdatesResponse;
(function (ListOTAUpdatesResponse) {
    ListOTAUpdatesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListOTAUpdatesResponse || (ListOTAUpdatesResponse = {}));
export var ListOutgoingCertificatesRequest;
(function (ListOutgoingCertificatesRequest) {
    ListOutgoingCertificatesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListOutgoingCertificatesRequest || (ListOutgoingCertificatesRequest = {}));
export var OutgoingCertificate;
(function (OutgoingCertificate) {
    OutgoingCertificate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OutgoingCertificate || (OutgoingCertificate = {}));
export var ListOutgoingCertificatesResponse;
(function (ListOutgoingCertificatesResponse) {
    ListOutgoingCertificatesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListOutgoingCertificatesResponse || (ListOutgoingCertificatesResponse = {}));
export var ListPoliciesRequest;
(function (ListPoliciesRequest) {
    ListPoliciesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPoliciesRequest || (ListPoliciesRequest = {}));
export var ListPoliciesResponse;
(function (ListPoliciesResponse) {
    ListPoliciesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPoliciesResponse || (ListPoliciesResponse = {}));
export var ListPolicyPrincipalsRequest;
(function (ListPolicyPrincipalsRequest) {
    ListPolicyPrincipalsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPolicyPrincipalsRequest || (ListPolicyPrincipalsRequest = {}));
export var ListPolicyPrincipalsResponse;
(function (ListPolicyPrincipalsResponse) {
    ListPolicyPrincipalsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPolicyPrincipalsResponse || (ListPolicyPrincipalsResponse = {}));
export var ListPolicyVersionsRequest;
(function (ListPolicyVersionsRequest) {
    ListPolicyVersionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPolicyVersionsRequest || (ListPolicyVersionsRequest = {}));
export var PolicyVersion;
(function (PolicyVersion) {
    PolicyVersion.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PolicyVersion || (PolicyVersion = {}));
export var ListPolicyVersionsResponse;
(function (ListPolicyVersionsResponse) {
    ListPolicyVersionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPolicyVersionsResponse || (ListPolicyVersionsResponse = {}));
export var ListPrincipalPoliciesRequest;
(function (ListPrincipalPoliciesRequest) {
    ListPrincipalPoliciesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPrincipalPoliciesRequest || (ListPrincipalPoliciesRequest = {}));
export var ListPrincipalPoliciesResponse;
(function (ListPrincipalPoliciesResponse) {
    ListPrincipalPoliciesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPrincipalPoliciesResponse || (ListPrincipalPoliciesResponse = {}));
export var ListPrincipalThingsRequest;
(function (ListPrincipalThingsRequest) {
    ListPrincipalThingsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPrincipalThingsRequest || (ListPrincipalThingsRequest = {}));
export var ListPrincipalThingsResponse;
(function (ListPrincipalThingsResponse) {
    ListPrincipalThingsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPrincipalThingsResponse || (ListPrincipalThingsResponse = {}));
export var ListProvisioningTemplatesRequest;
(function (ListProvisioningTemplatesRequest) {
    ListProvisioningTemplatesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListProvisioningTemplatesRequest || (ListProvisioningTemplatesRequest = {}));
export var ProvisioningTemplateSummary;
(function (ProvisioningTemplateSummary) {
    ProvisioningTemplateSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ProvisioningTemplateSummary || (ProvisioningTemplateSummary = {}));
export var ListProvisioningTemplatesResponse;
(function (ListProvisioningTemplatesResponse) {
    ListProvisioningTemplatesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListProvisioningTemplatesResponse || (ListProvisioningTemplatesResponse = {}));
export var ListProvisioningTemplateVersionsRequest;
(function (ListProvisioningTemplateVersionsRequest) {
    ListProvisioningTemplateVersionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListProvisioningTemplateVersionsRequest || (ListProvisioningTemplateVersionsRequest = {}));
export var ProvisioningTemplateVersionSummary;
(function (ProvisioningTemplateVersionSummary) {
    ProvisioningTemplateVersionSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ProvisioningTemplateVersionSummary || (ProvisioningTemplateVersionSummary = {}));
export var ListProvisioningTemplateVersionsResponse;
(function (ListProvisioningTemplateVersionsResponse) {
    ListProvisioningTemplateVersionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListProvisioningTemplateVersionsResponse || (ListProvisioningTemplateVersionsResponse = {}));
export var ListRoleAliasesRequest;
(function (ListRoleAliasesRequest) {
    ListRoleAliasesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListRoleAliasesRequest || (ListRoleAliasesRequest = {}));
export var ListRoleAliasesResponse;
(function (ListRoleAliasesResponse) {
    ListRoleAliasesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListRoleAliasesResponse || (ListRoleAliasesResponse = {}));
export var ListScheduledAuditsRequest;
(function (ListScheduledAuditsRequest) {
    ListScheduledAuditsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListScheduledAuditsRequest || (ListScheduledAuditsRequest = {}));
export var ScheduledAuditMetadata;
(function (ScheduledAuditMetadata) {
    ScheduledAuditMetadata.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ScheduledAuditMetadata || (ScheduledAuditMetadata = {}));
export var ListScheduledAuditsResponse;
(function (ListScheduledAuditsResponse) {
    ListScheduledAuditsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListScheduledAuditsResponse || (ListScheduledAuditsResponse = {}));
export var ListSecurityProfilesRequest;
(function (ListSecurityProfilesRequest) {
    ListSecurityProfilesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSecurityProfilesRequest || (ListSecurityProfilesRequest = {}));
export var SecurityProfileIdentifier;
(function (SecurityProfileIdentifier) {
    SecurityProfileIdentifier.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SecurityProfileIdentifier || (SecurityProfileIdentifier = {}));
export var ListSecurityProfilesResponse;
(function (ListSecurityProfilesResponse) {
    ListSecurityProfilesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSecurityProfilesResponse || (ListSecurityProfilesResponse = {}));
export var ListSecurityProfilesForTargetRequest;
(function (ListSecurityProfilesForTargetRequest) {
    ListSecurityProfilesForTargetRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSecurityProfilesForTargetRequest || (ListSecurityProfilesForTargetRequest = {}));
export var SecurityProfileTarget;
(function (SecurityProfileTarget) {
    SecurityProfileTarget.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SecurityProfileTarget || (SecurityProfileTarget = {}));
export var SecurityProfileTargetMapping;
(function (SecurityProfileTargetMapping) {
    SecurityProfileTargetMapping.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SecurityProfileTargetMapping || (SecurityProfileTargetMapping = {}));
export var ListSecurityProfilesForTargetResponse;
(function (ListSecurityProfilesForTargetResponse) {
    ListSecurityProfilesForTargetResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSecurityProfilesForTargetResponse || (ListSecurityProfilesForTargetResponse = {}));
export var ListStreamsRequest;
(function (ListStreamsRequest) {
    ListStreamsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListStreamsRequest || (ListStreamsRequest = {}));
export var StreamSummary;
(function (StreamSummary) {
    StreamSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StreamSummary || (StreamSummary = {}));
export var ListStreamsResponse;
(function (ListStreamsResponse) {
    ListStreamsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListStreamsResponse || (ListStreamsResponse = {}));
export var ListTagsForResourceRequest;
(function (ListTagsForResourceRequest) {
    ListTagsForResourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsForResourceRequest || (ListTagsForResourceRequest = {}));
export var ListTagsForResourceResponse;
(function (ListTagsForResourceResponse) {
    ListTagsForResourceResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsForResourceResponse || (ListTagsForResourceResponse = {}));
export var ListTargetsForPolicyRequest;
(function (ListTargetsForPolicyRequest) {
    ListTargetsForPolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTargetsForPolicyRequest || (ListTargetsForPolicyRequest = {}));
export var ListTargetsForPolicyResponse;
(function (ListTargetsForPolicyResponse) {
    ListTargetsForPolicyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTargetsForPolicyResponse || (ListTargetsForPolicyResponse = {}));
export var ListTargetsForSecurityProfileRequest;
(function (ListTargetsForSecurityProfileRequest) {
    ListTargetsForSecurityProfileRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTargetsForSecurityProfileRequest || (ListTargetsForSecurityProfileRequest = {}));
export var ListTargetsForSecurityProfileResponse;
(function (ListTargetsForSecurityProfileResponse) {
    ListTargetsForSecurityProfileResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTargetsForSecurityProfileResponse || (ListTargetsForSecurityProfileResponse = {}));
export var ListThingGroupsRequest;
(function (ListThingGroupsRequest) {
    ListThingGroupsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListThingGroupsRequest || (ListThingGroupsRequest = {}));
export var ListThingGroupsResponse;
(function (ListThingGroupsResponse) {
    ListThingGroupsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListThingGroupsResponse || (ListThingGroupsResponse = {}));
export var ListThingGroupsForThingRequest;
(function (ListThingGroupsForThingRequest) {
    ListThingGroupsForThingRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListThingGroupsForThingRequest || (ListThingGroupsForThingRequest = {}));
export var ListThingGroupsForThingResponse;
(function (ListThingGroupsForThingResponse) {
    ListThingGroupsForThingResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListThingGroupsForThingResponse || (ListThingGroupsForThingResponse = {}));
export var ListThingPrincipalsRequest;
(function (ListThingPrincipalsRequest) {
    ListThingPrincipalsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListThingPrincipalsRequest || (ListThingPrincipalsRequest = {}));
export var ListThingPrincipalsResponse;
(function (ListThingPrincipalsResponse) {
    ListThingPrincipalsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListThingPrincipalsResponse || (ListThingPrincipalsResponse = {}));
export var ReportType;
(function (ReportType) {
    ReportType["ERRORS"] = "ERRORS";
    ReportType["RESULTS"] = "RESULTS";
})(ReportType || (ReportType = {}));
